{
  "nbformat": 4,
  "nbformat_minor": 0,
  "metadata": {
    "colab": {
      "name": " Genomic-Analysis-WES2.ipynb",
      "provenance": [],
      "collapsed_sections": [],
      "include_colab_link": true
    },
    "kernelspec": {
      "name": "python3",
      "display_name": "Python 3"
    },
    "language_info": {
      "name": "python"
    }
  },
  "cells": [
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "view-in-github",
        "colab_type": "text"
      },
      "source": [
        "<a href=\"https://colab.research.google.com/github/pegahahadian/omics/blob/main/Genomic_Analysis_WES2.ipynb\" target=\"_parent\"><img src=\"https://colab.research.google.com/assets/colab-badge.svg\" alt=\"Open In Colab\"/></a>"
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "**connect to your google drive (optional)**"
      ],
      "metadata": {
        "id": "5T4lNa9RWYDW"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "from google.colab import drive\n",
        "drive.mount('/content/drive')"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "VTUBLtO2Xep_",
        "outputId": "f0ed0dfc-1df0-41be-efbb-b5aecc7ef02e"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Mounted at /content/drive\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!wget ftp://gsapubftp-anonymous@ftp.broadinstitute.org/bundle/hg19/Mills_and_1000G_gold_standard.indels.hg19.sites.vcf.gz\n",
        "!wget ftp://gsapubftp-anonymous@ftp.broadinstitute.org/bundle/hg19/Mills_and_1000G_gold_standard.indels.hg19.sites.vcf.idx.gz\n",
        "!wget ftp://gsapubftp-anonymous@ftp.broadinstitute.org/bundle/hg19/hapmap_3.3_hg19_pop_stratified_af.vcf.gz\n",
        "!wget ftp://gsapubftp-anonymous@ftp.broadinstitute.org/bundle/hg19/hapmap_3.3_hg19_pop_stratified_af.vcf.gz.tbi\n",
        "!wget ftp://gsapubftp-anonymous@ftp.broadinstitute.org/bundle/hg19/hapmap_3.3.hg19.sites.vcf.idx.gz\n",
        "!wget ftp://gsapubftp-anonymous@ftp.broadinstitute.org/bundle/hg19/hapmap_3.3.hg19.sites.vcf.gz\n",
        "!wget ftp://gsapubftp-anonymous@ftp.broadinstitute.org/bundle/hg19/dbsnp_138.hg19.vcf.idx.gz\n",
        "!wget ftp://gsapubftp-anonymous@ftp.broadinstitute.org/bundle/hg19/dbsnp_138.hg19.vcf.gz\n",
        "!wget ftp://gsapubftp-anonymous@ftp.broadinstitute.org/bundle/hg19/1000G_phase1.snps.high_confidence.hg19.sites.vcf.idx.gz\n",
        "!wget ftp://gsapubftp-anonymous@ftp.broadinstitute.org/bundle/hg19/1000G_phase1.snps.high_confidence.hg19.sites.vcf.gz\n",
        "!wget ftp://gsapubftp-anonymous@ftp.broadinstitute.org/bundle/hg19/1000G_phase1.indels.hg19.sites.vcf.idx.gz\n",
        "!wget ftp://gsapubftp-anonymous@ftp.broadinstitute.org/bundle/hg19/1000G_phase1.indels.hg19.sites.vcf.gz\n",
        "!wget ftp://gsapubftp-anonymous@ftp.broadinstitute.org/bundle/hg19/1000G_omni2.5.hg19.sites.vcf.idx.gz\n",
        "!wget ftp://gsapubftp-anonymous@ftp.broadinstitute.org/bundle/hg19/1000G_omni2.5.hg19.sites.vcf.gz"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "Ozp09lb8CWn_",
        "outputId": "6a516bf3-7909-49b6-a3e9-b58dee6ed360"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2022-07-26 05:28:00--  ftp://gsapubftp-anonymous@ftp.broadinstitute.org/bundle/hg19/Mills_and_1000G_gold_standard.indels.hg19.sites.vcf.gz\n",
            "           => ‘Mills_and_1000G_gold_standard.indels.hg19.sites.vcf.gz’\n",
            "Resolving ftp.broadinstitute.org (ftp.broadinstitute.org)... 69.173.70.223\n",
            "Connecting to ftp.broadinstitute.org (ftp.broadinstitute.org)|69.173.70.223|:21... connected.\n",
            "Logging in as gsapubftp-anonymous ... Logged in!\n",
            "==> SYST ... done.    ==> PWD ... done.\n",
            "==> TYPE I ... done.  ==> CWD (1) /bundle/hg19 ... done.\n",
            "==> SIZE Mills_and_1000G_gold_standard.indels.hg19.sites.vcf.gz ... 19980713\n",
            "==> PASV ... done.    ==> RETR Mills_and_1000G_gold_standard.indels.hg19.sites.vcf.gz ... done.\n",
            "Length: 19980713 (19M) (unauthoritative)\n",
            "\n",
            "          Mills_and   0%[                    ]       0  --.-KB/s               ^C\n",
            "--2022-07-26 05:28:03--  ftp://gsapubftp-anonymous@ftp.broadinstitute.org/bundle/hg19/Mills_and_1000G_gold_standard.indels.hg19.sites.vcf.idx.gz\n",
            "           => ‘Mills_and_1000G_gold_standard.indels.hg19.sites.vcf.idx.gz’\n",
            "Resolving ftp.broadinstitute.org (ftp.broadinstitute.org)... 69.173.70.223\n",
            "Connecting to ftp.broadinstitute.org (ftp.broadinstitute.org)|69.173.70.223|:21... connected.\n",
            "Logging in as gsapubftp-anonymous ... Logged in!\n",
            "==> SYST ... done.    ==> PWD ... done.\n",
            "==> TYPE I ... done.  ==> CWD (1) /bundle/hg19 ... done.\n",
            "==> SIZE Mills_and_1000G_gold_standard.indels.hg19.sites.vcf.idx.gz ... 435725\n",
            "==> PASV ... done.    ==> RETR Mills_and_1000G_gold_standard.indels.hg19.sites.vcf.idx.gz ... done.\n",
            "Length: 435725 (426K) (unauthoritative)\n",
            "\n",
            "Mills_and_1000G_gol 100%[===================>] 425.51K  1.30MB/s    in 0.3s    \n",
            "\n",
            "2022-07-26 05:28:04 (1.30 MB/s) - ‘Mills_and_1000G_gold_standard.indels.hg19.sites.vcf.idx.gz’ saved [435725]\n",
            "\n",
            "--2022-07-26 05:28:04--  ftp://gsapubftp-anonymous@ftp.broadinstitute.org/bundle/hg19/hapmap_3.3_hg19_pop_stratified_af.vcf.gz\n",
            "           => ‘hapmap_3.3_hg19_pop_stratified_af.vcf.gz’\n",
            "Resolving ftp.broadinstitute.org (ftp.broadinstitute.org)... 69.173.70.223\n",
            "Connecting to ftp.broadinstitute.org (ftp.broadinstitute.org)|69.173.70.223|:21... connected.\n",
            "Logging in as gsapubftp-anonymous ... Logged in!\n",
            "==> SYST ... done.    ==> PWD ... done.\n",
            "==> TYPE I ... done.  ==> CWD (1) /bundle/hg19 ... done.\n",
            "==> SIZE hapmap_3.3_hg19_pop_stratified_af.vcf.gz ... 180125180\n",
            "==> PASV ... done.    ==> RETR hapmap_3.3_hg19_pop_stratified_af.vcf.gz ... done.\n",
            "Length: 180125180 (172M) (unauthoritative)\n",
            "\n",
            "hapmap_3.3_hg19_pop 100%[===================>] 171.78M  36.8MB/s    in 5.0s    \n",
            "\n",
            "2022-07-26 05:28:10 (34.1 MB/s) - ‘hapmap_3.3_hg19_pop_stratified_af.vcf.gz’ saved [180125180]\n",
            "\n",
            "--2022-07-26 05:28:10--  ftp://gsapubftp-anonymous@ftp.broadinstitute.org/bundle/hg19/hapmap_3.3_hg19_pop_stratified_af.vcf.gz.tbi\n",
            "           => ‘hapmap_3.3_hg19_pop_stratified_af.vcf.gz.tbi’\n",
            "Resolving ftp.broadinstitute.org (ftp.broadinstitute.org)... 69.173.70.223\n",
            "Connecting to ftp.broadinstitute.org (ftp.broadinstitute.org)|69.173.70.223|:21... connected.\n",
            "Logging in as gsapubftp-anonymous ... Logged in!\n",
            "==> SYST ... done.    ==> PWD ... done.\n",
            "==> TYPE I ... done.  ==> CWD (1) /bundle/hg19 ... done.\n",
            "==> SIZE hapmap_3.3_hg19_pop_stratified_af.vcf.gz.tbi ... 1552080\n",
            "==> PASV ... done.    ==> RETR hapmap_3.3_hg19_pop_stratified_af.vcf.gz.tbi ... done.\n",
            "Length: 1552080 (1.5M) (unauthoritative)\n",
            "\n",
            "hapmap_3.3_hg19_pop 100%[===================>]   1.48M  3.28MB/s    in 0.5s    \n",
            "\n",
            "2022-07-26 05:28:11 (3.28 MB/s) - ‘hapmap_3.3_hg19_pop_stratified_af.vcf.gz.tbi’ saved [1552080]\n",
            "\n",
            "--2022-07-26 05:28:12--  ftp://gsapubftp-anonymous@ftp.broadinstitute.org/bundle/hg19/hapmap_3.3.hg19.sites.vcf.idx.gz\n",
            "           => ‘hapmap_3.3.hg19.sites.vcf.idx.gz’\n",
            "Resolving ftp.broadinstitute.org (ftp.broadinstitute.org)... 69.173.70.223\n",
            "Connecting to ftp.broadinstitute.org (ftp.broadinstitute.org)|69.173.70.223|:21... connected.\n",
            "Logging in as gsapubftp-anonymous ... Logged in!\n",
            "==> SYST ... done.    ==> PWD ... done.\n",
            "==> TYPE I ... done.  ==> CWD (1) /bundle/hg19 ... done.\n",
            "==> SIZE hapmap_3.3.hg19.sites.vcf.idx.gz ... 826171\n",
            "==> PASV ... done.    ==> RETR hapmap_3.3.hg19.sites.vcf.idx.gz ... done.\n",
            "Length: 826171 (807K) (unauthoritative)\n",
            "\n",
            "hapmap_3.3.hg19.sit 100%[===================>] 806.81K  2.46MB/s    in 0.3s    \n",
            "\n",
            "2022-07-26 05:28:13 (2.46 MB/s) - ‘hapmap_3.3.hg19.sites.vcf.idx.gz’ saved [826171]\n",
            "\n",
            "--2022-07-26 05:28:13--  ftp://gsapubftp-anonymous@ftp.broadinstitute.org/bundle/hg19/hapmap_3.3.hg19.sites.vcf.gz\n",
            "           => ‘hapmap_3.3.hg19.sites.vcf.gz’\n",
            "Resolving ftp.broadinstitute.org (ftp.broadinstitute.org)... 69.173.70.223\n",
            "Connecting to ftp.broadinstitute.org (ftp.broadinstitute.org)|69.173.70.223|:21... connected.\n",
            "Logging in as gsapubftp-anonymous ... Logged in!\n",
            "==> SYST ... done.    ==> PWD ... done.\n",
            "==> TYPE I ... done.  ==> CWD (1) /bundle/hg19 ... done.\n",
            "==> SIZE hapmap_3.3.hg19.sites.vcf.gz ... 60780530\n",
            "==> PASV ... done.    ==> RETR hapmap_3.3.hg19.sites.vcf.gz ... done.\n",
            "Length: 60780530 (58M) (unauthoritative)\n",
            "\n",
            "hapmap_3.3.hg19.sit 100%[===================>]  57.96M  23.2MB/s    in 2.5s    \n",
            "\n",
            "2022-07-26 05:28:16 (23.2 MB/s) - ‘hapmap_3.3.hg19.sites.vcf.gz’ saved [60780530]\n",
            "\n",
            "--2022-07-26 05:28:16--  ftp://gsapubftp-anonymous@ftp.broadinstitute.org/bundle/hg19/dbsnp_138.hg19.vcf.idx.gz\n",
            "           => ‘dbsnp_138.hg19.vcf.idx.gz’\n",
            "Resolving ftp.broadinstitute.org (ftp.broadinstitute.org)... 69.173.70.223\n",
            "Connecting to ftp.broadinstitute.org (ftp.broadinstitute.org)|69.173.70.223|:21... connected.\n",
            "Logging in as gsapubftp-anonymous ... Logged in!\n",
            "==> SYST ... done.    ==> PWD ... done.\n",
            "==> TYPE I ... done.  ==> CWD (1) /bundle/hg19 ... done.\n",
            "==> SIZE dbsnp_138.hg19.vcf.idx.gz ... 4035849\n",
            "==> PASV ... done.    ==> RETR dbsnp_138.hg19.vcf.idx.gz ... done.\n",
            "Length: 4035849 (3.8M) (unauthoritative)\n",
            "\n",
            "dbsnp_138.hg19.vcf. 100%[===================>]   3.85M  7.50MB/s    in 0.5s    \n",
            "\n",
            "2022-07-26 05:28:18 (7.50 MB/s) - ‘dbsnp_138.hg19.vcf.idx.gz’ saved [4035849]\n",
            "\n",
            "--2022-07-26 05:28:18--  ftp://gsapubftp-anonymous@ftp.broadinstitute.org/bundle/hg19/dbsnp_138.hg19.vcf.gz\n",
            "           => ‘dbsnp_138.hg19.vcf.gz’\n",
            "Resolving ftp.broadinstitute.org (ftp.broadinstitute.org)... 69.173.70.223\n",
            "Connecting to ftp.broadinstitute.org (ftp.broadinstitute.org)|69.173.70.223|:21... connected.\n",
            "Logging in as gsapubftp-anonymous ... Logged in!\n",
            "==> SYST ... done.    ==> PWD ... done.\n",
            "==> TYPE I ... done.  ==> CWD (1) /bundle/hg19 ... done.\n",
            "==> SIZE dbsnp_138.hg19.vcf.gz ... 1470262409\n",
            "==> PASV ... done.    ==> RETR dbsnp_138.hg19.vcf.gz ... done.\n",
            "Length: 1470262409 (1.4G) (unauthoritative)\n",
            "\n",
            "dbsnp_138.hg19.vcf. 100%[===================>]   1.37G  14.3MB/s    in 98s     \n",
            "\n",
            "2022-07-26 05:29:57 (14.4 MB/s) - ‘dbsnp_138.hg19.vcf.gz’ saved [1470262409]\n",
            "\n",
            "--2022-07-26 05:29:57--  ftp://gsapubftp-anonymous@ftp.broadinstitute.org/bundle/hg19/1000G_phase1.snps.high_confidence.hg19.sites.vcf.idx.gz\n",
            "           => ‘1000G_phase1.snps.high_confidence.hg19.sites.vcf.idx.gz’\n",
            "Resolving ftp.broadinstitute.org (ftp.broadinstitute.org)... 69.173.70.223\n",
            "Connecting to ftp.broadinstitute.org (ftp.broadinstitute.org)|69.173.70.223|:21... connected.\n",
            "Logging in as gsapubftp-anonymous ... Logged in!\n",
            "==> SYST ... done.    ==> PWD ... done.\n",
            "==> TYPE I ... done.  ==> CWD (1) /bundle/hg19 ... done.\n",
            "==> SIZE 1000G_phase1.snps.high_confidence.hg19.sites.vcf.idx.gz ... 3536784\n",
            "==> PASV ... done.    ==> RETR 1000G_phase1.snps.high_confidence.hg19.sites.vcf.idx.gz ... done.\n",
            "Length: 3536784 (3.4M) (unauthoritative)\n",
            "\n",
            "1000G_phase1.snps.h 100%[===================>]   3.37M  6.60MB/s    in 0.5s    \n",
            "\n",
            "2022-07-26 05:29:58 (6.60 MB/s) - ‘1000G_phase1.snps.high_confidence.hg19.sites.vcf.idx.gz’ saved [3536784]\n",
            "\n",
            "--2022-07-26 05:29:58--  ftp://gsapubftp-anonymous@ftp.broadinstitute.org/bundle/hg19/1000G_phase1.snps.high_confidence.hg19.sites.vcf.gz\n",
            "           => ‘1000G_phase1.snps.high_confidence.hg19.sites.vcf.gz’\n",
            "Resolving ftp.broadinstitute.org (ftp.broadinstitute.org)... 69.173.70.223\n",
            "Connecting to ftp.broadinstitute.org (ftp.broadinstitute.org)|69.173.70.223|:21... connected.\n",
            "Logging in as gsapubftp-anonymous ... Logged in!\n",
            "==> SYST ... done.    ==> PWD ... done.\n",
            "==> TYPE I ... done.  ==> CWD (1) /bundle/hg19 ... done.\n",
            "==> SIZE 1000G_phase1.snps.high_confidence.hg19.sites.vcf.gz ... 1801561307\n",
            "==> PASV ... done.    ==> RETR 1000G_phase1.snps.high_confidence.hg19.sites.vcf.gz ... done.\n",
            "Length: 1801561307 (1.7G) (unauthoritative)\n",
            "\n",
            "1000G_phase1.snps.h 100%[===================>]   1.68G  16.6MB/s    in 2m 11s  \n",
            "\n",
            "2022-07-26 05:32:10 (13.2 MB/s) - ‘1000G_phase1.snps.high_confidence.hg19.sites.vcf.gz’ saved [1801561307]\n",
            "\n",
            "--2022-07-26 05:32:10--  ftp://gsapubftp-anonymous@ftp.broadinstitute.org/bundle/hg19/1000G_phase1.indels.hg19.sites.vcf.idx.gz\n",
            "           => ‘1000G_phase1.indels.hg19.sites.vcf.idx.gz’\n",
            "Resolving ftp.broadinstitute.org (ftp.broadinstitute.org)... 69.173.70.223\n",
            "Connecting to ftp.broadinstitute.org (ftp.broadinstitute.org)|69.173.70.223|:21... connected.\n",
            "Logging in as gsapubftp-anonymous ... Logged in!\n",
            "==> SYST ... done.    ==> PWD ... done.\n",
            "==> TYPE I ... done.  ==> CWD (1) /bundle/hg19 ... done.\n",
            "==> SIZE 1000G_phase1.indels.hg19.sites.vcf.idx.gz ... 334184\n",
            "==> PASV ... done.    ==> RETR 1000G_phase1.indels.hg19.sites.vcf.idx.gz ... done.\n",
            "Length: 334184 (326K) (unauthoritative)\n",
            "\n",
            "1000G_phase1.indels 100%[===================>] 326.35K  1.23MB/s    in 0.3s    \n",
            "\n",
            "2022-07-26 05:32:11 (1.23 MB/s) - ‘1000G_phase1.indels.hg19.sites.vcf.idx.gz’ saved [334184]\n",
            "\n",
            "--2022-07-26 05:32:11--  ftp://gsapubftp-anonymous@ftp.broadinstitute.org/bundle/hg19/1000G_phase1.indels.hg19.sites.vcf.gz\n",
            "           => ‘1000G_phase1.indels.hg19.sites.vcf.gz’\n",
            "Resolving ftp.broadinstitute.org (ftp.broadinstitute.org)... 69.173.70.223\n",
            "Connecting to ftp.broadinstitute.org (ftp.broadinstitute.org)|69.173.70.223|:21... connected.\n",
            "Logging in as gsapubftp-anonymous ... Logged in!\n",
            "==> SYST ... done.    ==> PWD ... done.\n",
            "==> TYPE I ... done.  ==> CWD (1) /bundle/hg19 ... done.\n",
            "==> SIZE 1000G_phase1.indels.hg19.sites.vcf.gz ... 45011283\n",
            "==> PASV ... done.    ==> RETR 1000G_phase1.indels.hg19.sites.vcf.gz ... done.\n",
            "Length: 45011283 (43M) (unauthoritative)\n",
            "\n",
            "1000G_phase1.indels 100%[===================>]  42.93M  13.9MB/s    in 3.1s    \n",
            "\n",
            "2022-07-26 05:32:15 (13.9 MB/s) - ‘1000G_phase1.indels.hg19.sites.vcf.gz’ saved [45011283]\n",
            "\n",
            "--2022-07-26 05:32:15--  ftp://gsapubftp-anonymous@ftp.broadinstitute.org/bundle/hg19/1000G_omni2.5.hg19.sites.vcf.idx.gz\n",
            "           => ‘1000G_omni2.5.hg19.sites.vcf.idx.gz’\n",
            "Resolving ftp.broadinstitute.org (ftp.broadinstitute.org)... 69.173.70.223\n",
            "Connecting to ftp.broadinstitute.org (ftp.broadinstitute.org)|69.173.70.223|:21... connected.\n",
            "Logging in as gsapubftp-anonymous ... Logged in!\n",
            "==> SYST ... done.    ==> PWD ... done.\n",
            "==> TYPE I ... done.  ==> CWD (1) /bundle/hg19 ... done.\n",
            "==> SIZE 1000G_omni2.5.hg19.sites.vcf.idx.gz ... 474857\n",
            "==> PASV ... done.    ==> RETR 1000G_omni2.5.hg19.sites.vcf.idx.gz ... done.\n",
            "Length: 474857 (464K) (unauthoritative)\n",
            "\n",
            "1000G_omni2.5.hg19. 100%[===================>] 463.73K  1.35MB/s    in 0.3s    \n",
            "\n",
            "2022-07-26 05:32:17 (1.35 MB/s) - ‘1000G_omni2.5.hg19.sites.vcf.idx.gz’ saved [474857]\n",
            "\n",
            "--2022-07-26 05:32:17--  ftp://gsapubftp-anonymous@ftp.broadinstitute.org/bundle/hg19/1000G_omni2.5.hg19.sites.vcf.gz\n",
            "           => ‘1000G_omni2.5.hg19.sites.vcf.gz’\n",
            "Resolving ftp.broadinstitute.org (ftp.broadinstitute.org)... 69.173.70.223\n",
            "Connecting to ftp.broadinstitute.org (ftp.broadinstitute.org)|69.173.70.223|:21... connected.\n",
            "Logging in as gsapubftp-anonymous ... Logged in!\n",
            "==> SYST ... done.    ==> PWD ... done.\n",
            "==> TYPE I ... done.  ==> CWD (1) /bundle/hg19 ... done.\n",
            "==> SIZE 1000G_omni2.5.hg19.sites.vcf.gz ... 51811347\n",
            "==> PASV ... done.    ==> RETR 1000G_omni2.5.hg19.sites.vcf.gz ... done.\n",
            "Length: 51811347 (49M) (unauthoritative)\n",
            "\n",
            "1000G_omni2.5.hg19. 100%[===================>]  49.41M  22.8MB/s    in 2.2s    \n",
            "\n",
            "2022-07-26 05:32:20 (22.8 MB/s) - ‘1000G_omni2.5.hg19.sites.vcf.gz’ saved [51811347]\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!wget ftp://gsapubftp-anonymous@ftp.broadinstitute.org/bundle/hg19/dbsnp_138.hg19.vcf.gz.md5\n",
        "!md5sum dbsnp_138.hg19.vcf.gz"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "YDzE2OWeQl4y",
        "outputId": "f167eca5-54d4-4793-db34-44cec757b313"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2022-07-25 12:09:06--  ftp://gsapubftp-anonymous@ftp.broadinstitute.org/bundle/hg19/dbsnp_138.hg19.vcf.gz.md5\n",
            "           => ‘dbsnp_138.hg19.vcf.gz.md5’\n",
            "Resolving ftp.broadinstitute.org (ftp.broadinstitute.org)... 69.173.70.223\n",
            "Connecting to ftp.broadinstitute.org (ftp.broadinstitute.org)|69.173.70.223|:21... connected.\n",
            "Logging in as gsapubftp-anonymous ... Logged in!\n",
            "==> SYST ... done.    ==> PWD ... done.\n",
            "==> TYPE I ... done.  ==> CWD (1) /bundle/hg19 ... done.\n",
            "==> SIZE dbsnp_138.hg19.vcf.gz.md5 ... 93\n",
            "==> PASV ... done.    ==> RETR dbsnp_138.hg19.vcf.gz.md5 ... done.\n",
            "Length: 93 (unauthoritative)\n",
            "\n",
            "dbsnp_138.hg19.vcf. 100%[===================>]      93  --.-KB/s    in 0.001s  \n",
            "\n",
            "2022-07-25 12:09:10 (95.2 KB/s) - ‘dbsnp_138.hg19.vcf.gz.md5’ saved [93]\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!mkdir WES"
      ],
      "metadata": {
        "id": "0dzHVkrcX7Jn"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!ls"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "fBYXN2U0X_F3",
        "outputId": "9b0d0869-8b02-4094-9e73-f905c3c3265d"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "drive  sample_data  WES\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!cd WES/"
      ],
      "metadata": {
        "id": "rEKt7FnrYC-u"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "**get data from link**"
      ],
      "metadata": {
        "id": "2X1_LmC7YGJX"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!wget https://zenodo.org/record/3243160/files/proband_R1.fq.gz"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "tGfBT7o_YK2Y",
        "outputId": "0fe79191-98e9-4ef3-856a-d3e9b1bf0f4c"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2022-07-26 07:16:15--  https://zenodo.org/record/3243160/files/proband_R1.fq.gz\n",
            "Resolving zenodo.org (zenodo.org)... 137.138.76.77\n",
            "Connecting to zenodo.org (zenodo.org)|137.138.76.77|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 174028289 (166M) [application/octet-stream]\n",
            "Saving to: ‘proband_R1.fq.gz’\n",
            "\n",
            "proband_R1.fq.gz    100%[===================>] 165.97M  5.58MB/s    in 22s     \n",
            "\n",
            "2022-07-26 07:16:39 (7.45 MB/s) - ‘proband_R1.fq.gz’ saved [174028289/174028289]\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!wget https://zenodo.org/record/3243160/files/proband_R2.fq.gz"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "J0rNllcTYUq7",
        "outputId": "7b7e9fb1-de9f-4321-e99b-5ecebc9ef0d4"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2022-07-26 07:16:44--  https://zenodo.org/record/3243160/files/proband_R2.fq.gz\n",
            "Resolving zenodo.org (zenodo.org)... 137.138.76.77\n",
            "Connecting to zenodo.org (zenodo.org)|137.138.76.77|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 173473634 (165M) [application/octet-stream]\n",
            "Saving to: ‘proband_R2.fq.gz’\n",
            "\n",
            "proband_R2.fq.gz    100%[===================>] 165.44M  19.1MB/s    in 23s     \n",
            "\n",
            "2022-07-26 07:17:09 (7.11 MB/s) - ‘proband_R2.fq.gz’ saved [173473634/173473634]\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!cd /content/drive/MyDrive/WES\n",
        "!pwd"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "PoXbtn4UYn9A",
        "outputId": "faf6bfd6-a13f-4592-fff9-c7b1bf21d03f"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "/content\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!cp proband_R1.fq.gz /content/drive/MyDrive/WES/"
      ],
      "metadata": {
        "id": "PKux7_iXZxTJ"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!cp proband_R2.fq.gz /content/drive/MyDrive/WES/"
      ],
      "metadata": {
        "id": "CQyu9fgYZzk7"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "**Get reference genome (in this example chromosome 8 from hg19)**"
      ],
      "metadata": {
        "id": "XyPcOO1oaw-Z"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!wget https://zenodo.org/record/3243160/files/hg19_chr8.fa.gz\n",
        "#!cp hg19_chr8.fa.gz /content/drive/MyDrive/WES/"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "1wM_x1VvavF3",
        "outputId": "8884056a-8a3c-4d57-898a-75dc8c3324c1"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2022-07-26 07:17:26--  https://zenodo.org/record/3243160/files/hg19_chr8.fa.gz\n",
            "Resolving zenodo.org (zenodo.org)... 137.138.76.77\n",
            "Connecting to zenodo.org (zenodo.org)|137.138.76.77|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 46889263 (45M) [application/octet-stream]\n",
            "Saving to: ‘hg19_chr8.fa.gz’\n",
            "\n",
            "hg19_chr8.fa.gz     100%[===================>]  44.72M  7.80MB/s    in 7.0s    \n",
            "\n",
            "2022-07-26 07:17:35 (6.39 MB/s) - ‘hg19_chr8.fa.gz’ saved [46889263/46889263]\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!sudo apt search fastqc"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "kkK8bCKxbCzt",
        "outputId": "e653a940-5b83-4b0b-c96d-d0f38cfca3b8"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Sorting... Done\n",
            "Full Text Search... Done\n",
            "\u001b[32mfastqc\u001b[0m/bionic 0.11.5+dfsg-6 all\n",
            "  quality control for high throughput sequence data\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!sudo apt install fastqc"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "4lcH1hgRbJnE",
        "outputId": "e42ae0fd-7d1a-4e3a-bb08-be777d044571"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Reading package lists... Done\n",
            "Building dependency tree       \n",
            "Reading state information... Done\n",
            "The following package was automatically installed and is no longer required:\n",
            "  libnvidia-common-460\n",
            "Use 'sudo apt autoremove' to remove it.\n",
            "The following additional packages will be installed:\n",
            "  libapache-pom-java libcommons-compress-java libcommons-jexl2-java\n",
            "  libcommons-logging-java libcommons-math3-java libcommons-parent-java\n",
            "  libhtsjdk-java libjbzip2-java libngs-java libngs-sdk-dev libngs-sdk1\n",
            "  libsnappy-java libsnappy-jni libxz-java\n",
            "Suggested packages:\n",
            "  libcommons-jexl2-java-doc libavalon-framework-java\n",
            "  libcommons-logging-java-doc libexcalibur-logkit-java liblog4j1.2-java\n",
            "  picard-tools\n",
            "The following NEW packages will be installed:\n",
            "  fastqc libapache-pom-java libcommons-compress-java libcommons-jexl2-java\n",
            "  libcommons-logging-java libcommons-math3-java libcommons-parent-java\n",
            "  libhtsjdk-java libjbzip2-java libngs-java libngs-sdk-dev libngs-sdk1\n",
            "  libsnappy-java libsnappy-jni libxz-java\n",
            "0 upgraded, 15 newly installed, 0 to remove and 49 not upgraded.\n",
            "Need to get 5,140 kB of archives.\n",
            "After this operation, 8,129 kB of additional disk space will be used.\n",
            "Get:1 http://archive.ubuntu.com/ubuntu bionic/universe amd64 libcommons-math3-java all 3.6.1-3 [2,016 kB]\n",
            "Get:2 http://archive.ubuntu.com/ubuntu bionic/universe amd64 libjbzip2-java all 0.9.1-4 [42.6 kB]\n",
            "Get:3 http://archive.ubuntu.com/ubuntu bionic/universe amd64 libapache-pom-java all 18-1 [4,720 B]\n",
            "Get:4 http://archive.ubuntu.com/ubuntu bionic/universe amd64 libcommons-parent-java all 43-1 [10.8 kB]\n",
            "Get:5 http://archive.ubuntu.com/ubuntu bionic-updates/universe amd64 libcommons-compress-java all 1.18-1~18.04 [531 kB]\n",
            "Get:6 http://archive.ubuntu.com/ubuntu bionic/universe amd64 libcommons-logging-java all 1.2-2 [60.3 kB]\n",
            "Get:7 http://archive.ubuntu.com/ubuntu bionic/universe amd64 libcommons-jexl2-java all 2.1.1-4 [252 kB]\n",
            "Get:8 http://archive.ubuntu.com/ubuntu bionic/universe amd64 libngs-sdk1 amd64 1.3.0-3 [52.3 kB]\n",
            "Get:9 http://archive.ubuntu.com/ubuntu bionic/universe amd64 libngs-sdk-dev amd64 1.3.0-3 [92.8 kB]\n",
            "Get:10 http://archive.ubuntu.com/ubuntu bionic/universe amd64 libngs-java amd64 1.3.0-3 [76.4 kB]\n",
            "Get:11 http://archive.ubuntu.com/ubuntu bionic/universe amd64 libsnappy-jni amd64 1.1.4-1 [6,996 B]\n",
            "Get:12 http://archive.ubuntu.com/ubuntu bionic/universe amd64 libsnappy-java all 1.1.4-1 [67.2 kB]\n",
            "Get:13 http://archive.ubuntu.com/ubuntu bionic/universe amd64 libxz-java all 1.8-1 [134 kB]\n",
            "Get:14 http://archive.ubuntu.com/ubuntu bionic/universe amd64 libhtsjdk-java all 2.8.1+dfsg-2 [1,438 kB]\n",
            "Get:15 http://archive.ubuntu.com/ubuntu bionic/universe amd64 fastqc all 0.11.5+dfsg-6 [354 kB]\n",
            "Fetched 5,140 kB in 0s (12.5 MB/s)\n",
            "debconf: unable to initialize frontend: Dialog\n",
            "debconf: (No usable dialog-like program is installed, so the dialog based frontend cannot be used. at /usr/share/perl5/Debconf/FrontEnd/Dialog.pm line 76, <> line 15.)\n",
            "debconf: falling back to frontend: Readline\n",
            "debconf: unable to initialize frontend: Readline\n",
            "debconf: (This frontend requires a controlling tty.)\n",
            "debconf: falling back to frontend: Teletype\n",
            "dpkg-preconfigure: unable to re-open stdin: \n",
            "Selecting previously unselected package libcommons-math3-java.\n",
            "(Reading database ... 155653 files and directories currently installed.)\n",
            "Preparing to unpack .../00-libcommons-math3-java_3.6.1-3_all.deb ...\n",
            "Unpacking libcommons-math3-java (3.6.1-3) ...\n",
            "Selecting previously unselected package libjbzip2-java.\n",
            "Preparing to unpack .../01-libjbzip2-java_0.9.1-4_all.deb ...\n",
            "Unpacking libjbzip2-java (0.9.1-4) ...\n",
            "Selecting previously unselected package libapache-pom-java.\n",
            "Preparing to unpack .../02-libapache-pom-java_18-1_all.deb ...\n",
            "Unpacking libapache-pom-java (18-1) ...\n",
            "Selecting previously unselected package libcommons-parent-java.\n",
            "Preparing to unpack .../03-libcommons-parent-java_43-1_all.deb ...\n",
            "Unpacking libcommons-parent-java (43-1) ...\n",
            "Selecting previously unselected package libcommons-compress-java.\n",
            "Preparing to unpack .../04-libcommons-compress-java_1.18-1~18.04_all.deb ...\n",
            "Unpacking libcommons-compress-java (1.18-1~18.04) ...\n",
            "Selecting previously unselected package libcommons-logging-java.\n",
            "Preparing to unpack .../05-libcommons-logging-java_1.2-2_all.deb ...\n",
            "Unpacking libcommons-logging-java (1.2-2) ...\n",
            "Selecting previously unselected package libcommons-jexl2-java.\n",
            "Preparing to unpack .../06-libcommons-jexl2-java_2.1.1-4_all.deb ...\n",
            "Unpacking libcommons-jexl2-java (2.1.1-4) ...\n",
            "Selecting previously unselected package libngs-sdk1.\n",
            "Preparing to unpack .../07-libngs-sdk1_1.3.0-3_amd64.deb ...\n",
            "Unpacking libngs-sdk1 (1.3.0-3) ...\n",
            "Selecting previously unselected package libngs-sdk-dev.\n",
            "Preparing to unpack .../08-libngs-sdk-dev_1.3.0-3_amd64.deb ...\n",
            "Unpacking libngs-sdk-dev (1.3.0-3) ...\n",
            "Selecting previously unselected package libngs-java.\n",
            "Preparing to unpack .../09-libngs-java_1.3.0-3_amd64.deb ...\n",
            "Unpacking libngs-java (1.3.0-3) ...\n",
            "Selecting previously unselected package libsnappy-jni.\n",
            "Preparing to unpack .../10-libsnappy-jni_1.1.4-1_amd64.deb ...\n",
            "Unpacking libsnappy-jni (1.1.4-1) ...\n",
            "Selecting previously unselected package libsnappy-java.\n",
            "Preparing to unpack .../11-libsnappy-java_1.1.4-1_all.deb ...\n",
            "Unpacking libsnappy-java (1.1.4-1) ...\n",
            "Selecting previously unselected package libxz-java.\n",
            "Preparing to unpack .../12-libxz-java_1.8-1_all.deb ...\n",
            "Unpacking libxz-java (1.8-1) ...\n",
            "Selecting previously unselected package libhtsjdk-java.\n",
            "Preparing to unpack .../13-libhtsjdk-java_2.8.1+dfsg-2_all.deb ...\n",
            "Unpacking libhtsjdk-java (2.8.1+dfsg-2) ...\n",
            "Selecting previously unselected package fastqc.\n",
            "Preparing to unpack .../14-fastqc_0.11.5+dfsg-6_all.deb ...\n",
            "Unpacking fastqc (0.11.5+dfsg-6) ...\n",
            "Setting up libsnappy-jni (1.1.4-1) ...\n",
            "Setting up libcommons-math3-java (3.6.1-3) ...\n",
            "Setting up libjbzip2-java (0.9.1-4) ...\n",
            "Setting up libngs-sdk1 (1.3.0-3) ...\n",
            "Setting up libngs-sdk-dev (1.3.0-3) ...\n",
            "Setting up libxz-java (1.8-1) ...\n",
            "Setting up libsnappy-java (1.1.4-1) ...\n",
            "Setting up libapache-pom-java (18-1) ...\n",
            "Setting up libngs-java (1.3.0-3) ...\n",
            "Setting up libcommons-parent-java (43-1) ...\n",
            "Setting up libcommons-compress-java (1.18-1~18.04) ...\n",
            "Setting up libcommons-logging-java (1.2-2) ...\n",
            "Setting up libcommons-jexl2-java (2.1.1-4) ...\n",
            "Setting up libhtsjdk-java (2.8.1+dfsg-2) ...\n",
            "Setting up fastqc (0.11.5+dfsg-6) ...\n",
            "Processing triggers for man-db (2.8.3-2ubuntu0.1) ...\n",
            "Processing triggers for hicolor-icon-theme (0.17-2) ...\n",
            "Processing triggers for mime-support (3.60ubuntu1) ...\n",
            "Processing triggers for libc-bin (2.27-3ubuntu1.3) ...\n",
            "/sbin/ldconfig.real: /usr/local/lib/python3.7/dist-packages/ideep4py/lib/libmkldnn.so.0 is not a symbolic link\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!fastqc --help"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "HVTd-FlgbR5I",
        "outputId": "989d3ce9-43c8-472e-9a03-d30c0d79bb98"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\n",
            "            FastQC - A high throughput sequence QC analysis tool\n",
            "\n",
            "SYNOPSIS\n",
            "\n",
            "\tfastqc seqfile1 seqfile2 .. seqfileN\n",
            "\n",
            "    fastqc [-o output dir] [--(no)extract] [-f fastq|bam|sam] \n",
            "           [-c contaminant file] seqfile1 .. seqfileN\n",
            "\n",
            "DESCRIPTION\n",
            "\n",
            "    FastQC reads a set of sequence files and produces from each one a quality\n",
            "    control report consisting of a number of different modules, each one of \n",
            "    which will help to identify a different potential type of problem in your\n",
            "    data.\n",
            "    \n",
            "    If no files to process are specified on the command line then the program\n",
            "    will start as an interactive graphical application.  If files are provided\n",
            "    on the command line then the program will run with no user interaction\n",
            "    required.  In this mode it is suitable for inclusion into a standardised\n",
            "    analysis pipeline.\n",
            "    \n",
            "    The options for the program as as follows:\n",
            "    \n",
            "    -h --help       Print this help file and exit\n",
            "    \n",
            "    -v --version    Print the version of the program and exit\n",
            "    \n",
            "    -o --outdir     Create all output files in the specified output directory.\n",
            "                    Please note that this directory must exist as the program\n",
            "                    will not create it.  If this option is not set then the \n",
            "                    output file for each sequence file is created in the same\n",
            "                    directory as the sequence file which was processed.\n",
            "                    \n",
            "    --casava        Files come from raw casava output. Files in the same sample\n",
            "                    group (differing only by the group number) will be analysed\n",
            "                    as a set rather than individually. Sequences with the filter\n",
            "                    flag set in the header will be excluded from the analysis.\n",
            "                    Files must have the same names given to them by casava\n",
            "                    (including being gzipped and ending with .gz) otherwise they\n",
            "                    won't be grouped together correctly.\n",
            "                    \n",
            "    --nofilter      If running with --casava then don't remove read flagged by\n",
            "                    casava as poor quality when performing the QC analysis.\n",
            "                   \n",
            "    --extract       If set then the zipped output file will be uncompressed in\n",
            "                    the same directory after it has been created.  By default\n",
            "                    this option will be set if fastqc is run in non-interactive\n",
            "                    mode.\n",
            "                    \n",
            "    -j --java       Provides the full path to the java binary you want to use to\n",
            "                    launch fastqc. If not supplied then java is assumed to be in\n",
            "                    your path.\n",
            "                   \n",
            "    --noextract     Do not uncompress the output file after creating it.  You\n",
            "                    should set this option if you do not wish to uncompress\n",
            "                    the output when running in non-interactive mode.\n",
            "                    \n",
            "    --nogroup       Disable grouping of bases for reads >50bp. All reports will\n",
            "                    show data for every base in the read.  WARNING: Using this\n",
            "                    option will cause fastqc to crash and burn if you use it on\n",
            "                    really long reads, and your plots may end up a ridiculous size.\n",
            "                    You have been warned!\n",
            "                    \n",
            "    -f --format     Bypasses the normal sequence file format detection and\n",
            "                    forces the program to use the specified format.  Valid\n",
            "                    formats are bam,sam,bam_mapped,sam_mapped and fastq\n",
            "                    \n",
            "    -t --threads    Specifies the number of files which can be processed\n",
            "                    simultaneously.  Each thread will be allocated 250MB of\n",
            "                    memory so you shouldn't run more threads than your\n",
            "                    available memory will cope with, and not more than\n",
            "                    6 threads on a 32 bit machine\n",
            "                  \n",
            "    -c              Specifies a non-default file which contains the list of\n",
            "    --contaminants  contaminants to screen overrepresented sequences against.\n",
            "                    The file must contain sets of named contaminants in the\n",
            "                    form name[tab]sequence.  Lines prefixed with a hash will\n",
            "                    be ignored.\n",
            "\n",
            "    -a              Specifies a non-default file which contains the list of\n",
            "    --adapters      adapter sequences which will be explicity searched against\n",
            "                    the library. The file must contain sets of named adapters\n",
            "                    in the form name[tab]sequence.  Lines prefixed with a hash\n",
            "                    will be ignored.\n",
            "                    \n",
            "    -l              Specifies a non-default file which contains a set of criteria\n",
            "    --limits        which will be used to determine the warn/error limits for the\n",
            "                    various modules.  This file can also be used to selectively \n",
            "                    remove some modules from the output all together.  The format\n",
            "                    needs to mirror the default limits.txt file found in the\n",
            "                    Configuration folder.\n",
            "                    \n",
            "   -k --kmers       Specifies the length of Kmer to look for in the Kmer content\n",
            "                    module. Specified Kmer length must be between 2 and 10. Default\n",
            "                    length is 7 if not specified.\n",
            "                    \n",
            "   -q --quiet       Supress all progress messages on stdout and only report errors.\n",
            "   \n",
            "   -d --dir         Selects a directory to be used for temporary files written when\n",
            "                    generating report images. Defaults to system temp directory if\n",
            "                    not specified.\n",
            "                    \n",
            "BUGS\n",
            "\n",
            "    Any bugs in fastqc should be reported either to simon.andrews@babraham.ac.uk\n",
            "    or in www.bioinformatics.babraham.ac.uk/bugzilla/\n",
            "                   \n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!fastqc -t 2 /content/drive/MyDrive/WES/proband_R1.fq.gz "
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "psSbzouEbbaQ",
        "outputId": "c56d2ad2-7d38-4558-9a3a-d565b73739a2"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Started analysis of proband_R1.fq.gz\n",
            "Started analysis of proband_R1.fq.gz\n",
            "Approx 5% complete for proband_R1.fq.gz\n",
            "Approx 5% complete for proband_R1.fq.gz\n",
            "Approx 10% complete for proband_R1.fq.gz\n",
            "Approx 10% complete for proband_R1.fq.gz\n",
            "Approx 15% complete for proband_R1.fq.gz\n",
            "Approx 15% complete for proband_R1.fq.gz\n",
            "Approx 20% complete for proband_R1.fq.gz\n",
            "Approx 20% complete for proband_R1.fq.gz\n",
            "Approx 25% complete for proband_R1.fq.gz\n",
            "Approx 25% complete for proband_R1.fq.gz\n",
            "Approx 30% complete for proband_R1.fq.gz\n",
            "Approx 30% complete for proband_R1.fq.gz\n",
            "Approx 35% complete for proband_R1.fq.gz\n",
            "Approx 35% complete for proband_R1.fq.gz\n",
            "Approx 40% complete for proband_R1.fq.gz\n",
            "Approx 40% complete for proband_R1.fq.gz\n",
            "Approx 45% complete for proband_R1.fq.gz\n",
            "Approx 45% complete for proband_R1.fq.gz\n",
            "Approx 50% complete for proband_R1.fq.gz\n",
            "Approx 50% complete for proband_R1.fq.gz\n",
            "Approx 55% complete for proband_R1.fq.gz\n",
            "Approx 55% complete for proband_R1.fq.gz\n",
            "Approx 60% complete for proband_R1.fq.gz\n",
            "Approx 60% complete for proband_R1.fq.gz\n",
            "Approx 65% complete for proband_R1.fq.gz\n",
            "Approx 65% complete for proband_R1.fq.gz\n",
            "Approx 70% complete for proband_R1.fq.gz\n",
            "Approx 70% complete for proband_R1.fq.gz\n",
            "Approx 75% complete for proband_R1.fq.gz\n",
            "Approx 75% complete for proband_R1.fq.gz\n",
            "Approx 80% complete for proband_R1.fq.gz\n",
            "Approx 80% complete for proband_R1.fq.gz\n",
            "Approx 85% complete for proband_R1.fq.gz\n",
            "Approx 90% complete for proband_R1.fq.gz\n",
            "Approx 85% complete for proband_R1.fq.gz\n",
            "Approx 95% complete for proband_R1.fq.gz\n",
            "Approx 90% complete for proband_R1.fq.gz\n",
            "Analysis complete for proband_R1.fq.gz\n",
            "Approx 95% complete for proband_R1.fq.gz\n",
            "Analysis complete for proband_R1.fq.gz\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!fastqc -t 2 /content/drive/MyDrive/WES/proband_R2.fq.gz"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "0PVWoApjbyUI",
        "outputId": "861557f9-c8a4-4ffd-c41a-6c26aea79078"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Started analysis of proband_R2.fq.gz\n",
            "Approx 5% complete for proband_R2.fq.gz\n",
            "Approx 10% complete for proband_R2.fq.gz\n",
            "Approx 15% complete for proband_R2.fq.gz\n",
            "Approx 20% complete for proband_R2.fq.gz\n",
            "Approx 25% complete for proband_R2.fq.gz\n",
            "Approx 30% complete for proband_R2.fq.gz\n",
            "Approx 35% complete for proband_R2.fq.gz\n",
            "Approx 40% complete for proband_R2.fq.gz\n",
            "Approx 45% complete for proband_R2.fq.gz\n",
            "Approx 50% complete for proband_R2.fq.gz\n",
            "Approx 55% complete for proband_R2.fq.gz\n",
            "Approx 60% complete for proband_R2.fq.gz\n",
            "Approx 65% complete for proband_R2.fq.gz\n",
            "Approx 70% complete for proband_R2.fq.gz\n",
            "Approx 75% complete for proband_R2.fq.gz\n",
            "Approx 80% complete for proband_R2.fq.gz\n",
            "Approx 85% complete for proband_R2.fq.gz\n",
            "Approx 90% complete for proband_R2.fq.gz\n",
            "Approx 95% complete for proband_R2.fq.gz\n",
            "Analysis complete for proband_R2.fq.gz\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!sudo apt install bwa"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "dMu37Hy6c9HG",
        "outputId": "3a4d2acf-ecb9-4c52-c2c2-edca42ec15b6"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Reading package lists... Done\n",
            "Building dependency tree       \n",
            "Reading state information... Done\n",
            "The following package was automatically installed and is no longer required:\n",
            "  libnvidia-common-460\n",
            "Use 'sudo apt autoremove' to remove it.\n",
            "Suggested packages:\n",
            "  samtools\n",
            "The following NEW packages will be installed:\n",
            "  bwa\n",
            "0 upgraded, 1 newly installed, 0 to remove and 49 not upgraded.\n",
            "Need to get 176 kB of archives.\n",
            "After this operation, 404 kB of additional disk space will be used.\n",
            "Get:1 http://archive.ubuntu.com/ubuntu bionic-updates/universe amd64 bwa amd64 0.7.17-1ubuntu0.1 [176 kB]\n",
            "Fetched 176 kB in 1s (187 kB/s)\n",
            "debconf: unable to initialize frontend: Dialog\n",
            "debconf: (No usable dialog-like program is installed, so the dialog based frontend cannot be used. at /usr/share/perl5/Debconf/FrontEnd/Dialog.pm line 76, <> line 1.)\n",
            "debconf: falling back to frontend: Readline\n",
            "debconf: unable to initialize frontend: Readline\n",
            "debconf: (This frontend requires a controlling tty.)\n",
            "debconf: falling back to frontend: Teletype\n",
            "dpkg-preconfigure: unable to re-open stdin: \n",
            "Selecting previously unselected package bwa.\n",
            "(Reading database ... 155704 files and directories currently installed.)\n",
            "Preparing to unpack .../bwa_0.7.17-1ubuntu0.1_amd64.deb ...\n",
            "Unpacking bwa (0.7.17-1ubuntu0.1) ...\n",
            "Setting up bwa (0.7.17-1ubuntu0.1) ...\n",
            "Processing triggers for man-db (2.8.3-2ubuntu0.1) ...\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!bwa index hg19_chr8.fa.gz"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "iIj2Ddb7enfl",
        "outputId": "88dcaaaf-447d-4094-9615-e713d4fa0925"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "[bwa_index] Pack FASTA... 2.82 sec\n",
            "[bwa_index] Construct BWT for the packed sequence...\n",
            "[BWTIncCreate] textLength=292728044, availableWord=32597292\n",
            "[BWTIncConstructFromPacked] 10 iterations done. 53770876 characters processed.\n",
            "[BWTIncConstructFromPacked] 20 iterations done. 99337660 characters processed.\n",
            "[BWTIncConstructFromPacked] 30 iterations done. 139833516 characters processed.\n",
            "[BWTIncConstructFromPacked] 40 iterations done. 175822284 characters processed.\n",
            "[BWTIncConstructFromPacked] 50 iterations done. 207805164 characters processed.\n",
            "[BWTIncConstructFromPacked] 60 iterations done. 236227612 characters processed.\n",
            "[BWTIncConstructFromPacked] 70 iterations done. 261485516 characters processed.\n",
            "[BWTIncConstructFromPacked] 80 iterations done. 283930748 characters processed.\n",
            "[bwt_gen] Finished constructing BWT in 85 iterations.\n",
            "[bwa_index] 127.76 seconds elapse.\n",
            "[bwa_index] Update BWT... 0.96 sec\n",
            "[bwa_index] Pack forward-only FASTA... 1.79 sec\n",
            "[bwa_index] Construct SA from BWT and Occ... 60.27 sec\n",
            "[main] Version: 0.7.17-r1188\n",
            "[main] CMD: bwa index hg19_chr8.fa.gz\n",
            "[main] Real time: 195.841 sec; CPU: 193.631 sec\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!bwa mem -a -v 3 -t 2 -Y -R '@RG\\tID:rg1\\tSM:TEST\\tPL:illumina\\tLB:lib1' hg19_chr8.fa.gz proband_R1.fq.gz proband_R2.fq.gz > aln.pe.sam"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "McYxrbjnfT2O",
        "outputId": "78125b57-cd07-43e0-c0bd-d7112fd0d227"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
            "[M::process] read 198020 sequences (20000020 bp)...\n",
            "[M::process] read 198020 sequences (20000020 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 75213, 0, 2)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (254, 273, 293)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (176, 371)\n",
            "[M::mem_pestat] mean and std.dev: (273.43, 26.56)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (137, 410)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 198020 reads in 34.882 CPU sec, 18.069 real sec\n",
            "[M::process] read 198020 sequences (20000020 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 69705, 0, 7)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (253, 272, 291)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (177, 367)\n",
            "[M::mem_pestat] mean and std.dev: (272.12, 26.80)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (139, 405)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 198020 reads in 31.112 CPU sec, 18.253 real sec\n",
            "[M::process] read 198020 sequences (20000020 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 69452, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (256, 275, 294)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (180, 370)\n",
            "[M::mem_pestat] mean and std.dev: (275.52, 26.30)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (142, 408)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 198020 reads in 22.730 CPU sec, 11.787 real sec\n",
            "[M::process] read 198020 sequences (20000020 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (5, 97520, 0, 7)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (253, 272, 291)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (177, 367)\n",
            "[M::mem_pestat] mean and std.dev: (272.30, 26.67)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (139, 405)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 198020 reads in 18.161 CPU sec, 9.462 real sec\n",
            "[M::process] read 198020 sequences (20000020 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (6, 93968, 0, 5)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (253, 272, 292)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (175, 370)\n",
            "[M::mem_pestat] mean and std.dev: (272.54, 26.80)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (136, 409)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 198020 reads in 19.612 CPU sec, 10.183 real sec\n",
            "[M::process] read 198020 sequences (20000020 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 97558, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (254, 273, 292)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (178, 368)\n",
            "[M::mem_pestat] mean and std.dev: (272.99, 26.39)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (140, 406)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 198020 reads in 18.292 CPU sec, 9.442 real sec\n",
            "[M::process] read 198020 sequences (20000020 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (5, 96257, 1, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (256, 274, 293)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (182, 367)\n",
            "[M::mem_pestat] mean and std.dev: (274.63, 26.21)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (145, 404)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 198020 reads in 16.974 CPU sec, 8.882 real sec\n",
            "[M::process] read 198020 sequences (20000020 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (6, 97397, 0, 2)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (255, 273, 293)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (179, 369)\n",
            "[M::mem_pestat] mean and std.dev: (273.79, 26.52)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (141, 407)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 198020 reads in 18.101 CPU sec, 9.393 real sec\n",
            "[M::process] read 198020 sequences (20000020 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (3, 97581, 0, 5)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (255, 274, 293)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (179, 369)\n",
            "[M::mem_pestat] mean and std.dev: (274.31, 26.37)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (141, 407)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 198020 reads in 17.662 CPU sec, 9.119 real sec\n",
            "[M::process] read 198020 sequences (20000020 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (3, 97359, 0, 4)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (256, 274, 294)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (180, 370)\n",
            "[M::mem_pestat] mean and std.dev: (274.74, 26.36)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (142, 408)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 198020 reads in 17.157 CPU sec, 8.920 real sec\n",
            "[M::process] read 198020 sequences (20000020 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 97856, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (257, 275, 295)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (181, 371)\n",
            "[M::mem_pestat] mean and std.dev: (275.65, 26.45)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (143, 409)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 198020 reads in 16.201 CPU sec, 8.373 real sec\n",
            "[M::process] read 198020 sequences (20000020 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 96477, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (256, 275, 294)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (180, 370)\n",
            "[M::mem_pestat] mean and std.dev: (275.27, 26.44)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (142, 408)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 198020 reads in 17.562 CPU sec, 9.121 real sec\n",
            "[M::process] read 198020 sequences (20000020 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (3, 94337, 0, 2)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (257, 275, 295)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (181, 371)\n",
            "[M::mem_pestat] mean and std.dev: (275.90, 26.32)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (143, 409)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 198020 reads in 18.642 CPU sec, 9.668 real sec\n",
            "[M::process] read 198020 sequences (20000020 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 198020 reads in 75.393 CPU sec, 40.951 real sec\n",
            "[M::process] read 198020 sequences (20000020 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 1, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 198020 reads in 83.187 CPU sec, 42.865 real sec\n",
            "[M::process] read 198020 sequences (20000020 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 76339, 0, 1)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (257, 276, 295)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (181, 371)\n",
            "[M::mem_pestat] mean and std.dev: (276.11, 26.29)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (143, 409)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 198020 reads in 26.192 CPU sec, 13.794 real sec\n",
            "[M::process] read 198020 sequences (20000020 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 97667, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (256, 275, 294)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (180, 370)\n",
            "[M::mem_pestat] mean and std.dev: (275.34, 26.31)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (142, 408)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 198020 reads in 16.942 CPU sec, 8.765 real sec\n",
            "[M::process] read 198020 sequences (20000020 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 97938, 0, 1)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (256, 275, 294)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (180, 370)\n",
            "[M::mem_pestat] mean and std.dev: (275.27, 26.33)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (142, 408)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 198020 reads in 16.225 CPU sec, 8.382 real sec\n",
            "[M::process] read 198020 sequences (20000020 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 97981, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (256, 275, 295)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (178, 373)\n",
            "[M::mem_pestat] mean and std.dev: (275.28, 26.63)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (139, 412)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 198020 reads in 15.971 CPU sec, 8.330 real sec\n",
            "[M::process] read 198020 sequences (20000020 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (5, 97852, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (256, 275, 294)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (180, 370)\n",
            "[M::mem_pestat] mean and std.dev: (274.93, 26.48)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (142, 408)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 198020 reads in 16.636 CPU sec, 8.581 real sec\n",
            "[M::process] read 198020 sequences (20000020 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 98028, 0, 2)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (256, 274, 294)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (180, 370)\n",
            "[M::mem_pestat] mean and std.dev: (274.50, 26.40)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (142, 408)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 198020 reads in 16.220 CPU sec, 8.427 real sec\n",
            "[M::process] read 198020 sequences (20000020 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (3, 97816, 0, 2)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (254, 272, 292)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (178, 368)\n",
            "[M::mem_pestat] mean and std.dev: (272.86, 26.69)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (140, 406)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 198020 reads in 16.414 CPU sec, 8.452 real sec\n",
            "[M::process] read 198020 sequences (20000020 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (12, 90140, 0, 13)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (563, 1022, 1070)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2084)\n",
            "[M::mem_pestat] mean and std.dev: (801.09, 282.99)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2591)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (250, 268, 287)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (176, 361)\n",
            "[M::mem_pestat] mean and std.dev: (268.45, 26.37)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (139, 398)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (534, 688, 1117)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2283)\n",
            "[M::mem_pestat] mean and std.dev: (814.62, 365.22)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2866)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 198020 reads in 19.628 CPU sec, 10.187 real sec\n",
            "[M::process] read 32628 sequences (3295428 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (30, 94165, 0, 18)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (136, 671, 958)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2602)\n",
            "[M::mem_pestat] mean and std.dev: (628.73, 545.56)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 3424)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (250, 268, 287)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (176, 361)\n",
            "[M::mem_pestat] mean and std.dev: (268.52, 26.27)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (139, 398)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (647, 817, 1002)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1712)\n",
            "[M::mem_pestat] mean and std.dev: (707.27, 286.65)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2067)\n",
            "[M::mem_pestat] skip orientation FF\n",
            "[M::mem_pestat] skip orientation RR\n",
            "[M::mem_process_seqs] Processed 198020 reads in 16.329 CPU sec, 8.467 real sec\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 16047, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (257, 276, 294)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (183, 368)\n",
            "[M::mem_pestat] mean and std.dev: (275.86, 25.96)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (146, 405)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 32628 reads in 3.321 CPU sec, 2.800 real sec\n",
            "[main] Version: 0.7.17-r1188\n",
            "[main] CMD: bwa mem -a -v 3 -t 2 -Y -R @RG\\tID:rg1\\tSM:TEST\\tPL:illumina\\tLB:lib1 hg19_chr8.fa.gz proband_R1.fq.gz proband_R2.fq.gz\n",
            "[main] Real time: 311.968 sec; CPU: 590.597 sec\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!sudo apt install samtools\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "FBEXvJWWibXe",
        "outputId": "bb6987a7-b605-4e22-bf57-54b90b05487d"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Reading package lists... Done\n",
            "Building dependency tree       \n",
            "Reading state information... Done\n",
            "The following package was automatically installed and is no longer required:\n",
            "  libnvidia-common-460\n",
            "Use 'sudo apt autoremove' to remove it.\n",
            "The following additional packages will be installed:\n",
            "  cwltool libc-ares2 libhts2 libjs-bootstrap nodejs nodejs-doc\n",
            "  python-asn1crypto python-avro python-cachecontrol python-certifi\n",
            "  python-cffi-backend python-chardet python-cryptography python-enum34\n",
            "  python-html5lib python-idna python-ipaddress python-isodate python-lockfile\n",
            "  python-mistune python-openssl python-pkg-resources python-pyparsing\n",
            "  python-rdflib python-rdflib-jsonld python-requests python-ruamel.yaml\n",
            "  python-schema-salad python-shellescape python-six python-sparqlwrapper\n",
            "  python-typing python-urllib3 python-webencodings\n",
            "Suggested packages:\n",
            "  python-cryptography-doc python-cryptography-vectors python-enum34-doc\n",
            "  python-genshi python-lxml python-lockfile-doc python-openssl-doc\n",
            "  python-openssl-dbg python-setuptools python-pyparsing-doc python-rdflib-doc\n",
            "  python-rdflib-tools python-socks python-ntlm\n",
            "The following NEW packages will be installed:\n",
            "  cwltool libc-ares2 libhts2 libjs-bootstrap nodejs nodejs-doc\n",
            "  python-asn1crypto python-avro python-cachecontrol python-certifi\n",
            "  python-cffi-backend python-chardet python-cryptography python-enum34\n",
            "  python-html5lib python-idna python-ipaddress python-isodate python-lockfile\n",
            "  python-mistune python-openssl python-pkg-resources python-pyparsing\n",
            "  python-rdflib python-rdflib-jsonld python-requests python-ruamel.yaml\n",
            "  python-schema-salad python-shellescape python-six python-sparqlwrapper\n",
            "  python-typing python-urllib3 python-webencodings samtools\n",
            "0 upgraded, 35 newly installed, 0 to remove and 49 not upgraded.\n",
            "Need to get 8,568 kB of archives.\n",
            "After this operation, 41.0 MB of additional disk space will be used.\n",
            "Get:1 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-pkg-resources all 39.0.1-2 [128 kB]\n",
            "Get:2 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-six all 1.11.0-2 [11.3 kB]\n",
            "Get:3 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-isodate all 0.6.0-1 [23.8 kB]\n",
            "Get:4 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-pyparsing all 2.2.0+dfsg1-2 [52.1 kB]\n",
            "Get:5 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-rdflib all 4.2.1-2 [237 kB]\n",
            "Get:6 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-certifi all 2018.1.18-2 [144 kB]\n",
            "Get:7 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-chardet all 3.0.4-1 [80.3 kB]\n",
            "Get:8 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-idna all 2.6-1 [32.4 kB]\n",
            "Get:9 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 python-urllib3 all 1.22-1ubuntu0.18.04.2 [86.0 kB]\n",
            "Get:10 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 python-requests all 2.18.4-2ubuntu0.1 [58.5 kB]\n",
            "Get:11 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-ruamel.yaml amd64 0.15.34-1 [183 kB]\n",
            "Get:12 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-cachecontrol all 0.11.7-1 [15.2 kB]\n",
            "Get:13 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-lockfile all 1:0.12.2-2 [14.6 kB]\n",
            "Get:14 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-mistune all 0.8.3-2 [13.8 kB]\n",
            "Get:15 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-rdflib-jsonld all 0.4.0-3 [12.2 kB]\n",
            "Get:16 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-typing all 3.6.2-1 [22.2 kB]\n",
            "Get:17 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-avro all 1.8.2+dfsg-1 [23.2 kB]\n",
            "Get:18 http://archive.ubuntu.com/ubuntu bionic/universe amd64 libjs-bootstrap all 3.3.7+dfsg-2 [228 kB]\n",
            "Get:19 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-schema-salad all 2.6.20171201034858-3 [260 kB]\n",
            "Get:20 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-shellescape all 3.4.1-2 [4,432 B]\n",
            "Get:21 http://archive.ubuntu.com/ubuntu bionic/universe amd64 cwltool all 1.0.20180302231433-1 [143 kB]\n",
            "Get:22 http://archive.ubuntu.com/ubuntu bionic/universe amd64 libhts2 amd64 1.7-2 [300 kB]\n",
            "Get:23 http://archive.ubuntu.com/ubuntu bionic-updates/universe amd64 nodejs-doc all 8.10.0~dfsg-2ubuntu0.4 [752 kB]\n",
            "Get:24 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-asn1crypto all 0.24.0-1 [72.7 kB]\n",
            "Get:25 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-cffi-backend amd64 1.11.5-1 [63.4 kB]\n",
            "Get:26 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-enum34 all 1.1.6-2 [34.8 kB]\n",
            "Get:27 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-ipaddress all 1.0.17-1 [18.2 kB]\n",
            "Get:28 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 python-cryptography amd64 2.1.4-1ubuntu1.4 [276 kB]\n",
            "Get:29 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-webencodings all 0.5-2 [10.3 kB]\n",
            "Get:30 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-html5lib all 0.999999999-1 [83.6 kB]\n",
            "Get:31 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-openssl all 17.5.0-1ubuntu1 [41.3 kB]\n",
            "Get:32 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-sparqlwrapper all 1.7.6-2 [21.8 kB]\n",
            "Get:33 http://archive.ubuntu.com/ubuntu bionic/universe amd64 samtools amd64 1.7-1 [288 kB]\n",
            "Get:34 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 libc-ares2 amd64 1.14.0-1ubuntu0.1 [37.5 kB]\n",
            "Get:35 http://archive.ubuntu.com/ubuntu bionic-updates/universe amd64 nodejs amd64 8.10.0~dfsg-2ubuntu0.4 [4,796 kB]\n",
            "Fetched 8,568 kB in 2s (4,173 kB/s)\n",
            "debconf: unable to initialize frontend: Dialog\n",
            "debconf: (No usable dialog-like program is installed, so the dialog based frontend cannot be used. at /usr/share/perl5/Debconf/FrontEnd/Dialog.pm line 76, <> line 35.)\n",
            "debconf: falling back to frontend: Readline\n",
            "debconf: unable to initialize frontend: Readline\n",
            "debconf: (This frontend requires a controlling tty.)\n",
            "debconf: falling back to frontend: Teletype\n",
            "dpkg-preconfigure: unable to re-open stdin: \n",
            "Selecting previously unselected package python-pkg-resources.\n",
            "(Reading database ... 155721 files and directories currently installed.)\n",
            "Preparing to unpack .../00-python-pkg-resources_39.0.1-2_all.deb ...\n",
            "Unpacking python-pkg-resources (39.0.1-2) ...\n",
            "Selecting previously unselected package python-six.\n",
            "Preparing to unpack .../01-python-six_1.11.0-2_all.deb ...\n",
            "Unpacking python-six (1.11.0-2) ...\n",
            "Selecting previously unselected package python-isodate.\n",
            "Preparing to unpack .../02-python-isodate_0.6.0-1_all.deb ...\n",
            "Unpacking python-isodate (0.6.0-1) ...\n",
            "Selecting previously unselected package python-pyparsing.\n",
            "Preparing to unpack .../03-python-pyparsing_2.2.0+dfsg1-2_all.deb ...\n",
            "Unpacking python-pyparsing (2.2.0+dfsg1-2) ...\n",
            "Selecting previously unselected package python-rdflib.\n",
            "Preparing to unpack .../04-python-rdflib_4.2.1-2_all.deb ...\n",
            "Unpacking python-rdflib (4.2.1-2) ...\n",
            "Selecting previously unselected package python-certifi.\n",
            "Preparing to unpack .../05-python-certifi_2018.1.18-2_all.deb ...\n",
            "Unpacking python-certifi (2018.1.18-2) ...\n",
            "Selecting previously unselected package python-chardet.\n",
            "Preparing to unpack .../06-python-chardet_3.0.4-1_all.deb ...\n",
            "Unpacking python-chardet (3.0.4-1) ...\n",
            "Selecting previously unselected package python-idna.\n",
            "Preparing to unpack .../07-python-idna_2.6-1_all.deb ...\n",
            "Unpacking python-idna (2.6-1) ...\n",
            "Selecting previously unselected package python-urllib3.\n",
            "Preparing to unpack .../08-python-urllib3_1.22-1ubuntu0.18.04.2_all.deb ...\n",
            "Unpacking python-urllib3 (1.22-1ubuntu0.18.04.2) ...\n",
            "Selecting previously unselected package python-requests.\n",
            "Preparing to unpack .../09-python-requests_2.18.4-2ubuntu0.1_all.deb ...\n",
            "Unpacking python-requests (2.18.4-2ubuntu0.1) ...\n",
            "Selecting previously unselected package python-ruamel.yaml.\n",
            "Preparing to unpack .../10-python-ruamel.yaml_0.15.34-1_amd64.deb ...\n",
            "Unpacking python-ruamel.yaml (0.15.34-1) ...\n",
            "Selecting previously unselected package python-cachecontrol.\n",
            "Preparing to unpack .../11-python-cachecontrol_0.11.7-1_all.deb ...\n",
            "Unpacking python-cachecontrol (0.11.7-1) ...\n",
            "Selecting previously unselected package python-lockfile.\n",
            "Preparing to unpack .../12-python-lockfile_1%3a0.12.2-2_all.deb ...\n",
            "Unpacking python-lockfile (1:0.12.2-2) ...\n",
            "Selecting previously unselected package python-mistune.\n",
            "Preparing to unpack .../13-python-mistune_0.8.3-2_all.deb ...\n",
            "Unpacking python-mistune (0.8.3-2) ...\n",
            "Selecting previously unselected package python-rdflib-jsonld.\n",
            "Preparing to unpack .../14-python-rdflib-jsonld_0.4.0-3_all.deb ...\n",
            "Unpacking python-rdflib-jsonld (0.4.0-3) ...\n",
            "Selecting previously unselected package python-typing.\n",
            "Preparing to unpack .../15-python-typing_3.6.2-1_all.deb ...\n",
            "Unpacking python-typing (3.6.2-1) ...\n",
            "Selecting previously unselected package python-avro.\n",
            "Preparing to unpack .../16-python-avro_1.8.2+dfsg-1_all.deb ...\n",
            "Unpacking python-avro (1.8.2+dfsg-1) ...\n",
            "Selecting previously unselected package libjs-bootstrap.\n",
            "Preparing to unpack .../17-libjs-bootstrap_3.3.7+dfsg-2_all.deb ...\n",
            "Unpacking libjs-bootstrap (3.3.7+dfsg-2) ...\n",
            "Selecting previously unselected package python-schema-salad.\n",
            "Preparing to unpack .../18-python-schema-salad_2.6.20171201034858-3_all.deb ...\n",
            "Unpacking python-schema-salad (2.6.20171201034858-3) ...\n",
            "Selecting previously unselected package python-shellescape.\n",
            "Preparing to unpack .../19-python-shellescape_3.4.1-2_all.deb ...\n",
            "Unpacking python-shellescape (3.4.1-2) ...\n",
            "Selecting previously unselected package cwltool.\n",
            "Preparing to unpack .../20-cwltool_1.0.20180302231433-1_all.deb ...\n",
            "Unpacking cwltool (1.0.20180302231433-1) ...\n",
            "Selecting previously unselected package libhts2:amd64.\n",
            "Preparing to unpack .../21-libhts2_1.7-2_amd64.deb ...\n",
            "Unpacking libhts2:amd64 (1.7-2) ...\n",
            "Selecting previously unselected package nodejs-doc.\n",
            "Preparing to unpack .../22-nodejs-doc_8.10.0~dfsg-2ubuntu0.4_all.deb ...\n",
            "Unpacking nodejs-doc (8.10.0~dfsg-2ubuntu0.4) ...\n",
            "Selecting previously unselected package python-asn1crypto.\n",
            "Preparing to unpack .../23-python-asn1crypto_0.24.0-1_all.deb ...\n",
            "Unpacking python-asn1crypto (0.24.0-1) ...\n",
            "Selecting previously unselected package python-cffi-backend.\n",
            "Preparing to unpack .../24-python-cffi-backend_1.11.5-1_amd64.deb ...\n",
            "Unpacking python-cffi-backend (1.11.5-1) ...\n",
            "Selecting previously unselected package python-enum34.\n",
            "Preparing to unpack .../25-python-enum34_1.1.6-2_all.deb ...\n",
            "Unpacking python-enum34 (1.1.6-2) ...\n",
            "Selecting previously unselected package python-ipaddress.\n",
            "Preparing to unpack .../26-python-ipaddress_1.0.17-1_all.deb ...\n",
            "Unpacking python-ipaddress (1.0.17-1) ...\n",
            "Selecting previously unselected package python-cryptography.\n",
            "Preparing to unpack .../27-python-cryptography_2.1.4-1ubuntu1.4_amd64.deb ...\n",
            "Unpacking python-cryptography (2.1.4-1ubuntu1.4) ...\n",
            "Selecting previously unselected package python-webencodings.\n",
            "Preparing to unpack .../28-python-webencodings_0.5-2_all.deb ...\n",
            "Unpacking python-webencodings (0.5-2) ...\n",
            "Selecting previously unselected package python-html5lib.\n",
            "Preparing to unpack .../29-python-html5lib_0.999999999-1_all.deb ...\n",
            "Unpacking python-html5lib (0.999999999-1) ...\n",
            "Selecting previously unselected package python-openssl.\n",
            "Preparing to unpack .../30-python-openssl_17.5.0-1ubuntu1_all.deb ...\n",
            "Unpacking python-openssl (17.5.0-1ubuntu1) ...\n",
            "Selecting previously unselected package python-sparqlwrapper.\n",
            "Preparing to unpack .../31-python-sparqlwrapper_1.7.6-2_all.deb ...\n",
            "Unpacking python-sparqlwrapper (1.7.6-2) ...\n",
            "Selecting previously unselected package samtools.\n",
            "Preparing to unpack .../32-samtools_1.7-1_amd64.deb ...\n",
            "Unpacking samtools (1.7-1) ...\n",
            "Selecting previously unselected package libc-ares2:amd64.\n",
            "Preparing to unpack .../33-libc-ares2_1.14.0-1ubuntu0.1_amd64.deb ...\n",
            "Unpacking libc-ares2:amd64 (1.14.0-1ubuntu0.1) ...\n",
            "Selecting previously unselected package nodejs.\n",
            "Preparing to unpack .../34-nodejs_8.10.0~dfsg-2ubuntu0.4_amd64.deb ...\n",
            "Unpacking nodejs (8.10.0~dfsg-2ubuntu0.4) ...\n",
            "Setting up python-idna (2.6-1) ...\n",
            "Setting up nodejs-doc (8.10.0~dfsg-2ubuntu0.4) ...\n",
            "Setting up python-asn1crypto (0.24.0-1) ...\n",
            "Setting up python-mistune (0.8.3-2) ...\n",
            "Setting up python-certifi (2018.1.18-2) ...\n",
            "Setting up python-avro (1.8.2+dfsg-1) ...\n",
            "Setting up python-pkg-resources (39.0.1-2) ...\n",
            "Setting up python-ruamel.yaml (0.15.34-1) ...\n",
            "Setting up python-cffi-backend (1.11.5-1) ...\n",
            "Setting up python-six (1.11.0-2) ...\n",
            "Setting up python-isodate (0.6.0-1) ...\n",
            "Setting up python-shellescape (3.4.1-2) ...\n",
            "Setting up python-enum34 (1.1.6-2) ...\n",
            "Setting up libjs-bootstrap (3.3.7+dfsg-2) ...\n",
            "Setting up python-lockfile (1:0.12.2-2) ...\n",
            "Setting up python-ipaddress (1.0.17-1) ...\n",
            "Setting up libc-ares2:amd64 (1.14.0-1ubuntu0.1) ...\n",
            "Setting up libhts2:amd64 (1.7-2) ...\n",
            "Setting up python-webencodings (0.5-2) ...\n",
            "Setting up python-typing (3.6.2-1) ...\n",
            "Setting up python-pyparsing (2.2.0+dfsg1-2) ...\n",
            "Setting up python-urllib3 (1.22-1ubuntu0.18.04.2) ...\n",
            "Setting up python-chardet (3.0.4-1) ...\n",
            "Setting up nodejs (8.10.0~dfsg-2ubuntu0.4) ...\n",
            "update-alternatives: using /usr/bin/nodejs to provide /usr/bin/js (js) in auto mode\n",
            "Setting up python-rdflib (4.2.1-2) ...\n",
            "Setting up python-sparqlwrapper (1.7.6-2) ...\n",
            "Setting up python-rdflib-jsonld (0.4.0-3) ...\n",
            "Setting up python-cryptography (2.1.4-1ubuntu1.4) ...\n",
            "Setting up python-requests (2.18.4-2ubuntu0.1) ...\n",
            "Setting up python-cachecontrol (0.11.7-1) ...\n",
            "Setting up python-schema-salad (2.6.20171201034858-3) ...\n",
            "Setting up python-html5lib (0.999999999-1) ...\n",
            "Setting up samtools (1.7-1) ...\n",
            "Setting up python-openssl (17.5.0-1ubuntu1) ...\n",
            "Setting up cwltool (1.0.20180302231433-1) ...\n",
            "Processing triggers for libc-bin (2.27-3ubuntu1.3) ...\n",
            "/sbin/ldconfig.real: /usr/local/lib/python3.7/dist-packages/ideep4py/lib/libmkldnn.so.0 is not a symbolic link\n",
            "\n",
            "Processing triggers for man-db (2.8.3-2ubuntu0.1) ...\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!samtools view -Sb  aln.pe.sam >  aln.pe.bam"
      ],
      "metadata": {
        "id": "RYhQTmhWodjn"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!samtools flagstat aln.pe.bam\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "daNbSFlFpMBt",
        "outputId": "36796a92-1746-4887-8998-025b6b8be812"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "9666214 + 0 in total (QC-passed reads + QC-failed reads)\n",
            "4879783 + 0 secondary\n",
            "1323 + 0 supplementary\n",
            "0 + 0 duplicates\n",
            "9653076 + 0 mapped (99.86% : N/A)\n",
            "4785108 + 0 paired in sequencing\n",
            "2392554 + 0 read1\n",
            "2392554 + 0 read2\n",
            "4353926 + 0 properly paired (90.99% : N/A)\n",
            "4758832 + 0 with itself and mate mapped\n",
            "13138 + 0 singletons (0.27% : N/A)\n",
            "0 + 0 with mate mapped to a different chr\n",
            "0 + 0 with mate mapped to a different chr (mapQ>=5)\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!wget https://github.com/broadinstitute/picard/releases/download/2.27.4/picard.jar "
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "mTH_JLFupkf9",
        "outputId": "45a13b3c-93f1-44d1-de45-b056114c9e31"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2022-07-26 06:26:10--  https://github.com/broadinstitute/picard/releases/download/2.27.4/picard.jar\n",
            "Resolving github.com (github.com)... 192.30.255.113\n",
            "Connecting to github.com (github.com)|192.30.255.113|:443... connected.\n",
            "HTTP request sent, awaiting response... 302 Found\n",
            "Location: https://objects.githubusercontent.com/github-production-release-asset-2e65be/18225913/839cd9dd-e7dc-4c29-ab0d-6ab4bdcded4e?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAIWNJYAX4CSVEH53A%2F20220726%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20220726T062610Z&X-Amz-Expires=300&X-Amz-Signature=f44e322dd0b81322bb3e8c5c6d2f8c12b246f7d3f5b85c2cbac8bb3f200d80c9&X-Amz-SignedHeaders=host&actor_id=0&key_id=0&repo_id=18225913&response-content-disposition=attachment%3B%20filename%3Dpicard.jar&response-content-type=application%2Foctet-stream [following]\n",
            "--2022-07-26 06:26:10--  https://objects.githubusercontent.com/github-production-release-asset-2e65be/18225913/839cd9dd-e7dc-4c29-ab0d-6ab4bdcded4e?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAIWNJYAX4CSVEH53A%2F20220726%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20220726T062610Z&X-Amz-Expires=300&X-Amz-Signature=f44e322dd0b81322bb3e8c5c6d2f8c12b246f7d3f5b85c2cbac8bb3f200d80c9&X-Amz-SignedHeaders=host&actor_id=0&key_id=0&repo_id=18225913&response-content-disposition=attachment%3B%20filename%3Dpicard.jar&response-content-type=application%2Foctet-stream\n",
            "Resolving objects.githubusercontent.com (objects.githubusercontent.com)... 185.199.111.133, 185.199.109.133, 185.199.108.133, ...\n",
            "Connecting to objects.githubusercontent.com (objects.githubusercontent.com)|185.199.111.133|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 17544754 (17M) [application/octet-stream]\n",
            "Saving to: ‘picard.jar’\n",
            "\n",
            "picard.jar          100%[===================>]  16.73M  12.9MB/s    in 1.3s    \n",
            "\n",
            "2022-07-26 06:26:12 (12.9 MB/s) - ‘picard.jar’ saved [17544754/17544754]\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!java -version"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "ZgtwISXdqEIa",
        "outputId": "e478a970-a4b2-41ff-aec7-285c9f454057"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "openjdk version \"11.0.15\" 2022-04-19\n",
            "OpenJDK Runtime Environment (build 11.0.15+10-Ubuntu-0ubuntu0.18.04.1)\n",
            "OpenJDK 64-Bit Server VM (build 11.0.15+10-Ubuntu-0ubuntu0.18.04.1, mixed mode, sharing)\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!java -jar picard.jar MarkDuplicates -I aln.pe.bam -M sample.metrics.txt -ASO queryname -O dedup.bam"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "WblaEvQJqdHU",
        "outputId": "d1bd90e8-953e-4d01-de8e-8955b35d3ce8"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "07:46:27.468 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/content/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
            "[Tue Jul 26 07:46:27 UTC 2022] MarkDuplicates --INPUT aln.pe.bam --OUTPUT dedup.bam --METRICS_FILE sample.metrics.txt --ASSUME_SORT_ORDER queryname --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --ASSUME_SORTED false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --GA4GH_CLIENT_SECRETS client_secrets.json --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
            "[Tue Jul 26 07:46:27 UTC 2022] Executing as root@acad52d69833 on Linux 5.4.188+ amd64; OpenJDK 64-Bit Server VM 11.0.15+10-Ubuntu-0ubuntu0.18.04.1; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:2.27.4-SNAPSHOT\n",
            "INFO\t2022-07-26 07:46:27\tMarkDuplicates\tStart of doWork freeMemory: 21822752; totalMemory: 28311552; maxMemory: 3405774848\n",
            "INFO\t2022-07-26 07:46:27\tMarkDuplicates\tReading input file and constructing read end information.\n",
            "INFO\t2022-07-26 07:46:27\tMarkDuplicates\tWill retain up to 12339763 data points before spilling to disk.\n",
            "INFO\t2022-07-26 07:46:28\tMarkDuplicates\tSeen many non-increasing record positions. Printing Read-names as well.\n",
            "INFO\t2022-07-26 07:46:35\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:07s.  Time for last 1,000,000:    7s.  Last read position: chr8:26,240,595.  Last read name: DCW97JN1:309:C0C42ACXX:2:2102:6994:31893\n",
            "INFO\t2022-07-26 07:46:35\tMarkDuplicates\tTracking 1 as yet unmatched pairs. 1 records in RAM.\n",
            "INFO\t2022-07-26 07:46:39\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:12s.  Time for last 1,000,000:    4s.  Last read position: chr8:59,555,370.  Last read name: DCW97JN1:309:C0C42ACXX:3:2215:17225:42351\n",
            "INFO\t2022-07-26 07:46:39\tMarkDuplicates\tTracking 1 as yet unmatched pairs. 1 records in RAM.\n",
            "INFO\t2022-07-26 07:46:43\tMarkDuplicates\tRead     3,000,000 records.  Elapsed time: 00:00:15s.  Time for last 1,000,000:    3s.  Last read position: chr8:86,727,494.  Last read name: DCW97JN1:309:C0C42ACXX:2:1207:7470:7280\n",
            "INFO\t2022-07-26 07:46:43\tMarkDuplicates\tTracking 1 as yet unmatched pairs. 1 records in RAM.\n",
            "INFO\t2022-07-26 07:46:45\tMarkDuplicates\tRead     4,000,000 records.  Elapsed time: 00:00:17s.  Time for last 1,000,000:    1s.  Last read position: chr8:86,828,993.  Last read name: DCW97JN1:309:C0C42ACXX:2:1313:1740:25279\n",
            "INFO\t2022-07-26 07:46:45\tMarkDuplicates\tTracking 1 as yet unmatched pairs. 1 records in RAM.\n",
            "INFO\t2022-07-26 07:46:47\tMarkDuplicates\tRead     5,000,000 records.  Elapsed time: 00:00:20s.  Time for last 1,000,000:    2s.  Last read position: chr8:86,818,704.  Last read name: DCW97JN1:309:C0C42ACXX:2:2112:21245:71618\n",
            "INFO\t2022-07-26 07:46:47\tMarkDuplicates\tTracking 1 as yet unmatched pairs. 1 records in RAM.\n",
            "INFO\t2022-07-26 07:46:49\tMarkDuplicates\tRead     6,000,000 records.  Elapsed time: 00:00:21s.  Time for last 1,000,000:    1s.  Last read position: chr8:86,764,592.  Last read name: DCW97JN1:309:C0C42ACXX:2:2109:17928:90334\n",
            "INFO\t2022-07-26 07:46:49\tMarkDuplicates\tTracking 0 as yet unmatched pairs. 0 records in RAM.\n",
            "INFO\t2022-07-26 07:46:51\tMarkDuplicates\tRead     7,000,000 records.  Elapsed time: 00:00:23s.  Time for last 1,000,000:    1s.  Last read position: chr8:86,756,660.  Last read name: DCW97JN1:309:C0C42ACXX:3:2306:21114:71802\n",
            "INFO\t2022-07-26 07:46:51\tMarkDuplicates\tTracking 1 as yet unmatched pairs. 1 records in RAM.\n",
            "INFO\t2022-07-26 07:46:53\tMarkDuplicates\tRead     8,000,000 records.  Elapsed time: 00:00:25s.  Time for last 1,000,000:    2s.  Last read position: chr8:91,968,060.  Last read name: DCW97JN1:309:C0C42ACXX:3:2216:17617:49983\n",
            "INFO\t2022-07-26 07:46:53\tMarkDuplicates\tTracking 1 as yet unmatched pairs. 1 records in RAM.\n",
            "INFO\t2022-07-26 07:46:57\tMarkDuplicates\tRead     9,000,000 records.  Elapsed time: 00:00:29s.  Time for last 1,000,000:    3s.  Last read position: chr8:128,753,479.  Last read name: DCW97JN1:309:C0C42ACXX:2:1210:17220:66663\n",
            "INFO\t2022-07-26 07:46:57\tMarkDuplicates\tTracking 0 as yet unmatched pairs. 0 records in RAM.\n",
            "INFO\t2022-07-26 07:46:59\tMarkDuplicates\tRead 9666214 records. 0 pairs never matched.\n",
            "INFO\t2022-07-26 07:47:00\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 1011164096; totalMemory: 1633681408; maxMemory: 3405774848\n",
            "INFO\t2022-07-26 07:47:00\tMarkDuplicates\tWill retain up to 106430464 duplicate indices before spilling to disk.\n",
            "INFO\t2022-07-26 07:47:00\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
            "INFO\t2022-07-26 07:47:04\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
            "INFO\t2022-07-26 07:47:07\tMarkDuplicates\tSorting list of duplicate records.\n",
            "INFO\t2022-07-26 07:47:07\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 775254824; totalMemory: 1634729984; maxMemory: 3405774848\n",
            "INFO\t2022-07-26 07:47:07\tMarkDuplicates\tMarking 741918 records as duplicates.\n",
            "INFO\t2022-07-26 07:47:07\tMarkDuplicates\tFound 238 optical duplicate clusters.\n",
            "INFO\t2022-07-26 07:47:07\tMarkDuplicates\tReads are assumed to be ordered by: queryname\n",
            "INFO\t2022-07-26 07:47:07\tMarkDuplicates\tOutput will not be re-sorted. Output header will state SO:unknown GO:query\n",
            "INFO\t2022-07-26 07:47:08\tMarkDuplicates\tSeen many non-increasing record positions. Printing Read-names as well.\n",
            "INFO\t2022-07-26 07:48:31\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
            "INFO\t2022-07-26 07:48:31\tMarkDuplicates\tDuplicate Index cleanup.\n",
            "INFO\t2022-07-26 07:48:31\tMarkDuplicates\tGetting Memory Stats.\n",
            "INFO\t2022-07-26 07:48:31\tMarkDuplicates\tBefore output close freeMemory: 24296464; totalMemory: 31457280; maxMemory: 3405774848\n",
            "INFO\t2022-07-26 07:48:31\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
            "INFO\t2022-07-26 07:48:31\tMarkDuplicates\tAfter output close freeMemory: 24296464; totalMemory: 31457280; maxMemory: 3405774848\n",
            "[Tue Jul 26 07:48:31 UTC 2022] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 2.07 minutes.\n",
            "Runtime.totalMemory()=31457280\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!java -jar picard.jar SortSam -I dedup.bam -O dedup_sorted.bam -SORT_ORDER coordinate"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "veQt2Kur93Hk",
        "outputId": "73d8a44b-6680-4cea-ee8b-9ab6abcb7632"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "07:50:15.588 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/content/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
            "[Tue Jul 26 07:50:15 UTC 2022] SortSam --INPUT dedup.bam --OUTPUT dedup_sorted.bam --SORT_ORDER coordinate --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --GA4GH_CLIENT_SECRETS client_secrets.json --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
            "[Tue Jul 26 07:50:15 UTC 2022] Executing as root@acad52d69833 on Linux 5.4.188+ amd64; OpenJDK 64-Bit Server VM 11.0.15+10-Ubuntu-0ubuntu0.18.04.1; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:2.27.4-SNAPSHOT\n",
            "INFO\t2022-07-26 07:50:15\tSortSam\tSeen many non-increasing record positions. Printing Read-names as well.\n",
            "INFO\t2022-07-26 07:51:38\tSortSam\tFinished reading inputs, merging and writing to output now.\n",
            "[Tue Jul 26 07:52:47 UTC 2022] picard.sam.SortSam done. Elapsed time: 2.54 minutes.\n",
            "Runtime.totalMemory()=122683392\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!samtools index -@ 2 dedup_sorted.bam dedup_sorted.bai"
      ],
      "metadata": {
        "id": "5uBIEUER-3JY"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!wget https://github.com/broadinstitute/gatk/releases/download/4.2.6.1/gatk-4.2.6.1.zip"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "6gPahbjttUuT",
        "outputId": "54fe4d97-3592-4f51-cfa4-6d2d2acc3dd2"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2022-07-26 06:27:03--  https://github.com/broadinstitute/gatk/releases/download/4.2.6.1/gatk-4.2.6.1.zip\n",
            "Resolving github.com (github.com)... 192.30.255.113\n",
            "Connecting to github.com (github.com)|192.30.255.113|:443... connected.\n",
            "HTTP request sent, awaiting response... 302 Found\n",
            "Location: https://objects.githubusercontent.com/github-production-release-asset-2e65be/27452807/dd1c357e-1e7c-4aac-b133-886b16539c39?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAIWNJYAX4CSVEH53A%2F20220726%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20220726T062703Z&X-Amz-Expires=300&X-Amz-Signature=08b193d1dbd5327a6beebecd21261f5b4be710ce6d7a49ebf67d1eb738f0c589&X-Amz-SignedHeaders=host&actor_id=0&key_id=0&repo_id=27452807&response-content-disposition=attachment%3B%20filename%3Dgatk-4.2.6.1.zip&response-content-type=application%2Foctet-stream [following]\n",
            "--2022-07-26 06:27:03--  https://objects.githubusercontent.com/github-production-release-asset-2e65be/27452807/dd1c357e-1e7c-4aac-b133-886b16539c39?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAIWNJYAX4CSVEH53A%2F20220726%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20220726T062703Z&X-Amz-Expires=300&X-Amz-Signature=08b193d1dbd5327a6beebecd21261f5b4be710ce6d7a49ebf67d1eb738f0c589&X-Amz-SignedHeaders=host&actor_id=0&key_id=0&repo_id=27452807&response-content-disposition=attachment%3B%20filename%3Dgatk-4.2.6.1.zip&response-content-type=application%2Foctet-stream\n",
            "Resolving objects.githubusercontent.com (objects.githubusercontent.com)... 185.199.109.133, 185.199.110.133, 185.199.111.133, ...\n",
            "Connecting to objects.githubusercontent.com (objects.githubusercontent.com)|185.199.109.133|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 453893126 (433M) [application/octet-stream]\n",
            "Saving to: ‘gatk-4.2.6.1.zip’\n",
            "\n",
            "gatk-4.2.6.1.zip    100%[===================>] 432.87M  4.97MB/s    in 47s     \n",
            "\n",
            "2022-07-26 06:27:50 (9.27 MB/s) - ‘gatk-4.2.6.1.zip’ saved [453893126/453893126]\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!unzip gatk-4.2.6.1.zip"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "9tWdJQTO8LQj",
        "outputId": "e3feb886-4e2d-48c6-9fa1-842a02d92708"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Archive:  gatk-4.2.6.1.zip\n",
            "   creating: gatk-4.2.6.1/\n",
            "   creating: gatk-4.2.6.1/gatkdoc/\n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_rnaseq_ASEReadCounter.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_analysis_replicates_CollectIndependentReplicateMetrics.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_dragstr_CalibrateDragstrModel.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_ChromosomeCounts.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_engine_filters_LibraryReadFilter.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_vqsr_FilterVariantTranches.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_analysis_CollectWgsMetricsWithNonZeroCoverage.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$NonChimericOriginalAlignmentReadFilter.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_analysis_MeanQualityByCycle.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_illumina_quality_CollectHiSeqXPfFailMetrics.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_spark_pipelines_PrintReadsSpark.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_vcf_GenotypeConcordance.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_sam_CheckTerminatorBlock.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadLengthReadFilter.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_qc_Pileup.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_engine_filters_OverclippedReadFilter.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_sam_CleanSam.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_mutect_CreateSomaticPanelOfNormals.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_arrays_GtcToVcf.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_qc_CheckPileup.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_BwaMemIndexImageCreator.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_OrientationBiasReadCounts.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_vcf_RenameSampleInVcf.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_ReadAnonymizer.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$MappingQualityAvailableReadFilter.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_CountNs.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_AssemblyComplexity.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_analysis_CollectOxoGMetrics.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_arrays_VerifyIDIntensityContaminationMetrics.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$NotProperlyPairedReadFilter.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_analysis_FingerprintingDetailMetrics.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_engine_filters_ExcessiveEndClippedReadFilter.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_fasta_FastaReferenceMaker.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_illumina_BarcodeMetric.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_engine_filters_WellformedReadFilter.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_AlleleFraction.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_spark_pipelines_metrics_QualityScoreDistributionSpark.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_DepthPerSampleHC.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_engine_filters_SampleReadFilter.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_bqsr_AnalyzeCovariates.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_RMSMappingQuality.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$FirstOfPairReadFilter.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_analysis_artifacts_ConvertSequencingArtifactToOxoG.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_spark_pipelines_CountBasesSpark.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_spark_pipelines_CountReadsSpark.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_funcotator_Funcotator.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_analysis_CollectQualityYieldMetrics.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_copynumber_CreateReadCountPanelOfNormals.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_spark_pathseq_PathSeqPipelineSpark.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_spark_sv_evidence_CalcMetadataSpark.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$SecondOfPairReadFilter.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_engine_filters_AlignmentAgreesWithHeaderReadFilter.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_sam_markduplicates_CollectDuplicateMetrics.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_allelespecific_AS_StrandOddsRatio.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_vcf_filter_FilterVcf.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_analysis_CollectWgsMetricsWithNonZeroCoverage$WgsMetricsWithNonZeroCoverage.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_spark_ParallelCopyGCSDirectoryIntoHDFSSpark.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_sam_SamErrorMetric_IndelErrorMetric.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_fasta_ShiftFasta.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_varianteval_VariantEval.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_analysis_GcBiasSummaryMetrics.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_analysis_CollectRrbsMetrics.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_analysis_RnaSeqMetrics.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_vcf_MendelianViolations_FindMendelianViolations.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$NotSecondaryAlignmentReadFilter.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_CountReads.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_vcf_FixVcfHeader.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_analysis_directed_CollectHsMetrics.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_util_AccumulateQualityYieldMetrics.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_illumina_IlluminaBasecallingMetrics.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_gnarlyGenotyper_GnarlyGenotyper.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_analysis_JumpingLibraryMetrics.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$ValidAlignmentEndReadFilter.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_sam_markduplicates_EstimateLibraryComplexity.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_ReferenceBases.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_PrintReadsHeader.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_sam_ViewSam.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_validation_AnnotateVcfWithExpectedAlleleFraction.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_sam_SamErrorMetric_ErrorMetric.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_CountVariants.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_IndexFeatureFile.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$AllowAllReadsReadFilter.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_util_LiftOverIntervalList.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_engine_filters_MetricsReadFilter.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$NotDuplicateReadFilter.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_spark_sv_evidence_FindBadGenomicKmersSpark.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_sam_SamErrorMetric_OverlappingErrorMetric.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_fastq_BamToBfq.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_sam_AddCommentsToBam.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_allelespecific_AS_QualByDepth.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_AnalyzeSaturationMutagenesis.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_vcf_CollectVariantCallingMetrics$VariantCallingSummaryMetrics.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_GenotypeGVCFs.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_analysis_artifacts_SequencingArtifactMetrics$PreAdapterSummaryMetrics.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_TandemRepeat.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_arrays_CollectArraysVariantCallingMetrics$ArraysControlCodesSummaryMetrics.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_sam_BuildBamIndex.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_VariantAnnotator.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_sam_FilterSamReads.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_allelespecific_AS_BaseQualityRankSumTest.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_arrays_MergePedIntoVcf.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_validation_Concordance.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_engine_filters_FragmentLengthReadFilter.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_rnaseq_SplitNCigarReads.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$PrimaryLineReadFilter.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$NonZeroFragmentLengthReadFilter.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_arrays_BafRegressMetrics.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_spark_pipelines_metrics_CollectMultipleMetricsSpark.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_spark_sv_StructuralVariationDiscoveryPipelineSpark.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_validation_EvaluateInfoFieldConcordance.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_CountBases.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_StrandOddsRatio.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_vcf_SplitVcfs.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/index.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_sam_SplitSamByLibrary.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_funcotator_FilterFuncotations.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_vcf_MendelianViolations_MendelianViolationMetrics.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_fingerprint_CheckFingerprint.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_vqsr_CNNVariantTrain.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$HasReadGroupReadFilter.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_fingerprint_CrosscheckFingerprints.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_coverage_DepthOfCoverage.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_sam_SamErrorMetric_BaseErrorMetric.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_spark_PileupSpark.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_copynumber_DetermineGermlineContigPloidy.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_spark_sv_evidence_FindBreakpointEvidenceSpark.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_UniqueAltReadCount.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$ProperlyPairedReadFilter.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_MappingQuality.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$CigarContainsNoNOperator.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_copynumber_plotting_PlotDenoisedCopyRatios.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_QualByDepth.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_variantutils_VariantsToTable.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_spark_pathseq_PathSeqFilterSpark.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_vqsr_VariantRecalibrator.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_spark_pipelines_metrics_CollectBaseDistributionByCycleSpark.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_spark_sv_utils_ExtractOriginalAlignmentRecordsByNameSpark.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_reference_NormalizeFasta.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_UnmarkDuplicates.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_allelespecific_AS_RMSMappingQuality.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_illumina_quality_CollectHiSeqXPfFailMetrics$PFFailSummaryMetric.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_vqsr_CNNVariantWriteTensors.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_SplitIntervals.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_fasta_FastaAlternateReferenceMaker.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_spark_pipelines_CountVariantsSpark.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_variantutils_UpdateVCFSequenceDictionary.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_illumina_IlluminaBasecallsToFastq.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_sam_SamFormatConverter.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_qc_PostProcessReadsForRSEM.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_FisherStrand.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_rnaseq_GeneExpressionEvaluation.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_fingerprint_ClusterCrosscheckMetrics.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_ClipReads.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_ReadPosRankSumTest.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_sam_SamErrorMetric_CollectSamErrorMetrics.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_engine_filters_SoftClippedReadFilter.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_GetSampleName.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_allelespecific_AS_MappingQualityRankSumTest.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_engine_filters_IntervalOverlapReadFilter.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_spark_transforms_markduplicates_MarkDuplicatesSpark.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_DepthPerAlleleBySample.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$NonZeroReferenceLengthAlignmentReadFilter.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_sam_CalculateReadGroupChecksum.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_BaseQuality.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$SeqIsStoredReadFilter.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_sam_SetNmMdAndUqTags.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_metrics_GcBiasMetrics.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_analysis_CompareMetrics.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_funcotator_FuncotateSegments.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_sam_GatherBamFiles.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_analysis_artifacts_SequencingArtifactMetrics$BaitBiasSummaryMetrics.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_fingerprint_CalculateFingerprintMetrics.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_funcotator_FuncotatorDataSourceDownloader.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_ConvertHeaderlessHadoopBamShardToBam.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_spark_pathseq_PathSeqBuildReferenceTaxonomy.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_allelespecific_AS_FisherStrand.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_validation_CompareBaseQualities.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_analysis_replicates_IndependentReplicateMetric.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_spark_bwa_BwaSpark.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadGroupBlackListReadFilter.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_sam_SamComparisonMetric.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_sam_FastqToSam.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_analysis_CollectAlignmentSummaryMetrics.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_analysis_WgsMetrics.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_illumina_CollectIlluminaBasecallingMetrics.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_analysis_GcBiasDetailMetrics.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_sam_SortSam.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_validation_AnnotateVcfWithBamDepth.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_sam_SamToFastq.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_analysis_artifacts_CollectSequencingArtifactMetrics.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_spark_pathseq_PathSeqScoreSpark.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_sam_RevertSam.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_vcf_GenotypeConcordanceContingencyMetrics.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_vcf_GatherVcfs.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$MappedReadFilter.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_SplitReads.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$PassesVendorQualityCheckReadFilter.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_reference_ExtractSequences.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_sam_markduplicates_MarkDuplicatesWithMateCigar.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_analysis_CollectRnaSeqMetrics.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_bqsr_GatherBQSRReports.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_contamination_CalculateContamination.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_sam_MergeSamFiles.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_bqsr_ApplyBQSR.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_GatherVcfsCloud.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_spark_sv_discovery_inference_CpxVariantReInterpreterSpark.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_copynumber_AnnotateIntervals.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_spark_pathseq_PathSeqBuildKmers.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_vcf_MakeSitesOnlyVcf.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_util_IntervalListTools.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_InbreedingCoeff.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_illumina_IlluminaPhasingMetrics.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_analysis_CollectInsertSizeMetrics.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$ReadLengthEqualsCigarLengthReadFilter.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_arrays_CombineGenotypingArrayVcfs.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_StrandBiasBySample.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_util_BaitDesigner.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_FragmentDepthPerAlleleBySample.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_readorientation_LearnReadOrientationModel.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_sam_ValidateSamFile.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_copynumber_CollectReadCounts.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_arrays_illumina_BpmToNormalizationManifestCsv.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_analysis_InsertSizeMetrics.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_spark_pipelines_SortSamSpark.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_LeftAlignIndels.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_copynumber_FilterIntervals.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_MappingQualityRankSumTest.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$MappingQualityNotZeroReadFilter.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_vcf_SortVcf.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_mutect_GetNormalArtifactData.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_sam_FixMateInformation.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_spark_pipelines_PrintVariantsSpark.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_HaplotypeCallerSpark.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_analysis_CollectJumpingLibraryMetrics.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_sam_SplitSamByNumberOfReads.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_analysis_directed_HsMetrics.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_dragstr_ComposeSTRTableFile.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_MethylationTypeCaller.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_analysis_CollectOxoGMetrics$CpcgMetrics.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$MateDifferentStrandReadFilter.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_reference_NonNFastaSize.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_mutect_filtering_FilterMutectCalls.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_fingerprint_FingerprintMetrics.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_spark_BaseRecalibratorSpark.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_analysis_artifacts_SequencingArtifactMetrics$BaitBiasDetailMetrics.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_validation_CalculateMixingFractions.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_variantutils_SelectVariants.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_vcf_AccumulateVariantCallingMetrics.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_illumina_IlluminaBasecallsToSam.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_sv_SVCluster.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_FlagStat.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_PrintDistantMates.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_analysis_FingerprintingSummaryMetrics.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_copynumber_PreprocessIntervals.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_analysis_QualityScoreDistribution.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadGroupReadFilter.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_engine_filters_MappingQualityReadFilter.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/gatkDoc.css  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_analysis_CollectMultipleMetrics.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_variantutils_CalculateGenotypePosteriors.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_genomicsdb_GenomicsDBImport.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_vcf_MergeVcfs.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_spark_pipelines_metrics_CollectQualityYieldMetricsSpark.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_engine_filters_PlatformUnitReadFilter.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_vqsr_CNNScoreVariants.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_sam_RevertOriginalBaseQualitiesAndAddMateCigar.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_arrays_CreateBafRegressMetricsFile.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_spark_sv_evidence_ExtractSVEvidenceSpark.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_variantutils_ValidateVariants.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_copynumber_CollectAllelicCounts.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_spark_CreateHadoopBamSplittingIndex.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_PrintReads.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_arrays_CollectArraysVariantCallingMetrics$ArraysVariantCallingSummaryMetrics.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_vcf_GenotypeConcordanceSummaryMetrics.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_vcf_LiftoverVcf.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_spark_pipelines_BwaAndMarkDuplicatesPipelineSpark.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_sam_AddOATag.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_engine_filters_MateDistantReadFilter.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_haplotypecaller_HaplotypeCaller.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_analysis_AlignmentSummaryMetrics.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_spark_ApplyBQSRSpark.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_arrays_CreateVerifyIDIntensityContaminationMetricsFile.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_ReadPosition.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_Coverage.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$ValidAlignmentStartReadFilter.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$MateOnSameContigOrNoMappedMateReadFilter.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_spark_pipelines_ReadsPipelineSpark.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadStrandFilter.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_analysis_directed_TargetedPcrMetrics.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_sam_ReplaceSamHeader.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_validation_CountFalsePositives.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$MateUnmappedAndUnmappedReadFilter.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_analysis_directed_TargetMetrics.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_sam_AddOrReplaceReadGroups.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_vcf_VcfFormatConverter.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_sam_BamIndexStats.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_illumina_IlluminaLaneMetrics.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_mutect_Mutect2.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_arrays_CollectArraysVariantCallingMetrics.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_illumina_CollectIlluminaLaneMetrics.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_fasta_CountBasesInReference.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_fingerprint_ClusteredCrosscheckMetric.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_CombineGVCFs.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_analysis_CollectGcBiasMetrics.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_LocalAssembler.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_vcf_VcfToIntervalList.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_sam_SetNmAndUqTags.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_spark_pathseq_PathSeqBwaSpark.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_analysis_TheoreticalSensitivityMetrics.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_copynumber_plotting_PlotModeledSegments.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_illumina_MarkIlluminaAdapters.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_vcf_CollectVariantCallingMetrics.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_SampleList.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_vcf_UpdateVcfSequenceDictionary.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$MatchingBasesAndQualsReadFilter.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_spark_sv_discovery_DiscoverVariantsFromContigAlignmentsSAMSpark.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_copynumber_GermlineCNVCaller.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_spark_pipelines_metrics_MeanQualityByCycleSpark.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_FragmentLength.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_illumina_CheckIlluminaDirectory.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadNameReadFilter.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_analysis_directed_CollectTargetedPcrMetrics.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_analysis_CollectBaseDistributionByCycle.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_vqsr_ApplyVQSR.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_analysis_RrbsSummaryMetrics.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_variantutils_ReblockGVCF.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_analysis_RrbsCpgDetailMetrics.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_analysis_artifacts_ErrorSummaryMetrics.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_copynumber_CallCopyRatioSegments.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_allelespecific_AS_InbreedingCoeff.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_engine_filters_PlatformReadFilter.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_util_ScatterIntervalsByNs.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_analysis_CollectQualityYieldMetrics$QualityYieldMetrics.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_sam_MergeBamAlignment.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_FixMisencodedBaseQualityReads.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_MappingQualityZero.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_variantutils_LeftAlignAndTrimVariants.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_copynumber_ModelSegments.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_FeaturizedReadSets.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_bqsr_BaseRecalibrator.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_sam_DuplicationMetrics.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_vqsr_GatherTranches.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_spark_RevertSamSpark.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_vcf_GenotypeConcordanceDetailMetrics.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_illumina_ExtractIlluminaBarcodes.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_RevertBaseQualityScores.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_illumina_quality_CollectHiSeqXPfFailMetrics$PFFailDetailedMetric.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_sam_DownsampleSam.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_ClippingRankSumTest.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_engine_filters_AmbiguousBaseReadFilter.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_util_BedToIntervalList.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_ExcessHet.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_validation_RemoveNearbyIndels.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_filters_VariantFiltration.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_LikelihoodRankSumTest.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_spark_validation_CompareDuplicatesSpark.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_analysis_CollectRawWgsMetrics.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_PossibleDeNovo.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_arrays_CollectArraysVariantCallingMetrics$ArraysVariantCallingDetailMetrics.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_qc_TransferReadTags.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_spark_pipelines_metrics_CollectInsertSizeMetricsSpark.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_BaseQualityRankSumTest.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_spark_sv_discovery_SvDiscoverFromLocalAssemblyContigAlignmentsSpark.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_copynumber_DenoiseReadCounts.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_util_IntervalListToBed.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_sam_CompareSAMs.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_vcf_CollectVariantCallingMetrics$VariantCallingDetailMetrics.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_validation_basicshortmutpileup_ValidateBasicSomaticShortMutations.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_OriginalAlignment.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_spark_pipelines_BQSRPipelineSpark.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_sam_markduplicates_MarkDuplicates.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_sam_markduplicates_UmiAwareMarkDuplicatesWithMateCigar.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_sam_CreateSequenceDictionary.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_arrays_VcfToAdpc.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$GoodCigarReadFilter.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_analysis_BaseDistributionByCycleMetrics.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_realignmentfilter_FilterAlignmentArtifacts.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_sam_markduplicates_SimpleMarkDuplicatesWithMateCigar.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_GenotypeSummaries.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_spark_pipelines_FlagStatSpark.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_analysis_CollectWgsMetrics.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_sam_markduplicates_UmiMetrics.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_sam_ReorderSam.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_analysis_CollectRawWgsMetrics$RawWgsMetrics.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_AllelePseudoDepth.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_copynumber_PostprocessGermlineCNVCalls.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$NotSupplementaryAlignmentReadFilter.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_allelespecific_AS_ReadPosRankSumTest.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_util_FifoBuffer.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_sam_PositionBasedDownsampleSam.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_contamination_GetPileupSummaries.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$PairedReadFilter.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_analysis_artifacts_SequencingArtifactMetrics$PreAdapterDetailMetrics.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_sam_ReorderSam.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_analysis_CollectRawWgsMetrics$RawWgsMetrics.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_AllelePseudoDepth.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_copynumber_PostprocessGermlineCNVCalls.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$NotSupplementaryAlignmentReadFilter.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_analysis_BaseDistributionByCycleMetrics.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_GenotypeSummaries.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_spark_pipelines_FlagStatSpark.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_sam_markduplicates_SimpleMarkDuplicatesWithMateCigar.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_realignmentfilter_FilterAlignmentArtifacts.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_analysis_CollectWgsMetrics.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_sam_markduplicates_UmiMetrics.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$PairedReadFilter.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_analysis_artifacts_SequencingArtifactMetrics$PreAdapterDetailMetrics.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_allelespecific_AS_ReadPosRankSumTest.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_contamination_GetPileupSummaries.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_util_FifoBuffer.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_sam_PositionBasedDownsampleSam.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_engine_filters_AmbiguousBaseReadFilter.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_util_BedToIntervalList.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_ExcessHet.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_validation_RemoveNearbyIndels.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_filters_VariantFiltration.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_analysis_CollectRawWgsMetrics.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_spark_validation_CompareDuplicatesSpark.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_LikelihoodRankSumTest.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_bqsr_BaseRecalibrator.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_sam_DuplicationMetrics.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_illumina_ExtractIlluminaBarcodes.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_RevertBaseQualityScores.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_spark_RevertSamSpark.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_vcf_GenotypeConcordanceDetailMetrics.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_vqsr_GatherTranches.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_illumina_quality_CollectHiSeqXPfFailMetrics$PFFailDetailedMetric.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_sam_DownsampleSam.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_ClippingRankSumTest.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_validation_basicshortmutpileup_ValidateBasicSomaticShortMutations.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_vcf_CollectVariantCallingMetrics$VariantCallingDetailMetrics.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_sam_CompareSAMs.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_spark_pipelines_BQSRPipelineSpark.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_OriginalAlignment.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_sam_markduplicates_UmiAwareMarkDuplicatesWithMateCigar.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_sam_markduplicates_MarkDuplicates.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_arrays_VcfToAdpc.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$GoodCigarReadFilter.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_sam_CreateSequenceDictionary.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_arrays_CollectArraysVariantCallingMetrics$ArraysVariantCallingDetailMetrics.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_qc_TransferReadTags.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_PossibleDeNovo.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_spark_pipelines_metrics_CollectInsertSizeMetricsSpark.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_copynumber_DenoiseReadCounts.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_BaseQualityRankSumTest.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_spark_sv_discovery_SvDiscoverFromLocalAssemblyContigAlignmentsSpark.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_util_IntervalListToBed.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_vqsr_ApplyVQSR.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_analysis_RrbsSummaryMetrics.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_variantutils_ReblockGVCF.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_spark_pipelines_metrics_MeanQualityByCycleSpark.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_spark_sv_discovery_DiscoverVariantsFromContigAlignmentsSAMSpark.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_copynumber_GermlineCNVCaller.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_vcf_UpdateVcfSequenceDictionary.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$MatchingBasesAndQualsReadFilter.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_illumina_CheckIlluminaDirectory.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_FragmentLength.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_analysis_directed_CollectTargetedPcrMetrics.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadNameReadFilter.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_analysis_CollectBaseDistributionByCycle.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_FixMisencodedBaseQualityReads.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_sam_MergeBamAlignment.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_MappingQualityZero.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_FeaturizedReadSets.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_copynumber_ModelSegments.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_variantutils_LeftAlignAndTrimVariants.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_analysis_RrbsCpgDetailMetrics.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_analysis_artifacts_ErrorSummaryMetrics.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_copynumber_CallCopyRatioSegments.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_util_ScatterIntervalsByNs.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_allelespecific_AS_InbreedingCoeff.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_engine_filters_PlatformReadFilter.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_analysis_CollectQualityYieldMetrics$QualityYieldMetrics.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_illumina_IlluminaLaneMetrics.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_sam_BamIndexStats.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_mutect_Mutect2.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_arrays_CollectArraysVariantCallingMetrics.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_validation_CountFalsePositives.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$MateUnmappedAndUnmappedReadFilter.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_analysis_directed_TargetMetrics.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_vcf_VcfFormatConverter.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_sam_AddOrReplaceReadGroups.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_spark_pathseq_PathSeqBwaSpark.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_sam_SetNmAndUqTags.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_copynumber_plotting_PlotModeledSegments.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_vcf_CollectVariantCallingMetrics.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_illumina_MarkIlluminaAdapters.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_analysis_TheoreticalSensitivityMetrics.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_SampleList.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_CombineGVCFs.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_fingerprint_ClusteredCrosscheckMetric.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_fasta_CountBasesInReference.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_illumina_CollectIlluminaLaneMetrics.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_LocalAssembler.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_analysis_CollectGcBiasMetrics.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_vcf_VcfToIntervalList.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_vcf_LiftoverVcf.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_vcf_GenotypeConcordanceSummaryMetrics.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_arrays_CollectArraysVariantCallingMetrics$ArraysVariantCallingSummaryMetrics.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_spark_pipelines_BwaAndMarkDuplicatesPipelineSpark.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_sam_AddOATag.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_engine_filters_MateDistantReadFilter.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_spark_CreateHadoopBamSplittingIndex.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_PrintReads.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_ReadPosition.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_spark_pipelines_ReadsPipelineSpark.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$ValidAlignmentStartReadFilter.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$MateOnSameContigOrNoMappedMateReadFilter.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_Coverage.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_sam_ReplaceSamHeader.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadStrandFilter.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_analysis_directed_TargetedPcrMetrics.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_haplotypecaller_HaplotypeCaller.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_analysis_AlignmentSummaryMetrics.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_spark_ApplyBQSRSpark.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_arrays_CreateVerifyIDIntensityContaminationMetricsFile.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_variantutils_CalculateGenotypePosteriors.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_genomicsdb_GenomicsDBImport.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_vcf_MergeVcfs.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_analysis_CollectMultipleMetrics.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_arrays_CreateBafRegressMetricsFile.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_variantutils_ValidateVariants.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_spark_sv_evidence_ExtractSVEvidenceSpark.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_copynumber_CollectAllelicCounts.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_engine_filters_PlatformUnitReadFilter.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_spark_pipelines_metrics_CollectQualityYieldMetricsSpark.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_vqsr_CNNScoreVariants.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_sam_RevertOriginalBaseQualitiesAndAddMateCigar.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_variantutils_SelectVariants.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_illumina_IlluminaBasecallsToSam.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_vcf_AccumulateVariantCallingMetrics.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_dragstr_ComposeSTRTableFile.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_analysis_CollectJumpingLibraryMetrics.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_sam_SplitSamByNumberOfReads.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_analysis_directed_HsMetrics.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$MateDifferentStrandReadFilter.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_reference_NonNFastaSize.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_analysis_CollectOxoGMetrics$CpcgMetrics.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_MethylationTypeCaller.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_spark_BaseRecalibratorSpark.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_mutect_filtering_FilterMutectCalls.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_fingerprint_FingerprintMetrics.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_validation_CalculateMixingFractions.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_analysis_artifacts_SequencingArtifactMetrics$BaitBiasDetailMetrics.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_analysis_FingerprintingSummaryMetrics.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_copynumber_PreprocessIntervals.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_engine_filters_MappingQualityReadFilter.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_analysis_QualityScoreDistribution.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadGroupReadFilter.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_sv_SVCluster.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_PrintDistantMates.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_FlagStat.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_arrays_CombineGenotypingArrayVcfs.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_util_BaitDesigner.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_StrandBiasBySample.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_vcf_MakeSitesOnlyVcf.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_spark_pathseq_PathSeqBuildKmers.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_util_IntervalListTools.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_illumina_IlluminaPhasingMetrics.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_InbreedingCoeff.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$ReadLengthEqualsCigarLengthReadFilter.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_analysis_CollectInsertSizeMetrics.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_MappingQualityRankSumTest.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_copynumber_FilterIntervals.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_LeftAlignIndels.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$MappingQualityNotZeroReadFilter.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_sam_FixMateInformation.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_mutect_GetNormalArtifactData.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_vcf_SortVcf.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_HaplotypeCallerSpark.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_spark_pipelines_PrintVariantsSpark.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_copynumber_CollectReadCounts.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_sam_ValidateSamFile.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_readorientation_LearnReadOrientationModel.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_FragmentDepthPerAlleleBySample.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_arrays_illumina_BpmToNormalizationManifestCsv.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_spark_pipelines_SortSamSpark.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_analysis_InsertSizeMetrics.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_analysis_replicates_IndependentReplicateMetric.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_sam_FastqToSam.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_sam_SamComparisonMetric.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_spark_bwa_BwaSpark.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadGroupBlackListReadFilter.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_analysis_WgsMetrics.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_analysis_CollectAlignmentSummaryMetrics.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_sam_SamToFastq.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_sam_SortSam.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_validation_AnnotateVcfWithBamDepth.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_analysis_GcBiasDetailMetrics.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_illumina_CollectIlluminaBasecallingMetrics.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_sam_GatherBamFiles.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_analysis_artifacts_SequencingArtifactMetrics$BaitBiasSummaryMetrics.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_funcotator_FuncotatorDataSourceDownloader.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_fingerprint_CalculateFingerprintMetrics.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_allelespecific_AS_FisherStrand.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_spark_pathseq_PathSeqBuildReferenceTaxonomy.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_ConvertHeaderlessHadoopBamShardToBam.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_validation_CompareBaseQualities.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_sam_markduplicates_MarkDuplicatesWithMateCigar.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_sam_MergeSamFiles.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_bqsr_ApplyBQSR.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_contamination_CalculateContamination.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_analysis_CollectRnaSeqMetrics.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_bqsr_GatherBQSRReports.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_GatherVcfsCloud.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_copynumber_AnnotateIntervals.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_spark_sv_discovery_inference_CpxVariantReInterpreterSpark.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_analysis_artifacts_CollectSequencingArtifactMetrics.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_spark_pathseq_PathSeqScoreSpark.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_vcf_GenotypeConcordanceContingencyMetrics.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_sam_RevertSam.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_vcf_GatherVcfs.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_reference_ExtractSequences.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$PassesVendorQualityCheckReadFilter.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_SplitReads.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$MappedReadFilter.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_engine_filters_SoftClippedReadFilter.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_sam_SamErrorMetric_CollectSamErrorMetrics.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_allelespecific_AS_MappingQualityRankSumTest.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_GetSampleName.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_qc_PostProcessReadsForRSEM.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_sam_SamFormatConverter.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_FisherStrand.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_rnaseq_GeneExpressionEvaluation.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_fingerprint_ClusterCrosscheckMetrics.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_ClipReads.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_ReadPosRankSumTest.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_BaseQuality.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$SeqIsStoredReadFilter.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_sam_SetNmMdAndUqTags.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_metrics_GcBiasMetrics.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_analysis_CompareMetrics.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_funcotator_FuncotateSegments.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_DepthPerAlleleBySample.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_spark_transforms_markduplicates_MarkDuplicatesSpark.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_engine_filters_IntervalOverlapReadFilter.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_sam_CalculateReadGroupChecksum.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$NonZeroReferenceLengthAlignmentReadFilter.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_spark_pipelines_metrics_CollectBaseDistributionByCycleSpark.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_vqsr_VariantRecalibrator.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_spark_pathseq_PathSeqFilterSpark.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_spark_sv_utils_ExtractOriginalAlignmentRecordsByNameSpark.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_reference_NormalizeFasta.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_copynumber_plotting_PlotDenoisedCopyRatios.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_QualByDepth.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_variantutils_VariantsToTable.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_illumina_quality_CollectHiSeqXPfFailMetrics$PFFailSummaryMetric.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_SplitIntervals.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_vqsr_CNNVariantWriteTensors.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_fasta_FastaAlternateReferenceMaker.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_variantutils_UpdateVCFSequenceDictionary.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_illumina_IlluminaBasecallsToFastq.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_spark_pipelines_CountVariantsSpark.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_UnmarkDuplicates.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_allelespecific_AS_RMSMappingQuality.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_spark_pipelines_metrics_CollectMultipleMetricsSpark.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_spark_sv_StructuralVariationDiscoveryPipelineSpark.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_StrandOddsRatio.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_CountBases.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_validation_EvaluateInfoFieldConcordance.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_rnaseq_SplitNCigarReads.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$PrimaryLineReadFilter.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$NonZeroFragmentLengthReadFilter.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_arrays_BafRegressMetrics.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_coverage_DepthOfCoverage.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_copynumber_DetermineGermlineContigPloidy.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_spark_sv_evidence_FindBreakpointEvidenceSpark.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_UniqueAltReadCount.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_spark_PileupSpark.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_sam_SamErrorMetric_BaseErrorMetric.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$ProperlyPairedReadFilter.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$CigarContainsNoNOperator.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_MappingQuality.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_vcf_SplitVcfs.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_funcotator_FilterFuncotations.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_vcf_MendelianViolations_MendelianViolationMetrics.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_sam_SplitSamByLibrary.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_fingerprint_CrosscheckFingerprints.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$HasReadGroupReadFilter.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_fingerprint_CheckFingerprint.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_vqsr_CNNVariantTrain.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_vcf_CollectVariantCallingMetrics$VariantCallingSummaryMetrics.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_GenotypeGVCFs.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_arrays_CollectArraysVariantCallingMetrics$ArraysControlCodesSummaryMetrics.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_TandemRepeat.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_analysis_artifacts_SequencingArtifactMetrics$PreAdapterSummaryMetrics.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_spark_sv_evidence_FindBadGenomicKmersSpark.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_sam_SamErrorMetric_OverlappingErrorMetric.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_fastq_BamToBfq.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_sam_AddCommentsToBam.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_AnalyzeSaturationMutagenesis.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_allelespecific_AS_QualByDepth.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_VariantAnnotator.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_sam_FilterSamReads.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_allelespecific_AS_BaseQualityRankSumTest.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_arrays_MergePedIntoVcf.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_validation_Concordance.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_engine_filters_FragmentLengthReadFilter.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_sam_BuildBamIndex.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_analysis_RnaSeqMetrics.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_analysis_CollectRrbsMetrics.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$NotSecondaryAlignmentReadFilter.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_vcf_MendelianViolations_FindMendelianViolations.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_analysis_directed_CollectHsMetrics.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_vcf_FixVcfHeader.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_CountReads.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_analysis_CollectWgsMetricsWithNonZeroCoverage$WgsMetricsWithNonZeroCoverage.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_vcf_filter_FilterVcf.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_spark_ParallelCopyGCSDirectoryIntoHDFSSpark.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_sam_SamErrorMetric_IndelErrorMetric.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_fasta_ShiftFasta.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_varianteval_VariantEval.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_analysis_GcBiasSummaryMetrics.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_ReferenceBases.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_PrintReadsHeader.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_sam_ViewSam.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_validation_AnnotateVcfWithExpectedAlleleFraction.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_IndexFeatureFile.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$AllowAllReadsReadFilter.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_CountVariants.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_sam_SamErrorMetric_ErrorMetric.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_util_LiftOverIntervalList.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_engine_filters_MetricsReadFilter.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$NotDuplicateReadFilter.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_util_AccumulateQualityYieldMetrics.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_illumina_IlluminaBasecallingMetrics.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_analysis_JumpingLibraryMetrics.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_gnarlyGenotyper_GnarlyGenotyper.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_sam_markduplicates_EstimateLibraryComplexity.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$ValidAlignmentEndReadFilter.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_analysis_FingerprintingDetailMetrics.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_engine_filters_ExcessiveEndClippedReadFilter.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_illumina_BarcodeMetric.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_fasta_FastaReferenceMaker.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_engine_filters_SampleReadFilter.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_spark_pipelines_metrics_QualityScoreDistributionSpark.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_DepthPerSampleHC.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_engine_filters_WellformedReadFilter.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_AlleleFraction.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_analysis_artifacts_ConvertSequencingArtifactToOxoG.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$FirstOfPairReadFilter.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_RMSMappingQuality.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_bqsr_AnalyzeCovariates.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_OrientationBiasReadCounts.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_AssemblyComplexity.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_CountNs.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_vcf_RenameSampleInVcf.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_ReadAnonymizer.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$MappingQualityAvailableReadFilter.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_analysis_CollectOxoGMetrics.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$NotProperlyPairedReadFilter.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_arrays_VerifyIDIntensityContaminationMetrics.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_copynumber_CreateReadCountPanelOfNormals.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_spark_sv_evidence_CalcMetadataSpark.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_spark_pathseq_PathSeqPipelineSpark.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_sam_markduplicates_CollectDuplicateMetrics.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_engine_filters_AlignmentAgreesWithHeaderReadFilter.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$SecondOfPairReadFilter.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_allelespecific_AS_StrandOddsRatio.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_funcotator_Funcotator.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_spark_pipelines_CountReadsSpark.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_spark_pipelines_CountBasesSpark.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_analysis_CollectQualityYieldMetrics.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_analysis_MeanQualityByCycle.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_illumina_quality_CollectHiSeqXPfFailMetrics.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_vcf_GenotypeConcordance.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_spark_pipelines_PrintReadsSpark.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_analysis_replicates_CollectIndependentReplicateMetrics.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_rnaseq_ASEReadCounter.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_ChromosomeCounts.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_dragstr_CalibrateDragstrModel.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_vqsr_FilterVariantTranches.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_engine_filters_LibraryReadFilter.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$NonChimericOriginalAlignmentReadFilter.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_analysis_CollectWgsMetricsWithNonZeroCoverage.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_mutect_CreateSomaticPanelOfNormals.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_qc_CheckPileup.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_arrays_GtcToVcf.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_BwaMemIndexImageCreator.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_sam_CheckTerminatorBlock.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadLengthReadFilter.html  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_engine_filters_OverclippedReadFilter.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/org_broadinstitute_hellbender_tools_walkers_qc_Pileup.json  \n",
            "  inflating: gatk-4.2.6.1/gatkdoc/picard_sam_CleanSam.html  \n",
            "  inflating: gatk-4.2.6.1/gatkPythonPackageArchive.zip  \n",
            "  inflating: gatk-4.2.6.1/gatk-completion.sh  \n",
            "  inflating: gatk-4.2.6.1/gatk-package-4.2.6.1-local.jar  \n",
            "  inflating: gatk-4.2.6.1/gatkcondaenv.yml  \n",
            "  inflating: gatk-4.2.6.1/README.md  \n",
            "  inflating: gatk-4.2.6.1/gatk-package-4.2.6.1-spark.jar  \n",
            "   creating: gatk-4.2.6.1/scripts/\n",
            "   creating: gatk-4.2.6.1/scripts/sv/\n",
            "  inflating: gatk-4.2.6.1/scripts/sv/create_cluster.sh  \n",
            "  inflating: gatk-4.2.6.1/scripts/sv/delete_cluster.sh  \n",
            "  inflating: gatk-4.2.6.1/scripts/sv/sanity_checks.sh  \n",
            "  inflating: gatk-4.2.6.1/scripts/sv/manage_sv_pipeline.sh  \n",
            "  inflating: gatk-4.2.6.1/scripts/sv/run_whole_pipeline.sh  \n",
            "  inflating: gatk-4.2.6.1/scripts/sv/copy_sv_results.sh  \n",
            "  inflating: gatk-4.2.6.1/scripts/sv/default_init.sh  \n",
            "   creating: gatk-4.2.6.1/scripts/sv/stepByStep/\n",
            "  inflating: gatk-4.2.6.1/scripts/sv/stepByStep/scanBam.sh  \n",
            "  inflating: gatk-4.2.6.1/scripts/sv/stepByStep/discoverVariants.sh  \n",
            "  inflating: gatk-4.2.6.1/scripts/sv/stepByStep/svDiscover.sh  \n",
            "   creating: gatk-4.2.6.1/scripts/cnv_wdl/\n",
            "  inflating: gatk-4.2.6.1/scripts/cnv_wdl/cnv_common_tasks.wdl  \n",
            "   creating: gatk-4.2.6.1/scripts/cnv_wdl/germline/\n",
            "  inflating: gatk-4.2.6.1/scripts/cnv_wdl/germline/cnv_germline_cohort_workflow.wdl  \n",
            "  inflating: gatk-4.2.6.1/scripts/cnv_wdl/germline/README.md  \n",
            "  inflating: gatk-4.2.6.1/scripts/cnv_wdl/germline/cnv_germline_case_workflow.wdl  \n",
            "  inflating: gatk-4.2.6.1/scripts/cnv_wdl/germline/Structs.wdl  \n",
            "  inflating: gatk-4.2.6.1/scripts/cnv_wdl/germline/cnv_germline_case_scattered_workflow.wdl  \n",
            "  inflating: gatk-4.2.6.1/scripts/cnv_wdl/germline/joint_call_exome_cnvs.wdl  \n",
            "   creating: gatk-4.2.6.1/scripts/cnv_wdl/somatic/\n",
            "  inflating: gatk-4.2.6.1/scripts/cnv_wdl/somatic/cnv_somatic_pair_workflow.wdl  \n",
            "  inflating: gatk-4.2.6.1/scripts/cnv_wdl/somatic/cnv_somatic_oncotator_workflow.wdl  \n",
            "  inflating: gatk-4.2.6.1/scripts/cnv_wdl/somatic/cnv_somatic_panel_workflow.wdl  \n",
            "  inflating: gatk-4.2.6.1/scripts/cnv_wdl/somatic/README.md  \n",
            "  inflating: gatk-4.2.6.1/scripts/cnv_wdl/somatic/cnv_somatic_funcotate_seg_workflow.wdl  \n",
            "  inflating: gatk-4.2.6.1/scripts/dataproc-cluster-ui  \n",
            "   creating: gatk-4.2.6.1/scripts/mutect2_wdl/\n",
            "  inflating: gatk-4.2.6.1/scripts/mutect2_wdl/mutect2.wdl  \n",
            "   creating: gatk-4.2.6.1/scripts/mutect2_wdl/mutect_resources_json/\n",
            "  inflating: gatk-4.2.6.1/scripts/mutect2_wdl/mutect_resources_json/mutect_resources_process_gnomAD_2.1.json  \n",
            "  inflating: gatk-4.2.6.1/scripts/mutect2_wdl/README.md  \n",
            "  inflating: gatk-4.2.6.1/scripts/mutect2_wdl/mutect2_pon.wdl  \n",
            "  inflating: gatk-4.2.6.1/scripts/mutect2_wdl/mutect2_multi_sample.wdl  \n",
            "  inflating: gatk-4.2.6.1/scripts/mutect2_wdl/mutect_resources.wdl  \n",
            "  inflating: gatk-4.2.6.1/scripts/mutect2_wdl/mutect3_training_data.wdl  \n",
            "  inflating: gatk-4.2.6.1/gatk       \n",
            "  inflating: gatk-4.2.6.1/GATKConfig.EXAMPLE.properties  \n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "#!java -jar picard.jar CreateSequenceDictionary -R /content/drive/MyDrive/WES/hg19_chr8.fa -O /content/drive/MyDrive/WES/hg19_chr8.dict\n",
        "#!java -jar picard.jar CreateSequenceDictionary -R hg19_chr8.fa.gz -O hg19_chr8.fa.gz.dict\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "hq7_cmUEMHnx",
        "outputId": "8d864b96-d589-418b-c865-3607a6b11711"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "08:30:42.447 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/content/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
            "[Tue Jul 26 08:30:42 UTC 2022] CreateSequenceDictionary --OUTPUT hg19_chr8.fa.gz.dict --REFERENCE hg19_chr8.fa.gz --TRUNCATE_NAMES_AT_WHITESPACE true --NUM_SEQUENCES 2147483647 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --GA4GH_CLIENT_SECRETS client_secrets.json --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
            "[Tue Jul 26 08:30:42 UTC 2022] Executing as root@acad52d69833 on Linux 5.4.188+ amd64; OpenJDK 64-Bit Server VM 11.0.15+10-Ubuntu-0ubuntu0.18.04.1; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:2.27.4-SNAPSHOT\n",
            "[Tue Jul 26 08:30:45 UTC 2022] picard.sam.CreateSequenceDictionary done. Elapsed time: 0.05 minutes.\n",
            "Runtime.totalMemory()=620756992\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!gunzip hg19_chr8.fa.gz"
      ],
      "metadata": {
        "id": "WK_zTaqvMzVA"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!samtools faidx hg19_chr8.fa"
      ],
      "metadata": {
        "id": "9i48hKojsvgE"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!samtools dict hg19_chr8.fa>hg19_chr8.dict"
      ],
      "metadata": {
        "id": "sMmH-aZZtTYE"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "**Unzip the index files and rename them to have .idx at the end**"
      ],
      "metadata": {
        "id": "ghdt8FpswlzI"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!gunzip dbsnp_138.hg19.vcf.idx.gz\n",
        "!gunzip Mills_and_1000G_gold_standard.indels.hg19.sites.vcf.idx.gz\n",
        "!gunzip 1000G_omni2.5.hg19.sites.vcf.idx.gz\n",
        "!gunzip 1000G_phase1.indels.hg19.sites.vcf.idx.gz\n",
        "!gunzip 1000G_phase1.snps.high_confidence.hg19.sites.vcf.idx.gz\n",
        "!gunzip hapmap_3.3.hg19.sites.vcf.idx.gz"
      ],
      "metadata": {
        "id": "gEiLmCMYNzPv"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!gunzip hapmap_3.3.hg19.sites.vcf.idx.gz"
      ],
      "metadata": {
        "id": "huuHQjNpV22n"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "#!unzip gatk-4.zip \n",
        "#!java -jar /content/gatk-4/gatk-package-4.2.6.1-local.jar IndexFeatureFile -I dbsnp_138.hg19.vcf"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "MtuCPGuyNnx2",
        "outputId": "fdf602ca-e1e9-4261-a565-53324572daa0"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "09:01:39.480 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/content/gatk-4/gatk-package-4.2.6.1-local.jar!/com/intel/gkl/native/libgkl_compression.so\n",
            "09:01:39.716 INFO  IndexFeatureFile - ------------------------------------------------------------\n",
            "09:01:39.716 INFO  IndexFeatureFile - The Genome Analysis Toolkit (GATK) v4.2.6.1\n",
            "09:01:39.717 INFO  IndexFeatureFile - For support and documentation go to https://software.broadinstitute.org/gatk/\n",
            "09:01:39.717 INFO  IndexFeatureFile - Executing as root@acad52d69833 on Linux v5.4.188+ amd64\n",
            "09:01:39.717 INFO  IndexFeatureFile - Java runtime: OpenJDK 64-Bit Server VM v11.0.15+10-Ubuntu-0ubuntu0.18.04.1\n",
            "09:01:39.717 INFO  IndexFeatureFile - Start Date/Time: July 26, 2022 at 9:01:39 AM UTC\n",
            "09:01:39.718 INFO  IndexFeatureFile - ------------------------------------------------------------\n",
            "09:01:39.718 INFO  IndexFeatureFile - ------------------------------------------------------------\n",
            "09:01:39.720 INFO  IndexFeatureFile - HTSJDK Version: 2.24.1\n",
            "09:01:39.720 INFO  IndexFeatureFile - Picard Version: 2.27.1\n",
            "09:01:39.720 INFO  IndexFeatureFile - Built for Spark Version: 2.4.5\n",
            "09:01:39.720 INFO  IndexFeatureFile - HTSJDK Defaults.COMPRESSION_LEVEL : 2\n",
            "09:01:39.720 INFO  IndexFeatureFile - HTSJDK Defaults.USE_ASYNC_IO_READ_FOR_SAMTOOLS : false\n",
            "09:01:39.720 INFO  IndexFeatureFile - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_SAMTOOLS : true\n",
            "09:01:39.721 INFO  IndexFeatureFile - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_TRIBBLE : false\n",
            "09:01:39.721 INFO  IndexFeatureFile - Deflater: IntelDeflater\n",
            "09:01:39.721 INFO  IndexFeatureFile - Inflater: IntelInflater\n",
            "09:01:39.721 INFO  IndexFeatureFile - GCS max retries/reopens: 20\n",
            "09:01:39.721 INFO  IndexFeatureFile - Requester pays: disabled\n",
            "09:01:39.721 INFO  IndexFeatureFile - Initializing engine\n",
            "09:01:39.721 INFO  IndexFeatureFile - Done initializing engine\n",
            "09:01:39.915 INFO  FeatureManager - Using codec VCFCodec to read file file:///content/Mills_and_1000G_gold_standard.indels.hg19.sites.vcf.gz\n",
            "09:01:39.927 INFO  IndexFeatureFile - Shutting down engine\n",
            "[July 26, 2022 at 9:01:39 AM UTC] org.broadinstitute.hellbender.tools.IndexFeatureFile done. Elapsed time: 0.01 minutes.\n",
            "Runtime.totalMemory()=317718528\n",
            "***********************************************************************\n",
            "\n",
            "A USER ERROR has occurred: Error while trying to create index for Mills_and_1000G_gold_standard.indels.hg19.sites.vcf.gz. Error was: htsjdk.tribble.TribbleException.MalformedFeatureFile: Input file is not in valid block compressed format., for input source: /content/Mills_and_1000G_gold_standard.indels.hg19.sites.vcf.gz\n",
            "\n",
            "***********************************************************************\n",
            "Set the system property GATK_STACKTRACE_ON_USER_EXCEPTION (--java-options '-DGATK_STACKTRACE_ON_USER_EXCEPTION=true') to print the stack trace.\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!java -jar -XX:ParallelGCThreads=2 /content/gatk-4/gatk-package-4.2.6.1-local.jar BaseRecalibrator -I dedup_sorted.bam -R hg19_chr8.fa -O test_markdup_bqsr.report --known-sites dbsnp_138.hg19.vcf.gz --known-sites Mills_and_1000G_gold_standard.indels.hg19.sites.vcf.gz --known-sites 1000G_phase1.indels.hg19.sites.vcf.gz"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "h4rI398CLMAs",
        "outputId": "0350b45f-bdbf-4c17-964e-b9c4ff70acc5"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "09:06:18.011 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/content/gatk-4/gatk-package-4.2.6.1-local.jar!/com/intel/gkl/native/libgkl_compression.so\n",
            "09:06:18.236 INFO  BaseRecalibrator - ------------------------------------------------------------\n",
            "09:06:18.237 INFO  BaseRecalibrator - The Genome Analysis Toolkit (GATK) v4.2.6.1\n",
            "09:06:18.237 INFO  BaseRecalibrator - For support and documentation go to https://software.broadinstitute.org/gatk/\n",
            "09:06:18.237 INFO  BaseRecalibrator - Executing as root@acad52d69833 on Linux v5.4.188+ amd64\n",
            "09:06:18.237 INFO  BaseRecalibrator - Java runtime: OpenJDK 64-Bit Server VM v11.0.15+10-Ubuntu-0ubuntu0.18.04.1\n",
            "09:06:18.238 INFO  BaseRecalibrator - Start Date/Time: July 26, 2022 at 9:06:17 AM UTC\n",
            "09:06:18.238 INFO  BaseRecalibrator - ------------------------------------------------------------\n",
            "09:06:18.238 INFO  BaseRecalibrator - ------------------------------------------------------------\n",
            "09:06:18.239 INFO  BaseRecalibrator - HTSJDK Version: 2.24.1\n",
            "09:06:18.239 INFO  BaseRecalibrator - Picard Version: 2.27.1\n",
            "09:06:18.239 INFO  BaseRecalibrator - Built for Spark Version: 2.4.5\n",
            "09:06:18.240 INFO  BaseRecalibrator - HTSJDK Defaults.COMPRESSION_LEVEL : 2\n",
            "09:06:18.240 INFO  BaseRecalibrator - HTSJDK Defaults.USE_ASYNC_IO_READ_FOR_SAMTOOLS : false\n",
            "09:06:18.240 INFO  BaseRecalibrator - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_SAMTOOLS : true\n",
            "09:06:18.240 INFO  BaseRecalibrator - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_TRIBBLE : false\n",
            "09:06:18.240 INFO  BaseRecalibrator - Deflater: IntelDeflater\n",
            "09:06:18.240 INFO  BaseRecalibrator - Inflater: IntelInflater\n",
            "09:06:18.240 INFO  BaseRecalibrator - GCS max retries/reopens: 20\n",
            "09:06:18.241 INFO  BaseRecalibrator - Requester pays: disabled\n",
            "09:06:18.241 INFO  BaseRecalibrator - Initializing engine\n",
            "09:06:18.511 INFO  FeatureManager - Using codec VCFCodec to read file file:///content/dbsnp_138.hg19.vcf.gz\n",
            "09:06:19.016 INFO  FeatureManager - Using codec VCFCodec to read file file:///content/Mills_and_1000G_gold_standard.indels.hg19.sites.vcf.gz\n",
            "09:06:19.457 INFO  FeatureManager - Using codec VCFCodec to read file file:///content/1000G_phase1.indels.hg19.sites.vcf.gz\n",
            "09:06:19.502 INFO  BaseRecalibrator - Done initializing engine\n",
            "09:06:19.514 INFO  BaseRecalibrationEngine - The covariates being used here: \n",
            "09:06:19.515 INFO  BaseRecalibrationEngine - \tReadGroupCovariate\n",
            "09:06:19.515 INFO  BaseRecalibrationEngine - \tQualityScoreCovariate\n",
            "09:06:19.515 INFO  BaseRecalibrationEngine - \tContextCovariate\n",
            "09:06:19.515 INFO  BaseRecalibrationEngine - \tCycleCovariate\n",
            "09:06:19.518 INFO  ProgressMeter - Starting traversal\n",
            "09:06:19.519 INFO  ProgressMeter -        Current Locus  Elapsed Minutes       Reads Processed     Reads/Minute\n",
            "09:06:29.538 INFO  ProgressMeter -        chr8:11985545              0.2                225000        1347708.9\n",
            "09:06:39.558 INFO  ProgressMeter -        chr8:28574035              0.3                690000        2066074.5\n",
            "09:06:49.574 INFO  ProgressMeter -        chr8:54141236              0.5               1162000        2319824.3\n",
            "09:06:59.575 INFO  ProgressMeter -        chr8:82626334              0.7               1640000        2456560.8\n",
            "09:07:09.580 INFO  ProgressMeter -        chr8:99102059              0.8               1938000        2322766.2\n",
            "09:07:19.580 INFO  ProgressMeter -       chr8:126030145              1.0               2423000        2420539.1\n",
            "09:07:29.052 WARN  IntelInflater - Zero Bytes Written : 0\n",
            "09:07:29.054 INFO  BaseRecalibrator - 5378619 read(s) filtered by: MappingQualityNotZeroReadFilter \n",
            "0 read(s) filtered by: MappingQualityAvailableReadFilter \n",
            "0 read(s) filtered by: MappedReadFilter \n",
            "0 read(s) filtered by: NotSecondaryAlignmentReadFilter \n",
            "1410402 read(s) filtered by: NotDuplicateReadFilter \n",
            "0 read(s) filtered by: PassesVendorQualityCheckReadFilter \n",
            "0 read(s) filtered by: WellformedReadFilter \n",
            "6789021 total reads filtered\n",
            "09:07:29.055 INFO  ProgressMeter -       chr8:146280182              1.2               2877193        2482621.7\n",
            "09:07:29.055 INFO  ProgressMeter - Traversal complete. Processed 2877193 total reads in 1.2 minutes.\n",
            "09:07:29.252 INFO  BaseRecalibrator - Calculating quantized quality scores...\n",
            "09:07:29.274 INFO  BaseRecalibrator - Writing recalibration report...\n",
            "09:07:32.421 INFO  BaseRecalibrator - ...done!\n",
            "09:07:32.422 INFO  BaseRecalibrator - BaseRecalibrator was able to recalibrate 2877193 reads\n",
            "09:07:32.422 INFO  BaseRecalibrator - Shutting down engine\n",
            "[July 26, 2022 at 9:07:32 AM UTC] org.broadinstitute.hellbender.tools.walkers.bqsr.BaseRecalibrator done. Elapsed time: 1.24 minutes.\n",
            "Runtime.totalMemory()=961544192\n",
            "Tool returned:\n",
            "SUCCESS\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "**Apply the trained recalibration algorithm on your bam file**"
      ],
      "metadata": {
        "id": "A7M0qp_axUQw"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!java -jar -XX:ParallelGCThreads=2 /content/gatk-4/gatk-package-4.2.6.1-local.jar ApplyBQSR -I dedup_sorted.bam -R hg19_chr8.fa --bqsr-recal-file test_markdup_bqsr.report -O  final_markdup_bqsr.bam"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "jIyFlr2082VD",
        "outputId": "090f6728-9981-46cf-c52f-0ad96703e341"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "09:11:59.780 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/content/gatk-4/gatk-package-4.2.6.1-local.jar!/com/intel/gkl/native/libgkl_compression.so\n",
            "09:12:00.004 INFO  ApplyBQSR - ------------------------------------------------------------\n",
            "09:12:00.005 INFO  ApplyBQSR - The Genome Analysis Toolkit (GATK) v4.2.6.1\n",
            "09:12:00.005 INFO  ApplyBQSR - For support and documentation go to https://software.broadinstitute.org/gatk/\n",
            "09:12:00.006 INFO  ApplyBQSR - Executing as root@acad52d69833 on Linux v5.4.188+ amd64\n",
            "09:12:00.006 INFO  ApplyBQSR - Java runtime: OpenJDK 64-Bit Server VM v11.0.15+10-Ubuntu-0ubuntu0.18.04.1\n",
            "09:12:00.007 INFO  ApplyBQSR - Start Date/Time: July 26, 2022 at 9:11:59 AM UTC\n",
            "09:12:00.007 INFO  ApplyBQSR - ------------------------------------------------------------\n",
            "09:12:00.007 INFO  ApplyBQSR - ------------------------------------------------------------\n",
            "09:12:00.008 INFO  ApplyBQSR - HTSJDK Version: 2.24.1\n",
            "09:12:00.008 INFO  ApplyBQSR - Picard Version: 2.27.1\n",
            "09:12:00.008 INFO  ApplyBQSR - Built for Spark Version: 2.4.5\n",
            "09:12:00.009 INFO  ApplyBQSR - HTSJDK Defaults.COMPRESSION_LEVEL : 2\n",
            "09:12:00.009 INFO  ApplyBQSR - HTSJDK Defaults.USE_ASYNC_IO_READ_FOR_SAMTOOLS : false\n",
            "09:12:00.009 INFO  ApplyBQSR - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_SAMTOOLS : true\n",
            "09:12:00.009 INFO  ApplyBQSR - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_TRIBBLE : false\n",
            "09:12:00.009 INFO  ApplyBQSR - Deflater: IntelDeflater\n",
            "09:12:00.009 INFO  ApplyBQSR - Inflater: IntelInflater\n",
            "09:12:00.009 INFO  ApplyBQSR - GCS max retries/reopens: 20\n",
            "09:12:00.010 INFO  ApplyBQSR - Requester pays: disabled\n",
            "09:12:00.010 INFO  ApplyBQSR - Initializing engine\n",
            "09:12:00.370 INFO  ApplyBQSR - Done initializing engine\n",
            "09:12:00.395 INFO  ProgressMeter - Starting traversal\n",
            "09:12:00.396 INFO  ProgressMeter -        Current Locus  Elapsed Minutes       Reads Processed     Reads/Minute\n",
            "09:12:10.404 INFO  ProgressMeter -         chr8:7220598              0.2                208000        1247127.0\n",
            "09:12:20.426 INFO  ProgressMeter -        chr8:17074472              0.3                526000        1575636.5\n",
            "09:12:30.454 INFO  ProgressMeter -        chr8:23057021              0.5                859000        1714742.0\n",
            "09:12:40.468 INFO  ProgressMeter -        chr8:30510950              0.7               1186000        1775803.6\n",
            "09:12:50.476 INFO  ProgressMeter -        chr8:39634554              0.8               1495000        1791170.0\n",
            "09:13:00.475 INFO  ProgressMeter -        chr8:53569069              1.0               1794000        1791641.0\n",
            "09:13:10.485 INFO  ProgressMeter -        chr8:67475835              1.2               2117000        1812293.1\n",
            "09:13:20.497 INFO  ProgressMeter -        chr8:79485062              1.3               2436000        1824719.1\n",
            "09:13:30.519 INFO  ProgressMeter -        chr8:86727463              1.5               2697000        1795566.0\n",
            "09:13:40.539 INFO  ProgressMeter -        chr8:86792283              1.7               2876000        1723135.9\n",
            "09:13:50.564 INFO  ProgressMeter -        chr8:91029570              1.8               3109000        1693232.2\n",
            "09:14:00.570 INFO  ProgressMeter -       chr8:100990267              2.0               3432000        1713515.4\n",
            "09:14:10.587 INFO  ProgressMeter -       chr8:113326662              2.2               3754000        1730073.5\n",
            "09:14:20.602 INFO  ProgressMeter -       chr8:125989377              2.3               4077000        1744730.9\n",
            "09:14:30.621 INFO  ProgressMeter -       chr8:143692722              2.5               4398000        1756565.2\n",
            "09:14:40.651 INFO  ProgressMeter -       chr8:145999143              2.7               4718000        1766434.7\n",
            "09:14:42.780 WARN  IntelInflater - Zero Bytes Written : 0\n",
            "09:14:42.784 INFO  ApplyBQSR - 4879783 read(s) filtered by: WellformedReadFilter \n",
            "\n",
            "09:14:42.785 INFO  ProgressMeter -       chr8:146279981              2.7               4786431        1768505.6\n",
            "09:14:42.785 INFO  ProgressMeter - Traversal complete. Processed 4786431 total reads in 2.7 minutes.\n",
            "09:14:42.843 INFO  ApplyBQSR - Shutting down engine\n",
            "[July 26, 2022 at 9:14:42 AM UTC] org.broadinstitute.hellbender.tools.walkers.bqsr.ApplyBQSR done. Elapsed time: 2.72 minutes.\n",
            "Runtime.totalMemory()=384827392\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        " !java -jar -XX:ParallelGCThreads=2 /content/gatk-4/gatk-package-4.2.6.1-local.jar HaplotypeCaller -R hg19_chr8.fa -I final_markdup_bqsr.bam -L twist_hg19_f.bed -A QualByDepth -O raw2.vcf.gz"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "54BR7Bzbkg7p",
        "outputId": "a6c488cd-aabf-4b00-f752-4ec7a33a63eb"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "12:50:15.720 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/content/gatk-4/gatk-package-4.2.6.1-local.jar!/com/intel/gkl/native/libgkl_compression.so\n",
            "12:50:15.919 INFO  HaplotypeCaller - ------------------------------------------------------------\n",
            "12:50:15.919 INFO  HaplotypeCaller - The Genome Analysis Toolkit (GATK) v4.2.6.1\n",
            "12:50:15.919 INFO  HaplotypeCaller - For support and documentation go to https://software.broadinstitute.org/gatk/\n",
            "12:50:15.920 INFO  HaplotypeCaller - Executing as root@acad52d69833 on Linux v5.4.188+ amd64\n",
            "12:50:15.920 INFO  HaplotypeCaller - Java runtime: OpenJDK 64-Bit Server VM v11.0.15+10-Ubuntu-0ubuntu0.18.04.1\n",
            "12:50:15.920 INFO  HaplotypeCaller - Start Date/Time: July 26, 2022 at 12:50:15 PM UTC\n",
            "12:50:15.920 INFO  HaplotypeCaller - ------------------------------------------------------------\n",
            "12:50:15.920 INFO  HaplotypeCaller - ------------------------------------------------------------\n",
            "12:50:15.921 INFO  HaplotypeCaller - HTSJDK Version: 2.24.1\n",
            "12:50:15.921 INFO  HaplotypeCaller - Picard Version: 2.27.1\n",
            "12:50:15.921 INFO  HaplotypeCaller - Built for Spark Version: 2.4.5\n",
            "12:50:15.922 INFO  HaplotypeCaller - HTSJDK Defaults.COMPRESSION_LEVEL : 2\n",
            "12:50:15.922 INFO  HaplotypeCaller - HTSJDK Defaults.USE_ASYNC_IO_READ_FOR_SAMTOOLS : false\n",
            "12:50:15.922 INFO  HaplotypeCaller - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_SAMTOOLS : true\n",
            "12:50:15.922 INFO  HaplotypeCaller - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_TRIBBLE : false\n",
            "12:50:15.922 INFO  HaplotypeCaller - Deflater: IntelDeflater\n",
            "12:50:15.922 INFO  HaplotypeCaller - Inflater: IntelInflater\n",
            "12:50:15.922 INFO  HaplotypeCaller - GCS max retries/reopens: 20\n",
            "12:50:15.922 INFO  HaplotypeCaller - Requester pays: disabled\n",
            "12:50:15.923 INFO  HaplotypeCaller - Initializing engine\n",
            "12:50:16.199 INFO  FeatureManager - Using codec BEDCodec to read file file:///content/twist_hg19_f.bed\n",
            "12:50:16.321 INFO  IntervalArgumentCollection - Processing 1109709 bp from intervals\n",
            "12:50:16.354 INFO  HaplotypeCaller - Done initializing engine\n",
            "12:50:16.363 INFO  HaplotypeCallerEngine - Disabling physical phasing, which is supported only for reference-model confidence output\n",
            "12:50:16.381 INFO  NativeLibraryLoader - Loading libgkl_utils.so from jar:file:/content/gatk-4/gatk-package-4.2.6.1-local.jar!/com/intel/gkl/native/libgkl_utils.so\n",
            "12:50:16.383 INFO  NativeLibraryLoader - Loading libgkl_pairhmm_omp.so from jar:file:/content/gatk-4/gatk-package-4.2.6.1-local.jar!/com/intel/gkl/native/libgkl_pairhmm_omp.so\n",
            "12:50:16.422 INFO  IntelPairHmm - Flush-to-zero (FTZ) is enabled when running PairHMM\n",
            "12:50:16.423 INFO  IntelPairHmm - Available threads: 2\n",
            "12:50:16.423 INFO  IntelPairHmm - Requested threads: 4\n",
            "12:50:16.423 WARN  IntelPairHmm - Using 2 available threads, but 4 were requested\n",
            "12:50:16.423 INFO  PairHMM - Using the OpenMP multi-threaded AVX-accelerated native PairHMM implementation\n",
            "12:50:16.453 INFO  ProgressMeter - Starting traversal\n",
            "12:50:16.454 INFO  ProgressMeter -        Current Locus  Elapsed Minutes     Regions Processed   Regions/Minute\n",
            "12:50:17.839 WARN  InbreedingCoeff - InbreedingCoeff will not be calculated at position chr8:116895 and possibly subsequent; at least 10 samples must have called genotypes\n",
            "12:50:26.510 INFO  ProgressMeter -        chr8:10464718              0.2                   520           3102.9\n",
            "12:50:36.673 INFO  ProgressMeter -        chr8:12943320              0.3                   750           2225.6\n",
            "12:50:46.707 INFO  ProgressMeter -        chr8:27382879              0.5                  1950           3867.5\n",
            "12:50:56.929 INFO  ProgressMeter -        chr8:52733187              0.7                  3350           4966.0\n",
            "12:51:06.951 INFO  ProgressMeter -       chr8:100123326              0.8                  5300           6297.5\n",
            "12:51:16.987 INFO  ProgressMeter -       chr8:141461148              1.0                  7140           7077.2\n",
            "12:51:23.219 INFO  HaplotypeCaller - 21491 read(s) filtered by: MappingQualityReadFilter \n",
            "0 read(s) filtered by: MappingQualityAvailableReadFilter \n",
            "0 read(s) filtered by: MappedReadFilter \n",
            "0 read(s) filtered by: NotSecondaryAlignmentReadFilter \n",
            "821888 read(s) filtered by: NotDuplicateReadFilter \n",
            "0 read(s) filtered by: PassesVendorQualityCheckReadFilter \n",
            "0 read(s) filtered by: NonZeroReferenceLengthAlignmentReadFilter \n",
            "0 read(s) filtered by: GoodCigarReadFilter \n",
            "0 read(s) filtered by: WellformedReadFilter \n",
            "843379 total reads filtered\n",
            "12:51:23.220 INFO  ProgressMeter -       chr8:146171475              1.1                  8366           7518.3\n",
            "12:51:23.220 INFO  ProgressMeter - Traversal complete. Processed 8366 total regions in 1.1 minutes.\n",
            "12:51:23.231 INFO  VectorLoglessPairHMM - Time spent in setup for JNI call : 0.085027824\n",
            "12:51:23.232 INFO  PairHMM - Total compute time in PairHMM computeLogLikelihoods() : 7.369096923000001\n",
            "12:51:23.232 INFO  SmithWatermanAligner - Total compute time in java Smith-Waterman : 4.76 sec\n",
            "12:51:23.234 INFO  HaplotypeCaller - Shutting down engine\n",
            "[July 26, 2022 at 12:51:23 PM UTC] org.broadinstitute.hellbender.tools.walkers.haplotypecaller.HaplotypeCaller done. Elapsed time: 1.13 minutes.\n",
            "Runtime.totalMemory()=543162368\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        " #!java -jar -XX:ParallelGCThreads=2 /content/gatk-4/gatk-package-4.2.6.1-local.jar VariantAnnotator -R hg19_chr8.fa -I final_markdup_bqsr.bam -V  -O  -A QualByDepth"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "lVp_KOOiaL7J",
        "outputId": "29e6907c-5810-4e0d-b91e-7749f2937b14"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "12:47:19.074 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/content/gatk-4/gatk-package-4.2.6.1-local.jar!/com/intel/gkl/native/libgkl_compression.so\n",
            "12:47:19.290 INFO  VariantAnnotator - ------------------------------------------------------------\n",
            "12:47:19.290 INFO  VariantAnnotator - The Genome Analysis Toolkit (GATK) v4.2.6.1\n",
            "12:47:19.290 INFO  VariantAnnotator - For support and documentation go to https://software.broadinstitute.org/gatk/\n",
            "12:47:19.291 INFO  VariantAnnotator - Executing as root@acad52d69833 on Linux v5.4.188+ amd64\n",
            "12:47:19.291 INFO  VariantAnnotator - Java runtime: OpenJDK 64-Bit Server VM v11.0.15+10-Ubuntu-0ubuntu0.18.04.1\n",
            "12:47:19.291 INFO  VariantAnnotator - Start Date/Time: July 26, 2022 at 12:47:18 PM UTC\n",
            "12:47:19.291 INFO  VariantAnnotator - ------------------------------------------------------------\n",
            "12:47:19.291 INFO  VariantAnnotator - ------------------------------------------------------------\n",
            "12:47:19.292 INFO  VariantAnnotator - HTSJDK Version: 2.24.1\n",
            "12:47:19.293 INFO  VariantAnnotator - Picard Version: 2.27.1\n",
            "12:47:19.293 INFO  VariantAnnotator - Built for Spark Version: 2.4.5\n",
            "12:47:19.293 INFO  VariantAnnotator - HTSJDK Defaults.COMPRESSION_LEVEL : 2\n",
            "12:47:19.293 INFO  VariantAnnotator - HTSJDK Defaults.USE_ASYNC_IO_READ_FOR_SAMTOOLS : false\n",
            "12:47:19.293 INFO  VariantAnnotator - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_SAMTOOLS : true\n",
            "12:47:19.293 INFO  VariantAnnotator - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_TRIBBLE : false\n",
            "12:47:19.293 INFO  VariantAnnotator - Deflater: IntelDeflater\n",
            "12:47:19.293 INFO  VariantAnnotator - Inflater: IntelInflater\n",
            "12:47:19.293 INFO  VariantAnnotator - GCS max retries/reopens: 20\n",
            "12:47:19.294 INFO  VariantAnnotator - Requester pays: disabled\n",
            "12:47:19.294 INFO  VariantAnnotator - Initializing engine\n",
            "12:47:19.635 INFO  FeatureManager - Using codec VCFCodec to read file file:///content/hapmap_3.3.hg19.sites.vcf.gz\n",
            "12:47:19.767 INFO  VariantAnnotator - Done initializing engine\n",
            "12:47:19.770 INFO  VariantAnnotator - Shutting down engine\n",
            "[July 26, 2022 at 12:47:19 PM UTC] org.broadinstitute.hellbender.tools.walkers.annotator.VariantAnnotator done. Elapsed time: 0.01 minutes.\n",
            "Runtime.totalMemory()=349175808\n",
            "***********************************************************************\n",
            "\n",
            "A USER ERROR has occurred: Reads sample 'TEST' from readgroups tags does not match any sample in the variant genotypes\n",
            "\n",
            "***********************************************************************\n",
            "Set the system property GATK_STACKTRACE_ON_USER_EXCEPTION (--java-options '-DGATK_STACKTRACE_ON_USER_EXCEPTION=true') to print the stack trace.\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!java -jar picard.jar SplitVcfs -I raw2.vcf.gz --SNP_OUTPUT raw_snp.vcf --INDEL_OUTPUT raw_indel.vcf"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "UWoUunu_mDQK",
        "outputId": "a483068d-1611-4e58-ebeb-db42297058be"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "13:02:21.631 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/content/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
            "[Tue Jul 26 13:02:21 UTC 2022] SplitVcfs --INPUT raw2.vcf.gz --SNP_OUTPUT raw_snp.vcf --INDEL_OUTPUT raw_indel.vcf --STRICT true --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX true --CREATE_MD5_FILE false --GA4GH_CLIENT_SECRETS client_secrets.json --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
            "[Tue Jul 26 13:02:21 UTC 2022] Executing as root@acad52d69833 on Linux 5.4.188+ amd64; OpenJDK 64-Bit Server VM 11.0.15+10-Ubuntu-0ubuntu0.18.04.1; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:2.27.4-SNAPSHOT\n",
            "13:02:21.749 WARN  IntelInflater - Zero Bytes Written : 0\n",
            "13:02:21.840 WARN  IntelInflater - Zero Bytes Written : 0\n",
            "[Tue Jul 26 13:02:22 UTC 2022] picard.vcf.SplitVcfs done. Elapsed time: 0.01 minutes.\n",
            "Runtime.totalMemory()=216006656\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "**Hard filtering of variants**"
      ],
      "metadata": {
        "id": "uctErMq_khkr"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!java -jar -XX:ParallelGCThreads=2 /content/gatk-4/gatk-package-4.2.6.1-local.jar VariantFiltration -V raw_snp.vcf -filter \"QD<2.0\" --filter-name \"QD2\" -filter \"QUAL<30\" --filter-name \"QUAL30\" -filter \"DP<20\" --filter-name \"Depth20\" -O final_snp.vcf\n",
        "!java -jar -XX:ParallelGCThreads=2 /content/gatk-4/gatk-package-4.2.6.1-local.jar VariantFiltration -V raw_indel.vcf -filter \"QD<2.0\" --filter-name \"QD2\" -filter \"QUAL<30\" --filter-name \"QUAL30\" -filter \"DP<20\" --filter-name \"Depth20\" -O final_indel.vcf"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "n79CpLc3kg3Y",
        "outputId": "dda71cb6-a33c-4b14-9385-be4d60f975ec"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "13:08:42.503 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/content/gatk-4/gatk-package-4.2.6.1-local.jar!/com/intel/gkl/native/libgkl_compression.so\n",
            "13:08:42.778 INFO  VariantFiltration - ------------------------------------------------------------\n",
            "13:08:42.778 INFO  VariantFiltration - The Genome Analysis Toolkit (GATK) v4.2.6.1\n",
            "13:08:42.778 INFO  VariantFiltration - For support and documentation go to https://software.broadinstitute.org/gatk/\n",
            "13:08:42.779 INFO  VariantFiltration - Executing as root@acad52d69833 on Linux v5.4.188+ amd64\n",
            "13:08:42.779 INFO  VariantFiltration - Java runtime: OpenJDK 64-Bit Server VM v11.0.15+10-Ubuntu-0ubuntu0.18.04.1\n",
            "13:08:42.779 INFO  VariantFiltration - Start Date/Time: July 26, 2022 at 1:08:42 PM UTC\n",
            "13:08:42.779 INFO  VariantFiltration - ------------------------------------------------------------\n",
            "13:08:42.780 INFO  VariantFiltration - ------------------------------------------------------------\n",
            "13:08:42.780 INFO  VariantFiltration - HTSJDK Version: 2.24.1\n",
            "13:08:42.781 INFO  VariantFiltration - Picard Version: 2.27.1\n",
            "13:08:42.781 INFO  VariantFiltration - Built for Spark Version: 2.4.5\n",
            "13:08:42.781 INFO  VariantFiltration - HTSJDK Defaults.COMPRESSION_LEVEL : 2\n",
            "13:08:42.781 INFO  VariantFiltration - HTSJDK Defaults.USE_ASYNC_IO_READ_FOR_SAMTOOLS : false\n",
            "13:08:42.781 INFO  VariantFiltration - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_SAMTOOLS : true\n",
            "13:08:42.781 INFO  VariantFiltration - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_TRIBBLE : false\n",
            "13:08:42.781 INFO  VariantFiltration - Deflater: IntelDeflater\n",
            "13:08:42.781 INFO  VariantFiltration - Inflater: IntelInflater\n",
            "13:08:42.781 INFO  VariantFiltration - GCS max retries/reopens: 20\n",
            "13:08:42.782 INFO  VariantFiltration - Requester pays: disabled\n",
            "13:08:42.782 INFO  VariantFiltration - Initializing engine\n",
            "13:08:42.943 INFO  FeatureManager - Using codec VCFCodec to read file file:///content/raw_snp.vcf\n",
            "13:08:42.982 INFO  VariantFiltration - Done initializing engine\n",
            "13:08:43.056 INFO  ProgressMeter - Starting traversal\n",
            "13:08:43.056 INFO  ProgressMeter -        Current Locus  Elapsed Minutes    Variants Processed  Variants/Minute\n",
            "13:08:43.412 INFO  ProgressMeter -             unmapped              0.0                   682         115267.6\n",
            "13:08:43.414 INFO  ProgressMeter - Traversal complete. Processed 682 total variants in 0.0 minutes.\n",
            "13:08:43.517 INFO  VariantFiltration - Shutting down engine\n",
            "[July 26, 2022 at 1:08:43 PM UTC] org.broadinstitute.hellbender.tools.walkers.filters.VariantFiltration done. Elapsed time: 0.02 minutes.\n",
            "Runtime.totalMemory()=291504128\n",
            "13:08:47.044 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/content/gatk-4/gatk-package-4.2.6.1-local.jar!/com/intel/gkl/native/libgkl_compression.so\n",
            "13:08:47.258 INFO  VariantFiltration - ------------------------------------------------------------\n",
            "13:08:47.259 INFO  VariantFiltration - The Genome Analysis Toolkit (GATK) v4.2.6.1\n",
            "13:08:47.259 INFO  VariantFiltration - For support and documentation go to https://software.broadinstitute.org/gatk/\n",
            "13:08:47.259 INFO  VariantFiltration - Executing as root@acad52d69833 on Linux v5.4.188+ amd64\n",
            "13:08:47.259 INFO  VariantFiltration - Java runtime: OpenJDK 64-Bit Server VM v11.0.15+10-Ubuntu-0ubuntu0.18.04.1\n",
            "13:08:47.260 INFO  VariantFiltration - Start Date/Time: July 26, 2022 at 1:08:46 PM UTC\n",
            "13:08:47.260 INFO  VariantFiltration - ------------------------------------------------------------\n",
            "13:08:47.260 INFO  VariantFiltration - ------------------------------------------------------------\n",
            "13:08:47.262 INFO  VariantFiltration - HTSJDK Version: 2.24.1\n",
            "13:08:47.262 INFO  VariantFiltration - Picard Version: 2.27.1\n",
            "13:08:47.262 INFO  VariantFiltration - Built for Spark Version: 2.4.5\n",
            "13:08:47.262 INFO  VariantFiltration - HTSJDK Defaults.COMPRESSION_LEVEL : 2\n",
            "13:08:47.262 INFO  VariantFiltration - HTSJDK Defaults.USE_ASYNC_IO_READ_FOR_SAMTOOLS : false\n",
            "13:08:47.262 INFO  VariantFiltration - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_SAMTOOLS : true\n",
            "13:08:47.263 INFO  VariantFiltration - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_TRIBBLE : false\n",
            "13:08:47.263 INFO  VariantFiltration - Deflater: IntelDeflater\n",
            "13:08:47.263 INFO  VariantFiltration - Inflater: IntelInflater\n",
            "13:08:47.263 INFO  VariantFiltration - GCS max retries/reopens: 20\n",
            "13:08:47.264 INFO  VariantFiltration - Requester pays: disabled\n",
            "13:08:47.264 INFO  VariantFiltration - Initializing engine\n",
            "13:08:47.430 INFO  FeatureManager - Using codec VCFCodec to read file file:///content/raw_indel.vcf\n",
            "13:08:47.459 INFO  VariantFiltration - Done initializing engine\n",
            "13:08:47.557 INFO  ProgressMeter - Starting traversal\n",
            "13:08:47.560 INFO  ProgressMeter -        Current Locus  Elapsed Minutes    Variants Processed  Variants/Minute\n",
            "13:08:47.626 INFO  ProgressMeter -             unmapped              0.0                    13          11818.2\n",
            "13:08:47.627 INFO  ProgressMeter - Traversal complete. Processed 13 total variants in 0.0 minutes.\n",
            "13:08:47.696 INFO  VariantFiltration - Shutting down engine\n",
            "[July 26, 2022 at 1:08:47 PM UTC] org.broadinstitute.hellbender.tools.walkers.filters.VariantFiltration done. Elapsed time: 0.01 minutes.\n",
            "Runtime.totalMemory()=284164096\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "**Soft Filtering (Use hg38)**"
      ],
      "metadata": {
        "id": "OKTaEcr7oAYM"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!java -jar -XX:ParallelGCThreads=2 /content/gatk-4/gatk-package-4.2.6.1-local.jar VariantRecalibrator -R hg19_chr8.fa -V raw.vcf.gz --resource:hapmap,known=false,training=true,truth=true,prior=15.0 hapmap_3.3.hg19.sites.vcf.gz --resource:omni,known=false,training=true,truth=false,prior=12.0 1000G_omni2.5.hg19.sites.vcf.gz --resource:1000G,known=false,training=true,truth=false,prior=10.0 1000G_phase1.snps.high_confidence.hg19.sites.vcf.gz --resource:dbsnp,known=true,training=false,truth=false,prior=7.0 dbsnp_138.hg19.vcf.gz -an QD -an MQ -an MQRankSum -an ReadPosRankSum -an FS -an SOR -mode BOTH -O snpoutrecal --tranches-file snpout.tranches --rscript-file snpout.plotsR"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "ZaiqE8PX1xXL",
        "outputId": "b69fab22-84b0-4b7a-e411-4a86131db3ae"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "12:44:51.705 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/content/gatk-4/gatk-package-4.2.6.1-local.jar!/com/intel/gkl/native/libgkl_compression.so\n",
            "12:44:51.979 INFO  VariantRecalibrator - ------------------------------------------------------------\n",
            "12:44:51.979 INFO  VariantRecalibrator - The Genome Analysis Toolkit (GATK) v4.2.6.1\n",
            "12:44:51.980 INFO  VariantRecalibrator - For support and documentation go to https://software.broadinstitute.org/gatk/\n",
            "12:44:51.980 INFO  VariantRecalibrator - Executing as root@acad52d69833 on Linux v5.4.188+ amd64\n",
            "12:44:51.980 INFO  VariantRecalibrator - Java runtime: OpenJDK 64-Bit Server VM v11.0.15+10-Ubuntu-0ubuntu0.18.04.1\n",
            "12:44:51.980 INFO  VariantRecalibrator - Start Date/Time: July 26, 2022 at 12:44:51 PM UTC\n",
            "12:44:51.980 INFO  VariantRecalibrator - ------------------------------------------------------------\n",
            "12:44:51.981 INFO  VariantRecalibrator - ------------------------------------------------------------\n",
            "12:44:51.983 INFO  VariantRecalibrator - HTSJDK Version: 2.24.1\n",
            "12:44:51.983 INFO  VariantRecalibrator - Picard Version: 2.27.1\n",
            "12:44:51.983 INFO  VariantRecalibrator - Built for Spark Version: 2.4.5\n",
            "12:44:51.984 INFO  VariantRecalibrator - HTSJDK Defaults.COMPRESSION_LEVEL : 2\n",
            "12:44:51.984 INFO  VariantRecalibrator - HTSJDK Defaults.USE_ASYNC_IO_READ_FOR_SAMTOOLS : false\n",
            "12:44:51.984 INFO  VariantRecalibrator - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_SAMTOOLS : true\n",
            "12:44:51.985 INFO  VariantRecalibrator - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_TRIBBLE : false\n",
            "12:44:51.985 INFO  VariantRecalibrator - Deflater: IntelDeflater\n",
            "12:44:51.985 INFO  VariantRecalibrator - Inflater: IntelInflater\n",
            "12:44:51.986 INFO  VariantRecalibrator - GCS max retries/reopens: 20\n",
            "12:44:51.986 INFO  VariantRecalibrator - Requester pays: disabled\n",
            "12:44:51.986 INFO  VariantRecalibrator - Initializing engine\n",
            "12:44:52.447 INFO  FeatureManager - Using codec VCFCodec to read file file:///content/hapmap_3.3.hg19.sites.vcf.gz\n",
            "12:44:52.531 INFO  FeatureManager - Using codec VCFCodec to read file file:///content/1000G_omni2.5.hg19.sites.vcf.gz\n",
            "12:44:53.154 INFO  FeatureManager - Using codec VCFCodec to read file file:///content/1000G_phase1.snps.high_confidence.hg19.sites.vcf.gz\n",
            "12:44:53.243 INFO  FeatureManager - Using codec VCFCodec to read file file:///content/dbsnp_138.hg19.vcf.gz\n",
            "12:44:53.439 INFO  FeatureManager - Using codec VCFCodec to read file file:///content/raw.vcf.gz\n",
            "12:44:53.451 WARN  IntelInflater - Zero Bytes Written : 0\n",
            "12:44:53.544 WARN  IntelInflater - Zero Bytes Written : 0\n",
            "12:44:53.573 INFO  VariantRecalibrator - Done initializing engine\n",
            "12:44:53.576 INFO  TrainingSet - Found hapmap track: \tKnown = false \tTraining = true \tTruth = true \tPrior = Q15.0\n",
            "12:44:53.577 INFO  TrainingSet - Found omni track: \tKnown = false \tTraining = true \tTruth = false \tPrior = Q12.0\n",
            "12:44:53.577 INFO  TrainingSet - Found 1000G track: \tKnown = false \tTraining = true \tTruth = false \tPrior = Q10.0\n",
            "12:44:53.577 INFO  TrainingSet - Found dbsnp track: \tKnown = true \tTraining = false \tTruth = false \tPrior = Q7.0\n",
            "12:44:53.583 WARN  GATKVariantContextUtils - Can't determine output variant file format from output file extension \"\". Defaulting to VCF.\n",
            "12:44:53.658 INFO  ProgressMeter - Starting traversal\n",
            "12:44:53.659 INFO  ProgressMeter -        Current Locus  Elapsed Minutes    Variants Processed  Variants/Minute\n",
            "12:44:53.663 WARN  IntelInflater - Zero Bytes Written : 0\n",
            "12:44:54.035 INFO  ProgressMeter -             unmapped              0.0                   695         110904.3\n",
            "12:44:54.036 INFO  ProgressMeter - Traversal complete. Processed 695 total variants in 0.0 minutes.\n",
            "12:44:54.038 INFO  VariantRecalibrator - Shutting down engine\n",
            "[July 26, 2022 at 12:44:54 PM UTC] org.broadinstitute.hellbender.tools.walkers.vqsr.VariantRecalibrator done. Elapsed time: 0.04 minutes.\n",
            "Runtime.totalMemory()=1635778560\n",
            "***********************************************************************\n",
            "\n",
            "A USER ERROR has occurred: Bad input: Values for QD annotation not detected for ANY training variant in the input callset. VariantAnnotator may be used to add these annotations.\n",
            "\n",
            "***********************************************************************\n",
            "Set the system property GATK_STACKTRACE_ON_USER_EXCEPTION (--java-options '-DGATK_STACKTRACE_ON_USER_EXCEPTION=true') to print the stack trace.\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "**Generate your own BED file**"
      ],
      "metadata": {
        "id": "O_Q54VaWoH8s"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "#!sudo apt install bedtools\n",
        "\n",
        "#!sortBed -i /content/drive/MyDrive/Twist_Exome_Core_Covered_Targets_hg19_liftover.bed>/content/twist_hg19.bed\n",
        "!awk -F\"\\t\" '/chr8/' twist_hg19.bed > twist_hg19_f.bed"
      ],
      "metadata": {
        "id": "Oo1LNRrANVHx"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "**Install Conda to use VEP**"
      ],
      "metadata": {
        "id": "aou5W36soN-S"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!conda --version\n",
        "!pip install -q condacolab\n",
        "import condacolab\n",
        "condacolab.install()"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "TMZmgZrk4o1p",
        "outputId": "fc2a603c-5a1c-4e41-d172-329e35714cef"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "/bin/bash: conda: command not found\n",
            "⏬ Downloading https://github.com/jaimergp/miniforge/releases/latest/download/Mambaforge-colab-Linux-x86_64.sh...\n",
            "📦 Installing...\n",
            "📌 Adjusting configuration...\n",
            "🩹 Patching environment...\n",
            "⏲ Done in 0:00:31\n",
            "🔁 Restarting kernel...\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!conda --version"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "f8tDACIL5XeT",
        "outputId": "4e8304ea-22e0-4c48-9cc9-e070480d0129"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "conda 4.9.2\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!conda update -n base conda"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "krkxISJm-12M",
        "outputId": "3408a17b-2c18-4dac-9941-20c1b4e2a0a0"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Collecting package metadata (current_repodata.json): - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\bdone\n",
            "Solving environment: \\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\bdone\n",
            "\n",
            "## Package Plan ##\n",
            "\n",
            "  environment location: /usr/local\n",
            "\n",
            "  added / updated specs:\n",
            "    - conda\n",
            "\n",
            "\n",
            "The following packages will be downloaded:\n",
            "\n",
            "    package                    |            build\n",
            "    ---------------------------|-----------------\n",
            "    _openmp_mutex-4.5          |            2_gnu          23 KB  conda-forge\n",
            "    brotlipy-0.7.0             |py37h540881e_1004         342 KB  conda-forge\n",
            "    c-ares-1.18.1              |       h7f98852_0         113 KB  conda-forge\n",
            "    ca-certificates-2022.6.15  |       ha878542_0         149 KB  conda-forge\n",
            "    certifi-2022.6.15          |   py37h89c1867_0         155 KB  conda-forge\n",
            "    cffi-1.15.1                |   py37h43b0acd_0         227 KB  conda-forge\n",
            "    charset-normalizer-2.1.0   |     pyhd8ed1ab_0          35 KB  conda-forge\n",
            "    colorama-0.4.5             |     pyhd8ed1ab_0          18 KB  conda-forge\n",
            "    conda-4.13.0               |   py37h89c1867_1         989 KB  conda-forge\n",
            "    conda-package-handling-1.8.1|   py37h540881e_1         1.0 MB  conda-forge\n",
            "    cryptography-37.0.4        |   py37h38fbfac_0         1.5 MB  conda-forge\n",
            "    icu-70.1                   |       h27087fc_0        13.5 MB  conda-forge\n",
            "    idna-3.3                   |     pyhd8ed1ab_0          55 KB  conda-forge\n",
            "    keyutils-1.6.1             |       h166bdaf_0         115 KB  conda-forge\n",
            "    krb5-1.19.3                |       h3790be6_0         1.4 MB  conda-forge\n",
            "    ld_impl_linux-64-2.36.1    |       hea4e1c9_2         667 KB  conda-forge\n",
            "    libarchive-3.5.2           |       hb890918_3         1.6 MB  conda-forge\n",
            "    libcurl-7.83.1             |       h7bff187_0         342 KB  conda-forge\n",
            "    libffi-3.4.2               |       h7f98852_5          57 KB  conda-forge\n",
            "    libgcc-ng-12.1.0           |      h8d9b700_16         940 KB  conda-forge\n",
            "    libgomp-12.1.0             |      h8d9b700_16         459 KB  conda-forge\n",
            "    libnghttp2-1.47.0          |       h727a467_0         808 KB  conda-forge\n",
            "    libnsl-2.0.0               |       h7f98852_0          31 KB  conda-forge\n",
            "    libsolv-0.7.22             |       h6239696_0         443 KB  conda-forge\n",
            "    libssh2-1.10.0             |       ha56f1ee_2         233 KB  conda-forge\n",
            "    libstdcxx-ng-12.1.0        |      ha89aaad_16         4.3 MB  conda-forge\n",
            "    libxml2-2.9.14             |       h22db469_3         781 KB  conda-forge\n",
            "    libzlib-1.2.12             |       h166bdaf_2          63 KB  conda-forge\n",
            "    lz4-c-1.9.3                |       h9c3ff4c_1         179 KB  conda-forge\n",
            "    ncurses-6.3                |       h27087fc_1        1002 KB  conda-forge\n",
            "    openssl-1.1.1q             |       h166bdaf_0         2.1 MB  conda-forge\n",
            "    pycosat-0.6.3              |py37h540881e_1010         107 KB  conda-forge\n",
            "    pycparser-2.21             |     pyhd8ed1ab_0         100 KB  conda-forge\n",
            "    pyopenssl-22.0.0           |     pyhd8ed1ab_0          49 KB  conda-forge\n",
            "    pysocks-1.7.1              |   py37h89c1867_5          28 KB  conda-forge\n",
            "    python-3.7.12              |hb7a2778_100_cpython        57.3 MB  conda-forge\n",
            "    python_abi-3.7             |          2_cp37m           4 KB  conda-forge\n",
            "    readline-8.1.2             |       h0f457ee_0         291 KB  conda-forge\n",
            "    reproc-14.2.3              |       h7f98852_0          28 KB  conda-forge\n",
            "    reproc-cpp-14.2.3          |       h9c3ff4c_0          20 KB  conda-forge\n",
            "    requests-2.28.1            |     pyhd8ed1ab_0          53 KB  conda-forge\n",
            "    ruamel_yaml-0.15.80        |py37h540881e_1007         259 KB  conda-forge\n",
            "    setuptools-63.2.0          |   py37h89c1867_0         1.4 MB  conda-forge\n",
            "    six-1.16.0                 |     pyh6c4a22f_0          14 KB  conda-forge\n",
            "    sqlite-3.39.2              |       h4ff8645_0         1.5 MB  conda-forge\n",
            "    tk-8.6.12                  |       h27826a3_0         3.3 MB  conda-forge\n",
            "    tqdm-4.64.0                |     pyhd8ed1ab_0          81 KB  conda-forge\n",
            "    urllib3-1.26.11            |     pyhd8ed1ab_0         102 KB  conda-forge\n",
            "    wheel-0.37.1               |     pyhd8ed1ab_0          31 KB  conda-forge\n",
            "    yaml-0.2.5                 |       h7f98852_2          87 KB  conda-forge\n",
            "    zlib-1.2.12                |       h166bdaf_2          91 KB  conda-forge\n",
            "    zstd-1.5.2                 |       h8a70e8d_2         448 KB  conda-forge\n",
            "    ------------------------------------------------------------\n",
            "                                           Total:        98.9 MB\n",
            "\n",
            "The following NEW packages will be INSTALLED:\n",
            "\n",
            "  charset-normalizer conda-forge/noarch::charset-normalizer-2.1.0-pyhd8ed1ab_0\n",
            "  colorama           conda-forge/noarch::colorama-0.4.5-pyhd8ed1ab_0\n",
            "  keyutils           conda-forge/linux-64::keyutils-1.6.1-h166bdaf_0\n",
            "  libnsl             conda-forge/linux-64::libnsl-2.0.0-h7f98852_0\n",
            "  libzlib            conda-forge/linux-64::libzlib-1.2.12-h166bdaf_2\n",
            "\n",
            "The following packages will be REMOVED:\n",
            "\n",
            "  chardet-4.0.0-py37h89c1867_1\n",
            "\n",
            "The following packages will be UPDATED:\n",
            "\n",
            "  _openmp_mutex                                   4.5-1_gnu --> 4.5-2_gnu\n",
            "  brotlipy                          0.7.0-py37h5e8e339_1001 --> 0.7.0-py37h540881e_1004\n",
            "  c-ares                                  1.17.1-h7f98852_1 --> 1.18.1-h7f98852_0\n",
            "  ca-certificates                      2020.12.5-ha878542_0 --> 2022.6.15-ha878542_0\n",
            "  certifi                          2020.12.5-py37h89c1867_1 --> 2022.6.15-py37h89c1867_0\n",
            "  cffi                                1.14.5-py37hc58025e_0 --> 1.15.1-py37h43b0acd_0\n",
            "  conda                                4.9.2-py37h89c1867_0 --> 4.13.0-py37h89c1867_1\n",
            "  conda-package-han~                   1.7.2-py37hb5d75c8_0 --> 1.8.1-py37h540881e_1\n",
            "  cryptography                         3.4.5-py37h5d9358c_1 --> 37.0.4-py37h38fbfac_0\n",
            "  icu                                       68.1-h58526e2_0 --> 70.1-h27087fc_0\n",
            "  idna                                    2.10-pyh9f0ad1d_0 --> 3.3-pyhd8ed1ab_0\n",
            "  krb5                                    1.17.2-h926e7f8_0 --> 1.19.3-h3790be6_0\n",
            "  ld_impl_linux-64                        2.35.1-hea4e1c9_2 --> 2.36.1-hea4e1c9_2\n",
            "  libarchive                               3.5.1-h3f442fb_1 --> 3.5.2-hb890918_3\n",
            "  libcurl                                 7.75.0-hc4aaa36_0 --> 7.83.1-h7bff187_0\n",
            "  libffi                                     3.3-h58526e2_2 --> 3.4.2-h7f98852_5\n",
            "  libgcc-ng                               9.3.0-h2828fa1_18 --> 12.1.0-h8d9b700_16\n",
            "  libgomp                                 9.3.0-h2828fa1_18 --> 12.1.0-h8d9b700_16\n",
            "  libnghttp2                              1.43.0-h812cca2_0 --> 1.47.0-h727a467_0\n",
            "  libsolv                                 0.7.17-h780b84a_0 --> 0.7.22-h6239696_0\n",
            "  libssh2                                  1.9.0-ha56f1ee_6 --> 1.10.0-ha56f1ee_2\n",
            "  libstdcxx-ng                            9.3.0-h6de172a_18 --> 12.1.0-ha89aaad_16\n",
            "  libxml2                                 2.9.10-h72842e0_3 --> 2.9.14-h22db469_3\n",
            "  lz4-c                                    1.9.3-h9c3ff4c_0 --> 1.9.3-h9c3ff4c_1\n",
            "  ncurses                                    6.2-h58526e2_4 --> 6.3-h27087fc_1\n",
            "  openssl                                 1.1.1j-h7f98852_0 --> 1.1.1q-h166bdaf_0\n",
            "  pycosat                           0.6.3-py37h5e8e339_1006 --> 0.6.3-py37h540881e_1010\n",
            "  pycparser                               2.20-pyh9f0ad1d_2 --> 2.21-pyhd8ed1ab_0\n",
            "  pyopenssl                             20.0.1-pyhd8ed1ab_0 --> 22.0.0-pyhd8ed1ab_0\n",
            "  pysocks                              1.7.1-py37h89c1867_3 --> 1.7.1-py37h89c1867_5\n",
            "  python                        3.7.10-hffdb5ce_100_cpython --> 3.7.12-hb7a2778_100_cpython\n",
            "  python_abi                                    3.7-1_cp37m --> 3.7-2_cp37m\n",
            "  readline                                   8.0-he28a2e2_2 --> 8.1.2-h0f457ee_0\n",
            "  reproc                                  14.2.1-h36c2ea0_0 --> 14.2.3-h7f98852_0\n",
            "  reproc-cpp                              14.2.1-h58526e2_0 --> 14.2.3-h9c3ff4c_0\n",
            "  requests                              2.25.1-pyhd3deb0d_0 --> 2.28.1-pyhd8ed1ab_0\n",
            "  ruamel_yaml                     0.15.80-py37h5e8e339_1004 --> 0.15.80-py37h540881e_1007\n",
            "  setuptools                          49.6.0-py37h89c1867_3 --> 63.2.0-py37h89c1867_0\n",
            "  six                                   1.15.0-pyh9f0ad1d_0 --> 1.16.0-pyh6c4a22f_0\n",
            "  sqlite                                  3.34.0-h74cdb3f_0 --> 3.39.2-h4ff8645_0\n",
            "  tk                                      8.6.10-h21135ba_1 --> 8.6.12-h27826a3_0\n",
            "  tqdm                                  4.59.0-pyhd8ed1ab_0 --> 4.64.0-pyhd8ed1ab_0\n",
            "  urllib3                               1.26.3-pyhd8ed1ab_0 --> 1.26.11-pyhd8ed1ab_0\n",
            "  wheel                                 0.36.2-pyhd3deb0d_0 --> 0.37.1-pyhd8ed1ab_0\n",
            "  yaml                                     0.2.5-h516909a_0 --> 0.2.5-h7f98852_2\n",
            "  zlib                                 1.2.11-h516909a_1010 --> 1.2.12-h166bdaf_2\n",
            "  zstd                                     1.4.9-ha95c52a_0 --> 1.5.2-h8a70e8d_2\n",
            "\n",
            "\n",
            "\n",
            "Downloading and Extracting Packages\n",
            "lz4-c-1.9.3          | 179 KB    | : 100% 1.0/1 [00:00<00:00,  4.90it/s]                \n",
            "_openmp_mutex-4.5    | 23 KB     | : 100% 1.0/1 [00:00<00:00, 15.52it/s]\n",
            "ca-certificates-2022 | 149 KB    | : 100% 1.0/1 [00:00<00:00, 12.23it/s]\n",
            "readline-8.1.2       | 291 KB    | : 100% 1.0/1 [00:00<00:00,  7.98it/s]\n",
            "pycosat-0.6.3        | 107 KB    | : 100% 1.0/1 [00:00<00:00, 13.26it/s]\n",
            "ld_impl_linux-64-2.3 | 667 KB    | : 100% 1.0/1 [00:00<00:00,  4.30it/s]\n",
            "wheel-0.37.1         | 31 KB     | : 100% 1.0/1 [00:00<00:00, 15.57it/s]\n",
            "brotlipy-0.7.0       | 342 KB    | : 100% 1.0/1 [00:00<00:00,  8.71it/s]\n",
            "conda-package-handli | 1.0 MB    | : 100% 1.0/1 [00:00<00:00,  3.76it/s]\n",
            "setuptools-63.2.0    | 1.4 MB    | : 100% 1.0/1 [00:00<00:00,  1.71it/s]\n",
            "tqdm-4.64.0          | 81 KB     | : 100% 1.0/1 [00:00<00:00, 14.27it/s]\n",
            "python_abi-3.7       | 4 KB      | : 100% 1.0/1 [00:00<00:00, 19.16it/s]\n",
            "colorama-0.4.5       | 18 KB     | : 100% 1.0/1 [00:00<00:00, 19.36it/s]\n",
            "libzlib-1.2.12       | 63 KB     | : 100% 1.0/1 [00:00<00:00, 14.75it/s]\n",
            "sqlite-3.39.2        | 1.5 MB    | : 100% 1.0/1 [00:00<00:00,  3.14it/s]\n",
            "zlib-1.2.12          | 91 KB     | : 100% 1.0/1 [00:00<00:00, 13.88it/s]\n",
            "libnghttp2-1.47.0    | 808 KB    | : 100% 1.0/1 [00:00<00:00,  3.61it/s]\n",
            "certifi-2022.6.15    | 155 KB    | : 100% 1.0/1 [00:00<00:00, 12.58it/s]\n",
            "libgomp-12.1.0       | 459 KB    | : 100% 1.0/1 [00:00<00:00,  6.66it/s]\n",
            "reproc-cpp-14.2.3    | 20 KB     | : 100% 1.0/1 [00:00<00:00, 13.44it/s]\n",
            "krb5-1.19.3          | 1.4 MB    | : 100% 1.0/1 [00:00<00:00,  2.40it/s]\n",
            "cryptography-37.0.4  | 1.5 MB    | : 100% 1.0/1 [00:00<00:00,  1.88it/s]\n",
            "libarchive-3.5.2     | 1.6 MB    | : 100% 1.0/1 [00:00<00:00,  2.79it/s]\n",
            "libxml2-2.9.14       | 781 KB    | : 100% 1.0/1 [00:00<00:00,  4.88it/s]\n",
            "ncurses-6.3          | 1002 KB   | : 100% 1.0/1 [00:00<00:00,  1.71it/s]\n",
            "libstdcxx-ng-12.1.0  | 4.3 MB    | : 100% 1.0/1 [00:00<00:00,  1.03it/s]\n",
            "yaml-0.2.5           | 87 KB     | : 100% 1.0/1 [00:00<00:00, 13.48it/s]\n",
            "ruamel_yaml-0.15.80  | 259 KB    | : 100% 1.0/1 [00:00<00:00,  8.46it/s]\n",
            "cffi-1.15.1          | 227 KB    | : 100% 1.0/1 [00:00<00:00,  9.47it/s]\n",
            "libgcc-ng-12.1.0     | 940 KB    | : 100% 1.0/1 [00:00<00:00,  4.56it/s]\n",
            "pyopenssl-22.0.0     | 49 KB     | : 100% 1.0/1 [00:00<00:00, 15.22it/s]\n",
            "keyutils-1.6.1       | 115 KB    | : 100% 1.0/1 [00:00<00:00, 11.43it/s]\n",
            "openssl-1.1.1q       | 2.1 MB    | : 100% 1.0/1 [00:00<00:00,  2.30it/s]\n",
            "requests-2.28.1      | 53 KB     | : 100% 1.0/1 [00:00<00:00, 14.73it/s]\n",
            "python-3.7.12        | 57.3 MB   | : 100% 1.0/1 [00:09<00:00,  9.99s/it]               \n",
            "pycparser-2.21       | 100 KB    | : 100% 1.0/1 [00:00<00:00, 10.67it/s]\n",
            "libnsl-2.0.0         | 31 KB     | : 100% 1.0/1 [00:00<00:00, 17.84it/s]\n",
            "idna-3.3             | 55 KB     | : 100% 1.0/1 [00:00<00:00, 17.88it/s]\n",
            "conda-4.13.0         | 989 KB    | : 100% 1.0/1 [00:00<00:00,  3.11it/s]\n",
            "libssh2-1.10.0       | 233 KB    | : 100% 1.0/1 [00:00<00:00,  8.42it/s]\n",
            "libffi-3.4.2         | 57 KB     | : 100% 1.0/1 [00:00<00:00, 14.27it/s]\n",
            "icu-70.1             | 13.5 MB   | : 100% 1.0/1 [00:03<00:00,  3.05s/it]               \n",
            "zstd-1.5.2           | 448 KB    | : 100% 1.0/1 [00:00<00:00,  6.70it/s]\n",
            "six-1.16.0           | 14 KB     | : 100% 1.0/1 [00:00<00:00, 15.64it/s]\n",
            "libsolv-0.7.22       | 443 KB    | : 100% 1.0/1 [00:00<00:00,  6.91it/s]\n",
            "tk-8.6.12            | 3.3 MB    | : 100% 1.0/1 [00:00<00:00,  1.28it/s]\n",
            "reproc-14.2.3        | 28 KB     | : 100% 1.0/1 [00:00<00:00, 12.48it/s]\n",
            "pysocks-1.7.1        | 28 KB     | : 100% 1.0/1 [00:00<00:00, 13.20it/s]\n",
            "charset-normalizer-2 | 35 KB     | : 100% 1.0/1 [00:00<00:00, 16.98it/s]\n",
            "c-ares-1.18.1        | 113 KB    | : 100% 1.0/1 [00:00<00:00, 13.47it/s]\n",
            "urllib3-1.26.11      | 102 KB    | : 100% 1.0/1 [00:00<00:00, 11.13it/s]\n",
            "libcurl-7.83.1       | 342 KB    | : 100% 1.0/1 [00:00<00:00,  8.48it/s]\n",
            "Preparing transaction: | \b\b/ \b\b- \b\b\\ \b\bdone\n",
            "Verifying transaction: / \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\bdone\n",
            "Executing transaction: / \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\bdone\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!conda create --name vep"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "GmndF31P7miP",
        "outputId": "d6827596-3297-4cfc-ed64-be539ffdf7f6"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Collecting package metadata (current_repodata.json): - \b\b\\ \b\bdone\n",
            "Solving environment: / \b\bdone\n",
            "\n",
            "## Package Plan ##\n",
            "\n",
            "  environment location: /usr/local/envs/vep\n",
            "\n",
            "\n",
            "\n",
            "Preparing transaction: \\ \b\bdone\n",
            "Verifying transaction: / \b\bdone\n",
            "Executing transaction: \\ \b\bdone\n",
            "#\n",
            "# To activate this environment, use\n",
            "#\n",
            "#     $ conda activate vep\n",
            "#\n",
            "# To deactivate an active environment, use\n",
            "#\n",
            "#     $ conda deactivate\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!conda install -c bioconda ensembl-vep "
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "Xn7rCUD58dtL",
        "outputId": "a9eab9e0-34a2-4264-e5d4-15be6c2bcacc"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Collecting package metadata (current_repodata.json): - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\bdone\n",
            "Solving environment: / \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\bfailed with initial frozen solve. Retrying with flexible solve.\n",
            "Solving environment: - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\bfailed with repodata from current_repodata.json, will retry with next repodata source.\n",
            "Collecting package metadata (repodata.json): - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\bdone\n",
            "Solving environment: / \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \n",
            "The environment is inconsistent, please check the package plan carefully\n",
            "The following packages are causing the inconsistency:\n",
            "\n",
            "  - conda-forge/linux-64::mamba==0.8.0=py37h7f483ca_0\n",
            "\b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\bdone\n",
            "\n",
            "## Package Plan ##\n",
            "\n",
            "  environment location: /usr/local\n",
            "\n",
            "  added / updated specs:\n",
            "    - ensembl-vep\n",
            "\n",
            "\n",
            "The following packages will be downloaded:\n",
            "\n",
            "    package                    |            build\n",
            "    ---------------------------|-----------------\n",
            "    clustalw-2.1               |       h9f5acd7_7         311 KB  bioconda\n",
            "    curl-7.83.1                |       h7bff187_0          89 KB  conda-forge\n",
            "    ensembl-vep-107.0          | pl5321h4a94de4_0         3.1 MB  bioconda\n",
            "    expat-2.4.8                |       h27087fc_0         187 KB  conda-forge\n",
            "    htslib-1.14                |       h9753748_2         2.0 MB  bioconda\n",
            "    libdeflate-1.10            |       h7f98852_0          77 KB  conda-forge\n",
            "    libgcc-7.2.0               |       h69d50b8_2         304 KB  conda-forge\n",
            "    libmamba-0.24.0            |       hd8a31e3_1         1.4 MB  conda-forge\n",
            "    libmambapy-0.24.0          |   py37h1ee4b26_1         276 KB  conda-forge\n",
            "    mamba-0.24.0               |   py37h6dacc13_1          45 KB  conda-forge\n",
            "    mysql-connector-c-6.1.11   |    h6eb9d5d_1007         2.7 MB  conda-forge\n",
            "    paml-4.9                   |       hec16e2b_7         994 KB  bioconda\n",
            "    perl-5.32.1                | 2_h7f98852_perl5        14.4 MB  conda-forge\n",
            "    perl-algorithm-diff-1.1903 | pl5321hdfd78af_3          36 KB  bioconda\n",
            "    perl-base-2.23             | pl5321hdfd78af_2          12 KB  bioconda\n",
            "    perl-bio-asn1-entrezgene-1.73| pl5321hdfd78af_3          24 KB  bioconda\n",
            "    perl-bio-coordinate-1.007001| pl5321hdfd78af_3          36 KB  bioconda\n",
            "    perl-bio-db-hts-3.01       | pl5321h6233b05_5         121 KB  bioconda\n",
            "    perl-bio-featureio-1.6.905 | pl5321hdfd78af_4          41 KB  bioconda\n",
            "    perl-bio-samtools-1.43     | pl5321h7132678_3         149 KB  bioconda\n",
            "    perl-bio-tools-phylo-paml-1.7.3| pl5321hdfd78af_3          58 KB  bioconda\n",
            "    perl-bio-tools-run-alignment-clustalw-1.7.4| pl5321hdfd78af_3          25 KB  bioconda\n",
            "    perl-bio-tools-run-alignment-tcoffee-1.7.4| pl5321hdfd78af_4          25 KB  bioconda\n",
            "    perl-bioperl-1.7.8         |       hdfd78af_0           8 KB  bioconda\n",
            "    perl-bioperl-core-1.7.8    | pl5321h9ee0642_0         1.8 MB  bioconda\n",
            "    perl-bioperl-run-1.007003  | pl5321hdfd78af_0         340 KB  bioconda\n",
            "    perl-business-isbn-3.007   | pl5321hdfd78af_0          18 KB  bioconda\n",
            "    perl-business-isbn-data-20210112.006| pl5321hdfd78af_0          21 KB  bioconda\n",
            "    perl-capture-tiny-0.48     | pl5321hdfd78af_2          19 KB  bioconda\n",
            "    perl-carp-1.38             | pl5321hdfd78af_4          17 KB  bioconda\n",
            "    perl-class-data-inheritable-0.09| pl5321hdfd78af_0          10 KB  bioconda\n",
            "    perl-common-sense-3.75     | pl5321hdfd78af_0          13 KB  bioconda\n",
            "    perl-compress-raw-bzip2-2.201| pl5321h87f3376_0          47 KB  bioconda\n",
            "    perl-compress-raw-zlib-2.105| pl5321h87f3376_0          75 KB  bioconda\n",
            "    perl-constant-1.33         | pl5321hdfd78af_2          12 KB  bioconda\n",
            "    perl-data-dumper-2.183     | pl5321hec16e2b_1          38 KB  bioconda\n",
            "    perl-dbd-mysql-4.046       | pl5321h9f5acd7_4         252 KB  bioconda\n",
            "    perl-dbi-1.643             | pl5321hec16e2b_1         605 KB  bioconda\n",
            "    perl-devel-stacktrace-2.04 | pl5321hdfd78af_1          17 KB  bioconda\n",
            "    perl-digest-hmac-1.04      | pl5321hdfd78af_0          10 KB  bioconda\n",
            "    perl-digest-md5-2.58       | pl5321hec16e2b_1          19 KB  bioconda\n",
            "    perl-encode-3.18           | pl5321hec16e2b_0         2.2 MB  bioconda\n",
            "    perl-encode-locale-1.05    | pl5321hdfd78af_7          13 KB  bioconda\n",
            "    perl-exception-class-1.45  | pl5321hdfd78af_0          21 KB  bioconda\n",
            "    perl-exporter-5.72         | pl5321hdfd78af_2          16 KB  bioconda\n",
            "    perl-extutils-makemaker-6.66|                0           2 KB  bioconda\n",
            "    perl-file-listing-6.15     | pl5321hdfd78af_0          14 KB  bioconda\n",
            "    perl-file-slurp-tiny-0.004 | pl5321hdfd78af_2          10 KB  bioconda\n",
            "    perl-file-sort-1.01        | pl5321hdfd78af_3          20 KB  bioconda\n",
            "    perl-file-spec-3.48_01     | pl5321hdfd78af_2           6 KB  bioconda\n",
            "    perl-getopt-long-2.52      | pl5321hdfd78af_0          30 KB  bioconda\n",
            "    perl-html-parser-3.78      | pl5321h9f5acd7_0          67 KB  bioconda\n",
            "    perl-html-tagset-3.20      | pl5321hdfd78af_4          14 KB  bioconda\n",
            "    perl-http-cookies-6.10     | pl5321hdfd78af_0          22 KB  bioconda\n",
            "    perl-http-daemon-6.14      | pl5321hdfd78af_0          20 KB  bioconda\n",
            "    perl-http-date-6.05        | pl5321hdfd78af_0          13 KB  bioconda\n",
            "    perl-http-message-6.36     | pl5321hdfd78af_0          54 KB  bioconda\n",
            "    perl-http-negotiate-6.01   | pl5321hdfd78af_4          14 KB  bioconda\n",
            "    perl-io-compress-2.106     | pl5321h87f3376_0          83 KB  bioconda\n",
            "    perl-io-html-1.004         | pl5321hdfd78af_0          16 KB  bioconda\n",
            "    perl-io-socket-ssl-2.074   | pl5321hdfd78af_0         162 KB  bioconda\n",
            "    perl-io-string-1.08        | pl5321hdfd78af_4          12 KB  bioconda\n",
            "    perl-io-tty-1.16           | pl5321hec16e2b_1          32 KB  bioconda\n",
            "    perl-ipc-run-20200505.0    | pl5321hdfd78af_0          76 KB  bioconda\n",
            "    perl-json-4.07             | pl5321hdfd78af_0          55 KB  bioconda\n",
            "    perl-json-xs-2.34          | pl5321h9f5acd7_5          64 KB  bioconda\n",
            "    perl-libwww-perl-6.67      | pl5321hdfd78af_0          97 KB  bioconda\n",
            "    perl-libxml-perl-0.08      | pl5321hdfd78af_3          40 KB  bioconda\n",
            "    perl-lwp-mediatypes-6.04   | pl5321hdfd78af_1          24 KB  bioconda\n",
            "    perl-mime-base64-3.16      | pl5321hec16e2b_2          17 KB  bioconda\n",
            "    perl-net-http-6.22         | pl5321hdfd78af_0          21 KB  bioconda\n",
            "    perl-net-ssleay-1.92       | pl5321h0e0aaa8_1         302 KB  bioconda\n",
            "    perl-ntlm-1.09             | pl5321hdfd78af_5          16 KB  bioconda\n",
            "    perl-parent-0.236          | pl5321hdfd78af_2           8 KB  bioconda\n",
            "    perl-perlio-gzip-0.20      | pl5321h7132678_3          18 KB  bioconda\n",
            "    perl-scalar-list-utils-1.62| pl5321hec16e2b_0          43 KB  bioconda\n",
            "    perl-sereal-4.019          | pl5321hdfd78af_0          11 KB  bioconda\n",
            "    perl-sereal-decoder-4.023  | pl5321hec16e2b_1         169 KB  bioconda\n",
            "    perl-sereal-encoder-4.023  | pl5321hec16e2b_0         282 KB  bioconda\n",
            "    perl-set-intervaltree-0.12 | pl5321h87f3376_2          31 KB  bioconda\n",
            "    perl-socket-2.027          | pl5321hec16e2b_3          33 KB  bioconda\n",
            "    perl-sub-uplevel-0.2800    | pl5321hec16e2b_4          14 KB  bioconda\n",
            "    perl-test-deep-1.130       | pl5321hdfd78af_0          37 KB  bioconda\n",
            "    perl-test-differences-0.69 | pl5321hdfd78af_0          19 KB  bioconda\n",
            "    perl-test-exception-0.43   | pl5321hdfd78af_3          15 KB  bioconda\n",
            "    perl-test-harness-3.44     | pl5321hdfd78af_0          85 KB  bioconda\n",
            "    perl-test-most-0.37        | pl5321hdfd78af_0          20 KB  bioconda\n",
            "    perl-test-warn-0.36        | pl5321hdfd78af_2          14 KB  bioconda\n",
            "    perl-text-csv-2.01         | pl5321hdfd78af_0          70 KB  bioconda\n",
            "    perl-text-diff-1.45        | pl5321hdfd78af_1          23 KB  bioconda\n",
            "    perl-threaded-5.32.1       |       hdfd78af_1           5 KB  bioconda\n",
            "    perl-time-local-1.30       | pl5321hdfd78af_0          12 KB  bioconda\n",
            "    perl-timedate-2.33         | pl5321hdfd78af_2          31 KB  bioconda\n",
            "    perl-tree-dag_node-1.32    | pl5321hdfd78af_0          43 KB  bioconda\n",
            "    perl-try-tiny-0.31         | pl5321hdfd78af_0          18 KB  bioconda\n",
            "    perl-types-serialiser-1.01 | pl5321hdfd78af_0          13 KB  bioconda\n",
            "    perl-uri-5.12              | pl5321hdfd78af_0          59 KB  bioconda\n",
            "    perl-url-encode-0.03       | pl5321h9ee0642_0          15 KB  bioconda\n",
            "    perl-www-robotrules-6.02   | pl5321hdfd78af_4          14 KB  bioconda\n",
            "    perl-xml-dom-1.46          | pl5321hdfd78af_1          59 KB  bioconda\n",
            "    perl-xml-dom-xpath-0.14    | pl5321hdfd78af_2          12 KB  bioconda\n",
            "    perl-xml-parser-2.44       |                4         151 KB  bioconda\n",
            "    perl-xml-regexp-0.04       | pl5321hdfd78af_3          10 KB  bioconda\n",
            "    perl-xml-xpathengine-0.14  | pl5321hdfd78af_3          28 KB  bioconda\n",
            "    pybind11-abi-4             |       hd8ed1ab_3          10 KB  conda-forge\n",
            "    t_coffee-11.0.8            |   py37hea885bf_8        33.2 MB  bioconda\n",
            "    unzip-6.0                  |       h7f98852_3         165 KB  conda-forge\n",
            "    yaml-cpp-0.7.0             |       h27087fc_1         214 KB  conda-forge\n",
            "    ------------------------------------------------------------\n",
            "                                           Total:        67.9 MB\n",
            "\n",
            "The following NEW packages will be INSTALLED:\n",
            "\n",
            "  clustalw           bioconda/linux-64::clustalw-2.1-h9f5acd7_7\n",
            "  curl               conda-forge/linux-64::curl-7.83.1-h7bff187_0\n",
            "  ensembl-vep        bioconda/noarch::ensembl-vep-107.0-pl5321h4a94de4_0\n",
            "  expat              conda-forge/linux-64::expat-2.4.8-h27087fc_0\n",
            "  htslib             bioconda/linux-64::htslib-1.14-h9753748_2\n",
            "  libdeflate         conda-forge/linux-64::libdeflate-1.10-h7f98852_0\n",
            "  libgcc             conda-forge/linux-64::libgcc-7.2.0-h69d50b8_2\n",
            "  libmamba           conda-forge/linux-64::libmamba-0.24.0-hd8a31e3_1\n",
            "  libmambapy         conda-forge/linux-64::libmambapy-0.24.0-py37h1ee4b26_1\n",
            "  mysql-connector-c  conda-forge/linux-64::mysql-connector-c-6.1.11-h6eb9d5d_1007\n",
            "  paml               bioconda/linux-64::paml-4.9-hec16e2b_7\n",
            "  perl               conda-forge/linux-64::perl-5.32.1-2_h7f98852_perl5\n",
            "  perl-algorithm-di~ bioconda/noarch::perl-algorithm-diff-1.1903-pl5321hdfd78af_3\n",
            "  perl-base          bioconda/noarch::perl-base-2.23-pl5321hdfd78af_2\n",
            "  perl-bio-asn1-ent~ bioconda/noarch::perl-bio-asn1-entrezgene-1.73-pl5321hdfd78af_3\n",
            "  perl-bio-coordina~ bioconda/noarch::perl-bio-coordinate-1.007001-pl5321hdfd78af_3\n",
            "  perl-bio-db-hts    bioconda/linux-64::perl-bio-db-hts-3.01-pl5321h6233b05_5\n",
            "  perl-bio-featureio bioconda/noarch::perl-bio-featureio-1.6.905-pl5321hdfd78af_4\n",
            "  perl-bio-samtools  bioconda/linux-64::perl-bio-samtools-1.43-pl5321h7132678_3\n",
            "  perl-bio-tools-ph~ bioconda/noarch::perl-bio-tools-phylo-paml-1.7.3-pl5321hdfd78af_3\n",
            "  perl-bio-tools-ru~ bioconda/noarch::perl-bio-tools-run-alignment-clustalw-1.7.4-pl5321hdfd78af_3\n",
            "  perl-bio-tools-ru~ bioconda/noarch::perl-bio-tools-run-alignment-tcoffee-1.7.4-pl5321hdfd78af_4\n",
            "  perl-bioperl       bioconda/noarch::perl-bioperl-1.7.8-hdfd78af_0\n",
            "  perl-bioperl-core  bioconda/linux-64::perl-bioperl-core-1.7.8-pl5321h9ee0642_0\n",
            "  perl-bioperl-run   bioconda/noarch::perl-bioperl-run-1.007003-pl5321hdfd78af_0\n",
            "  perl-business-isbn bioconda/noarch::perl-business-isbn-3.007-pl5321hdfd78af_0\n",
            "  perl-business-isb~ bioconda/noarch::perl-business-isbn-data-20210112.006-pl5321hdfd78af_0\n",
            "  perl-capture-tiny  bioconda/noarch::perl-capture-tiny-0.48-pl5321hdfd78af_2\n",
            "  perl-carp          bioconda/noarch::perl-carp-1.38-pl5321hdfd78af_4\n",
            "  perl-class-data-i~ bioconda/noarch::perl-class-data-inheritable-0.09-pl5321hdfd78af_0\n",
            "  perl-common-sense  bioconda/noarch::perl-common-sense-3.75-pl5321hdfd78af_0\n",
            "  perl-compress-raw~ bioconda/linux-64::perl-compress-raw-bzip2-2.201-pl5321h87f3376_0\n",
            "  perl-compress-raw~ bioconda/linux-64::perl-compress-raw-zlib-2.105-pl5321h87f3376_0\n",
            "  perl-constant      bioconda/noarch::perl-constant-1.33-pl5321hdfd78af_2\n",
            "  perl-data-dumper   bioconda/linux-64::perl-data-dumper-2.183-pl5321hec16e2b_1\n",
            "  perl-dbd-mysql     bioconda/linux-64::perl-dbd-mysql-4.046-pl5321h9f5acd7_4\n",
            "  perl-dbi           bioconda/linux-64::perl-dbi-1.643-pl5321hec16e2b_1\n",
            "  perl-devel-stackt~ bioconda/noarch::perl-devel-stacktrace-2.04-pl5321hdfd78af_1\n",
            "  perl-digest-hmac   bioconda/noarch::perl-digest-hmac-1.04-pl5321hdfd78af_0\n",
            "  perl-digest-md5    bioconda/linux-64::perl-digest-md5-2.58-pl5321hec16e2b_1\n",
            "  perl-encode        bioconda/linux-64::perl-encode-3.18-pl5321hec16e2b_0\n",
            "  perl-encode-locale bioconda/noarch::perl-encode-locale-1.05-pl5321hdfd78af_7\n",
            "  perl-exception-cl~ bioconda/noarch::perl-exception-class-1.45-pl5321hdfd78af_0\n",
            "  perl-exporter      bioconda/noarch::perl-exporter-5.72-pl5321hdfd78af_2\n",
            "  perl-extutils-mak~ bioconda/linux-64::perl-extutils-makemaker-6.66-0\n",
            "  perl-file-listing  bioconda/noarch::perl-file-listing-6.15-pl5321hdfd78af_0\n",
            "  perl-file-slurp-t~ bioconda/noarch::perl-file-slurp-tiny-0.004-pl5321hdfd78af_2\n",
            "  perl-file-sort     bioconda/noarch::perl-file-sort-1.01-pl5321hdfd78af_3\n",
            "  perl-file-spec     bioconda/noarch::perl-file-spec-3.48_01-pl5321hdfd78af_2\n",
            "  perl-getopt-long   bioconda/noarch::perl-getopt-long-2.52-pl5321hdfd78af_0\n",
            "  perl-html-parser   bioconda/linux-64::perl-html-parser-3.78-pl5321h9f5acd7_0\n",
            "  perl-html-tagset   bioconda/noarch::perl-html-tagset-3.20-pl5321hdfd78af_4\n",
            "  perl-http-cookies  bioconda/noarch::perl-http-cookies-6.10-pl5321hdfd78af_0\n",
            "  perl-http-daemon   bioconda/noarch::perl-http-daemon-6.14-pl5321hdfd78af_0\n",
            "  perl-http-date     bioconda/noarch::perl-http-date-6.05-pl5321hdfd78af_0\n",
            "  perl-http-message  bioconda/noarch::perl-http-message-6.36-pl5321hdfd78af_0\n",
            "  perl-http-negotia~ bioconda/noarch::perl-http-negotiate-6.01-pl5321hdfd78af_4\n",
            "  perl-io-compress   bioconda/linux-64::perl-io-compress-2.106-pl5321h87f3376_0\n",
            "  perl-io-html       bioconda/noarch::perl-io-html-1.004-pl5321hdfd78af_0\n",
            "  perl-io-socket-ssl bioconda/noarch::perl-io-socket-ssl-2.074-pl5321hdfd78af_0\n",
            "  perl-io-string     bioconda/noarch::perl-io-string-1.08-pl5321hdfd78af_4\n",
            "  perl-io-tty        bioconda/linux-64::perl-io-tty-1.16-pl5321hec16e2b_1\n",
            "  perl-ipc-run       bioconda/noarch::perl-ipc-run-20200505.0-pl5321hdfd78af_0\n",
            "  perl-json          bioconda/noarch::perl-json-4.07-pl5321hdfd78af_0\n",
            "  perl-json-xs       bioconda/linux-64::perl-json-xs-2.34-pl5321h9f5acd7_5\n",
            "  perl-libwww-perl   bioconda/noarch::perl-libwww-perl-6.67-pl5321hdfd78af_0\n",
            "  perl-libxml-perl   bioconda/noarch::perl-libxml-perl-0.08-pl5321hdfd78af_3\n",
            "  perl-lwp-mediatyp~ bioconda/noarch::perl-lwp-mediatypes-6.04-pl5321hdfd78af_1\n",
            "  perl-mime-base64   bioconda/linux-64::perl-mime-base64-3.16-pl5321hec16e2b_2\n",
            "  perl-net-http      bioconda/noarch::perl-net-http-6.22-pl5321hdfd78af_0\n",
            "  perl-net-ssleay    bioconda/linux-64::perl-net-ssleay-1.92-pl5321h0e0aaa8_1\n",
            "  perl-ntlm          bioconda/noarch::perl-ntlm-1.09-pl5321hdfd78af_5\n",
            "  perl-parent        bioconda/noarch::perl-parent-0.236-pl5321hdfd78af_2\n",
            "  perl-perlio-gzip   bioconda/linux-64::perl-perlio-gzip-0.20-pl5321h7132678_3\n",
            "  perl-scalar-list-~ bioconda/linux-64::perl-scalar-list-utils-1.62-pl5321hec16e2b_0\n",
            "  perl-sereal        bioconda/noarch::perl-sereal-4.019-pl5321hdfd78af_0\n",
            "  perl-sereal-decod~ bioconda/linux-64::perl-sereal-decoder-4.023-pl5321hec16e2b_1\n",
            "  perl-sereal-encod~ bioconda/linux-64::perl-sereal-encoder-4.023-pl5321hec16e2b_0\n",
            "  perl-set-interval~ bioconda/linux-64::perl-set-intervaltree-0.12-pl5321h87f3376_2\n",
            "  perl-socket        bioconda/linux-64::perl-socket-2.027-pl5321hec16e2b_3\n",
            "  perl-sub-uplevel   bioconda/linux-64::perl-sub-uplevel-0.2800-pl5321hec16e2b_4\n",
            "  perl-test-deep     bioconda/noarch::perl-test-deep-1.130-pl5321hdfd78af_0\n",
            "  perl-test-differe~ bioconda/noarch::perl-test-differences-0.69-pl5321hdfd78af_0\n",
            "  perl-test-excepti~ bioconda/noarch::perl-test-exception-0.43-pl5321hdfd78af_3\n",
            "  perl-test-harness  bioconda/noarch::perl-test-harness-3.44-pl5321hdfd78af_0\n",
            "  perl-test-most     bioconda/noarch::perl-test-most-0.37-pl5321hdfd78af_0\n",
            "  perl-test-warn     bioconda/noarch::perl-test-warn-0.36-pl5321hdfd78af_2\n",
            "  perl-text-csv      bioconda/noarch::perl-text-csv-2.01-pl5321hdfd78af_0\n",
            "  perl-text-diff     bioconda/noarch::perl-text-diff-1.45-pl5321hdfd78af_1\n",
            "  perl-threaded      bioconda/noarch::perl-threaded-5.32.1-hdfd78af_1\n",
            "  perl-time-local    bioconda/noarch::perl-time-local-1.30-pl5321hdfd78af_0\n",
            "  perl-timedate      bioconda/noarch::perl-timedate-2.33-pl5321hdfd78af_2\n",
            "  perl-tree-dag_node bioconda/noarch::perl-tree-dag_node-1.32-pl5321hdfd78af_0\n",
            "  perl-try-tiny      bioconda/noarch::perl-try-tiny-0.31-pl5321hdfd78af_0\n",
            "  perl-types-serial~ bioconda/noarch::perl-types-serialiser-1.01-pl5321hdfd78af_0\n",
            "  perl-uri           bioconda/noarch::perl-uri-5.12-pl5321hdfd78af_0\n",
            "  perl-url-encode    bioconda/linux-64::perl-url-encode-0.03-pl5321h9ee0642_0\n",
            "  perl-www-robotrul~ bioconda/noarch::perl-www-robotrules-6.02-pl5321hdfd78af_4\n",
            "  perl-xml-dom       bioconda/noarch::perl-xml-dom-1.46-pl5321hdfd78af_1\n",
            "  perl-xml-dom-xpath bioconda/noarch::perl-xml-dom-xpath-0.14-pl5321hdfd78af_2\n",
            "  perl-xml-parser    bioconda/linux-64::perl-xml-parser-2.44-4\n",
            "  perl-xml-regexp    bioconda/noarch::perl-xml-regexp-0.04-pl5321hdfd78af_3\n",
            "  perl-xml-xpatheng~ bioconda/noarch::perl-xml-xpathengine-0.14-pl5321hdfd78af_3\n",
            "  pybind11-abi       conda-forge/noarch::pybind11-abi-4-hd8ed1ab_3\n",
            "  t_coffee           bioconda/linux-64::t_coffee-11.0.8-py37hea885bf_8\n",
            "  unzip              conda-forge/linux-64::unzip-6.0-h7f98852_3\n",
            "  yaml-cpp           conda-forge/linux-64::yaml-cpp-0.7.0-h27087fc_1\n",
            "\n",
            "The following packages will be UPDATED:\n",
            "\n",
            "  mamba                                0.8.0-py37h7f483ca_0 --> 0.24.0-py37h6dacc13_1\n",
            "\n",
            "\n",
            "\n",
            "Downloading and Extracting Packages\n",
            "perl-bio-asn1-entrez | 24 KB     | : 100% 1.0/1 [00:00<00:00,  6.72it/s]               \n",
            "perl-algorithm-diff- | 36 KB     | : 100% 1.0/1 [00:00<00:00, 13.02it/s]\n",
            "perl-base-2.23       | 12 KB     | : 100% 1.0/1 [00:00<00:00, 16.30it/s]\n",
            "perl-data-dumper-2.1 | 38 KB     | : 100% 1.0/1 [00:00<00:00, 12.91it/s]\n",
            "perl-bioperl-1.7.8   | 8 KB      | : 100% 1.0/1 [00:00<00:00,  4.61it/s]\n",
            "perl-business-isbn-d | 21 KB     | : 100% 1.0/1 [00:00<00:00, 16.49it/s]\n",
            "perl-io-string-1.08  | 12 KB     | : 100% 1.0/1 [00:00<00:00, 19.28it/s]\n",
            "perl-file-spec-3.48_ | 6 KB      | : 100% 1.0/1 [00:00<00:00, 17.13it/s]\n",
            "perl-bio-samtools-1. | 149 KB    | : 100% 1.0/1 [00:00<00:00, 11.32it/s]\n",
            "perl-xml-regexp-0.04 | 10 KB     | : 100% 1.0/1 [00:00<00:00, 19.48it/s]\n",
            "perl-file-listing-6. | 14 KB     | : 100% 1.0/1 [00:00<00:00, 13.54it/s]\n",
            "perl-timedate-2.33   | 31 KB     | : 100% 1.0/1 [00:00<00:00, 11.38it/s]\n",
            "perl-html-parser-3.7 | 67 KB     | : 100% 1.0/1 [00:00<00:00, 11.81it/s]\n",
            "perl-dbi-1.643       | 605 KB    | : 100% 1.0/1 [00:00<00:00,  4.64it/s]\n",
            "perl-compress-raw-bz | 47 KB     | : 100% 1.0/1 [00:00<00:00, 14.19it/s]\n",
            "perl-lwp-mediatypes- | 24 KB     | : 100% 1.0/1 [00:00<00:00, 15.91it/s]\n",
            "perl-http-negotiate- | 14 KB     | : 100% 1.0/1 [00:00<00:00, 12.24it/s]\n",
            "libgcc-7.2.0         | 304 KB    | : 100% 1.0/1 [00:00<00:00,  9.32it/s]\n",
            "perl-ipc-run-2020050 | 76 KB     | : 100% 1.0/1 [00:00<00:00, 11.28it/s]\n",
            "perl-file-sort-1.01  | 20 KB     | : 100% 1.0/1 [00:00<00:00, 18.94it/s]\n",
            "perl-xml-xpathengine | 28 KB     | : 100% 1.0/1 [00:00<00:00, 14.02it/s]\n",
            "perl-http-cookies-6. | 22 KB     | : 100% 1.0/1 [00:00<00:00, 15.63it/s]\n",
            "htslib-1.14          | 2.0 MB    | : 100% 1.0/1 [00:00<00:00,  1.99it/s]\n",
            "perl-time-local-1.30 | 12 KB     | : 100% 1.0/1 [00:00<00:00, 17.01it/s]\n",
            "perl-parent-0.236    | 8 KB      | : 100% 1.0/1 [00:00<00:00, 13.87it/s]\n",
            "perl-business-isbn-3 | 18 KB     | : 100% 1.0/1 [00:00<00:00, 18.35it/s]\n",
            "perl-io-tty-1.16     | 32 KB     | : 100% 1.0/1 [00:00<00:00, 17.18it/s]\n",
            "ensembl-vep-107.0    | 3.1 MB    | : 100% 1.0/1 [00:01<00:00,  1.58s/it]               \n",
            "perl-xml-dom-1.46    | 59 KB     | : 100% 1.0/1 [00:00<00:00, 14.49it/s]\n",
            "perl-bioperl-core-1. | 1.8 MB    | : 100% 1.0/1 [00:01<00:00,  1.87s/it]               \n",
            "perl-libxml-perl-0.0 | 40 KB     | : 100% 1.0/1 [00:00<00:00, 12.19it/s]\n",
            "perl-bio-featureio-1 | 41 KB     | : 100% 1.0/1 [00:00<00:00, 14.51it/s]\n",
            "perl-io-compress-2.1 | 83 KB     | : 100% 1.0/1 [00:00<00:00,  7.59it/s]\n",
            "yaml-cpp-0.7.0       | 214 KB    | : 100% 1.0/1 [00:00<00:00,  8.93it/s]\n",
            "perl-url-encode-0.03 | 15 KB     | : 100% 1.0/1 [00:00<00:00, 16.80it/s]\n",
            "perl-getopt-long-2.5 | 30 KB     | : 100% 1.0/1 [00:00<00:00, 15.00it/s]\n",
            "perl-json-4.07       | 55 KB     | : 100% 1.0/1 [00:00<00:00, 14.88it/s]\n",
            "perl-net-http-6.22   | 21 KB     | : 100% 1.0/1 [00:00<00:00, 16.11it/s]\n",
            "perl-test-difference | 19 KB     | : 100% 1.0/1 [00:00<00:00, 14.14it/s]\n",
            "curl-7.83.1          | 89 KB     | : 100% 1.0/1 [00:00<00:00, 13.93it/s]\n",
            "perl-scalar-list-uti | 43 KB     | : 100% 1.0/1 [00:00<00:00, 15.07it/s]\n",
            "perl-exporter-5.72   | 16 KB     | : 100% 1.0/1 [00:00<00:00, 12.70it/s]\n",
            "perl-test-exception- | 15 KB     | : 100% 1.0/1 [00:00<00:00, 17.72it/s]\n",
            "perl-sereal-4.019    | 11 KB     | : 100% 1.0/1 [00:00<00:00,  7.80it/s]\n",
            "perl-xml-dom-xpath-0 | 12 KB     | : 100% 1.0/1 [00:00<00:00, 19.94it/s]\n",
            "perl-file-slurp-tiny | 10 KB     | : 100% 1.0/1 [00:00<00:00, 20.02it/s]\n",
            "perl-text-diff-1.45  | 23 KB     | : 100% 1.0/1 [00:00<00:00, 16.88it/s]\n",
            "mysql-connector-c-6. | 2.7 MB    | : 100% 1.0/1 [00:00<00:00,  1.96it/s]\n",
            "perl-ntlm-1.09       | 16 KB     | : 100% 1.0/1 [00:00<00:00, 20.04it/s]\n",
            "pybind11-abi-4       | 10 KB     | : 100% 1.0/1 [00:00<00:00, 11.72it/s]\n",
            "perl-devel-stacktrac | 17 KB     | : 100% 1.0/1 [00:00<00:00, 15.62it/s]\n",
            "perl-bio-tools-run-a | 25 KB     | : 100% 1.0/1 [00:00<00:00, 15.91it/s]\n",
            "perl-sub-uplevel-0.2 | 14 KB     | : 100% 1.0/1 [00:00<00:00, 16.50it/s]\n",
            "perl-io-socket-ssl-2 | 162 KB    | : 100% 1.0/1 [00:00<00:00, 10.86it/s]\n",
            "perl-bio-tools-run-a | 25 KB     | : 100% 1.0/1 [00:00<00:00, 13.77it/s]\n",
            "perl-constant-1.33   | 12 KB     | : 100% 1.0/1 [00:00<00:00, 18.25it/s]\n",
            "perl-exception-class | 21 KB     | : 100% 1.0/1 [00:00<00:00, 16.12it/s]\n",
            "perl-http-daemon-6.1 | 20 KB     | : 100% 1.0/1 [00:00<00:00, 12.92it/s]\n",
            "unzip-6.0            | 165 KB    | : 100% 1.0/1 [00:00<00:00, 10.88it/s]\n",
            "perl-try-tiny-0.31   | 18 KB     | : 100% 1.0/1 [00:00<00:00, 16.79it/s]\n",
            "perl-5.32.1          | 14.4 MB   | : 100% 1.0/1 [00:03<00:00,  3.90s/it]               \n",
            "perl-threaded-5.32.1 | 5 KB      | : 100% 1.0/1 [00:00<00:00, 16.03it/s]\n",
            "perl-set-intervaltre | 31 KB     | : 100% 1.0/1 [00:00<00:00,  5.03it/s]               \n",
            "perl-bio-coordinate- | 36 KB     | : 100% 1.0/1 [00:00<00:00, 13.17it/s]\n",
            "perl-socket-2.027    | 33 KB     | : 100% 1.0/1 [00:00<00:00, 19.46it/s]\n",
            "expat-2.4.8          | 187 KB    | : 100% 1.0/1 [00:00<00:00,  8.47it/s]\n",
            "perl-encode-3.18     | 2.2 MB    | : 100% 1.0/1 [00:00<00:00,  1.77it/s]               \n",
            "perl-extutils-makema | 2 KB      | : 100% 1.0/1 [00:00<00:00,  9.71it/s]\n",
            "perl-http-date-6.05  | 13 KB     | : 100% 1.0/1 [00:00<00:00, 12.53it/s]\n",
            "perl-types-serialise | 13 KB     | : 100% 1.0/1 [00:00<00:00, 14.99it/s]\n",
            "perl-json-xs-2.34    | 64 KB     | : 100% 1.0/1 [00:00<00:00,  8.02it/s]              \n",
            "perl-digest-md5-2.58 | 19 KB     | : 100% 1.0/1 [00:00<00:00, 15.85it/s]\n",
            "libmamba-0.24.0      | 1.4 MB    | : 100% 1.0/1 [00:00<00:00,  2.60it/s]\n",
            "perl-perlio-gzip-0.2 | 18 KB     | : 100% 1.0/1 [00:00<00:00,  6.75it/s]               \n",
            "perl-html-tagset-3.2 | 14 KB     | : 100% 1.0/1 [00:00<00:00, 18.39it/s]\n",
            "libmambapy-0.24.0    | 276 KB    | : 100% 1.0/1 [00:00<00:00,  7.26it/s]\n",
            "perl-test-warn-0.36  | 14 KB     | : 100% 1.0/1 [00:00<00:00, 10.68it/s]\n",
            "perl-io-html-1.004   | 16 KB     | : 100% 1.0/1 [00:00<00:00, 11.13it/s]\n",
            "perl-sereal-decoder- | 169 KB    | : 100% 1.0/1 [00:00<00:00,  4.90it/s]                \n",
            "perl-libwww-perl-6.6 | 97 KB     | : 100% 1.0/1 [00:00<00:00, 12.97it/s]\n",
            "perl-text-csv-2.01   | 70 KB     | : 100% 1.0/1 [00:00<00:00,  6.18it/s]                \n",
            "perl-bio-tools-phylo | 58 KB     | : 100% 1.0/1 [00:00<00:00, 12.44it/s]\n",
            "perl-uri-5.12        | 59 KB     | : 100% 1.0/1 [00:00<00:00, 12.47it/s]\n",
            "t_coffee-11.0.8      | 33.2 MB   | : 100% 1.0/1 [00:06<00:00,  6.13s/it]               \n",
            "perl-digest-hmac-1.0 | 10 KB     | : 100% 1.0/1 [00:00<00:00, 10.08it/s]\n",
            "perl-sereal-encoder- | 282 KB    | : 100% 1.0/1 [00:00<00:00,  3.88it/s]                \n",
            "perl-bio-db-hts-3.01 | 121 KB    | : 100% 1.0/1 [00:00<00:00,  3.54it/s]                \n",
            "libdeflate-1.10      | 77 KB     | : 100% 1.0/1 [00:00<00:00, 11.02it/s]\n",
            "perl-capture-tiny-0. | 19 KB     | : 100% 1.0/1 [00:00<00:00, 11.60it/s]\n",
            "perl-common-sense-3. | 13 KB     | : 100% 1.0/1 [00:00<00:00, 13.17it/s]\n",
            "perl-bioperl-run-1.0 | 340 KB    | : 100% 1.0/1 [00:00<00:00,  3.93it/s]\n",
            "perl-test-most-0.37  | 20 KB     | : 100% 1.0/1 [00:00<00:00, 11.78it/s]\n",
            "perl-dbd-mysql-4.046 | 252 KB    | : 100% 1.0/1 [00:00<00:00,  4.75it/s]                \n",
            "perl-test-deep-1.130 | 37 KB     | : 100% 1.0/1 [00:00<00:00, 14.41it/s]\n",
            "clustalw-2.1         | 311 KB    | : 100% 1.0/1 [00:00<00:00,  8.67it/s]\n",
            "mamba-0.24.0         | 45 KB     | : 100% 1.0/1 [00:00<00:00, 11.33it/s]\n",
            "paml-4.9             | 994 KB    | : 100% 1.0/1 [00:00<00:00,  3.29it/s]\n",
            "perl-tree-dag_node-1 | 43 KB     | : 100% 1.0/1 [00:00<00:00, 16.02it/s]\n",
            "perl-test-harness-3. | 85 KB     | : 100% 1.0/1 [00:00<00:00, 10.95it/s]\n",
            "perl-www-robotrules- | 14 KB     | : 100% 1.0/1 [00:00<00:00, 17.70it/s]\n",
            "perl-net-ssleay-1.92 | 302 KB    | : 100% 1.0/1 [00:00<00:00,  5.74it/s]\n",
            "perl-encode-locale-1 | 13 KB     | : 100% 1.0/1 [00:00<00:00, 17.02it/s]\n",
            "perl-class-data-inhe | 10 KB     | : 100% 1.0/1 [00:00<00:00, 13.15it/s]\n",
            "perl-compress-raw-zl | 75 KB     | : 100% 1.0/1 [00:00<00:00, 13.28it/s]\n",
            "perl-carp-1.38       | 17 KB     | : 100% 1.0/1 [00:00<00:00, 16.26it/s]\n",
            "perl-xml-parser-2.44 | 151 KB    | : 100% 1.0/1 [00:00<00:00, 12.10it/s]\n",
            "perl-mime-base64-3.1 | 17 KB     | : 100% 1.0/1 [00:00<00:00, 15.15it/s]\n",
            "perl-http-message-6. | 54 KB     | : 100% 1.0/1 [00:00<00:00, 12.95it/s]\n",
            "Preparing transaction: | \b\b/ \b\b- \b\b\\ \b\b| \b\bdone\n",
            "Verifying transaction: - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\bdone\n",
            "Executing transaction: \\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ This package installs only the variant effect predictor (VEP) library\n",
            "code. To install data libraries, you can use the 'vep_install' command\n",
            "installed along with it. For example, to install the VEP library for human\n",
            "GRCh38 to a directory\n",
            "\n",
            "vep_install -a cf -s homo_sapiens -y GRCh38 -c /output/path/to/GRCh38/vep --CONVERT\n",
            "\n",
            "(note that vep_install is renamed from INSTALL.pl\n",
            " to avoid having generic script names in the PATH)\n",
            "\n",
            "The --CONVERT flag is not required but improves lookup speeds during\n",
            "runs. See the VEP documentation for more details\n",
            "\n",
            "http://www.ensembl.org/info/docs/tools/vep/script/vep_cache.html\n",
            "\n",
            "\b\b| \b\b/ \b\b- \b\b\\ \b\bdone\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!activate\n",
        "!source activate\n",
        "!vep --help"
      ],
      "metadata": {
        "id": "zTXvlaE--idB"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!vep -i final_snp.vcf --database --vcf --force"
      ],
      "metadata": {
        "id": "P9854uPc-2Fb"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "import requests, sys\n",
        " \n",
        "server = \"https://rest.ensembl.org\"\n",
        "ext = \"/vep/human/id\"\n",
        "headers={ \"Content-Type\" : \"application/json\", \"Accept\" : \"application/json\"}\n",
        "r = requests.post(server+ext, headers=headers, data='{ \"ids\" : [\"rs56116432\", \"COSM476\" ] }')\n",
        " \n",
        "if not r.ok:\n",
        "  r.raise_for_status()\n",
        "  sys.exit()\n",
        " \n",
        "decoded = r.json()\n",
        "print(repr(decoded))"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "w9rjnGmXRHoC",
        "outputId": "b87af106-ba7a-4fb6-e553-ef1df2a70978"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "[{'end': 133256042, 'id': 'rs56116432', 'colocated_variants': [{'start': 133256042, 'end': 133256042, 'strand': 1, 'seq_region_name': '9', 'id': 'rs56116432', 'frequencies': {'T': {'gnomadg_mid': 0, 'gnomadg_eas': 0, 'sas': 0.001, 'gnomade_nfe': 0.003597, 'gnomade_eas': 0, 'gnomadg_asj': 0.001441, 'gnomadg_sas': 0.0006206, 'gnomadg_fin': 0.01281, 'gnomadg_oth': 0.00239, 'gnomade_asj': 0.002299, 'gnomade_oth': 0.005249, 'eas': 0, 'gnomade_amr': 0.002312, 'eur': 0.0109, 'gnomade_afr': 0.0007352, 'gnomade_sas': 0.001379, 'gnomadg': 0.003052, 'amr': 0.0014, 'afr': 0, 'gnomadg_nfe': 0.003927, 'gnomadg_afr': 0.0005082, 'gnomadg_ami': 0, 'gnomade': 0.003617, 'gnomade_fin': 0.01381, 'gnomadg_amr': 0.001767}}, 'allele_string': 'C/T'}], 'input': 'rs56116432', 'most_severe_consequence': 'missense_variant', 'transcript_consequences': [{'transcript_id': 'ENST00000453660', 'variant_allele': 'T', 'consequence_terms': ['non_coding_transcript_exon_variant'], 'cdna_end': 718, 'strand': -1, 'gene_symbol': 'ABO', 'biotype': 'processed_transcript', 'gene_id': 'ENSG00000175164', 'impact': 'MODIFIER', 'hgnc_id': 'HGNC:79', 'cdna_start': 718, 'gene_symbol_source': 'HGNC'}, {'hgnc_id': 'HGNC:79', 'cdna_start': 711, 'impact': 'MODERATE', 'protein_end': 229, 'cdna_end': 711, 'variant_allele': 'T', 'gene_id': 'ENSG00000175164', 'cds_end': 686, 'biotype': 'protein_coding', 'gene_symbol': 'ABO', 'polyphen_score': 0.995, 'cds_start': 686, 'protein_start': 229, 'polyphen_prediction': 'probably_damaging', 'gene_symbol_source': 'HGNC', 'sift_prediction': 'deleterious', 'strand': -1, 'consequence_terms': ['missense_variant'], 'transcript_id': 'ENST00000538324', 'codons': 'gGc/gAc', 'sift_score': 0, 'amino_acids': 'G/D'}, {'polyphen_prediction': 'probably_damaging', 'polyphen_score': 0.998, 'protein_start': 229, 'cds_start': 686, 'sift_prediction': 'deleterious', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000611156', 'consequence_terms': ['missense_variant'], 'strand': -1, 'amino_acids': 'G/D', 'sift_score': 0, 'codons': 'gGc/gAc', 'impact': 'MODERATE', 'cdna_start': 711, 'hgnc_id': 'HGNC:79', 'variant_allele': 'T', 'cdna_end': 711, 'protein_end': 229, 'gene_symbol': 'ABO', 'biotype': 'protein_coding', 'cds_end': 686, 'gene_id': 'ENSG00000175164'}, {'impact': 'MODIFIER', 'hgnc_id': 'HGNC:79', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'strand': -1, 'transcript_id': 'ENST00000647353', 'variant_allele': 'T', 'gene_id': 'ENSG00000175164', 'gene_symbol': 'ABO', 'biotype': 'processed_transcript'}, {'transcript_id': 'ENST00000651471', 'variant_allele': 'T', 'consequence_terms': ['downstream_gene_variant'], 'strand': -1, 'gene_symbol': 'ABO', 'biotype': 'processed_transcript', 'gene_id': 'ENSG00000175164', 'impact': 'MODIFIER', 'hgnc_id': 'HGNC:79', 'distance': 64, 'gene_symbol_source': 'HGNC'}, {'gene_symbol_source': 'HGNC', 'hgnc_id': 'HGNC:79', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'gene_symbol': 'ABO', 'gene_id': 'ENSG00000175164', 'variant_allele': 'T', 'transcript_id': 'ENST00000679909', 'strand': -1, 'consequence_terms': ['intron_variant']}, {'transcript_id': 'ENST00000680600', 'variant_allele': 'T', 'consequence_terms': ['upstream_gene_variant'], 'strand': -1, 'gene_symbol': 'ABO', 'biotype': 'processed_transcript', 'gene_id': 'ENSG00000175164', 'impact': 'MODIFIER', 'hgnc_id': 'HGNC:79', 'distance': 3445, 'gene_symbol_source': 'HGNC'}], 'seq_region_name': '9', 'assembly_name': 'GRCh38', 'allele_string': 'C/T', 'strand': 1, 'start': 133256042}, {'colocated_variants': [{'allele_string': 'C/T', 'seq_region_name': 'CHR_HG2030_PATCH', 'strand': 1, 'id': 'rs56116432', 'start': 133256189, 'end': 133256189}], 'id': 'rs56116432', 'end': 133256189, 'strand': 1, 'start': 133256189, 'seq_region_name': 'CHR_HG2030_PATCH', 'most_severe_consequence': 'missense_variant', 'transcript_consequences': [{'hgnc_id': 'HGNC:79', 'cdna_start': 689, 'impact': 'MODERATE', 'gene_id': 'ENSG00000281879', 'biotype': 'protein_coding', 'cds_end': 689, 'gene_symbol': 'ABO', 'protein_end': 230, 'cdna_end': 689, 'variant_allele': 'T', 'gene_symbol_source': 'HGNC', 'sift_prediction': 'deleterious', 'cds_start': 689, 'polyphen_score': 0.996, 'protein_start': 230, 'polyphen_prediction': 'probably_damaging', 'codons': 'gGc/gAc', 'sift_score': 0, 'amino_acids': 'G/D', 'strand': -1, 'consequence_terms': ['missense_variant'], 'transcript_id': 'ENST00000626615'}, {'sift_prediction': 'deleterious', 'gene_symbol_source': 'HGNC', 'polyphen_prediction': 'probably_damaging', 'polyphen_score': 0.996, 'protein_start': 230, 'cds_start': 689, 'amino_acids': 'G/D', 'sift_score': 0, 'codons': 'gGc/gAc', 'transcript_id': 'ENST00000644422', 'consequence_terms': ['missense_variant'], 'strand': -1, 'impact': 'MODERATE', 'hgnc_id': 'HGNC:79', 'cdna_start': 701, 'gene_symbol': 'ABO', 'biotype': 'protein_coding', 'cds_end': 689, 'gene_id': 'ENSG00000281879', 'variant_allele': 'T', 'cdna_end': 701, 'protein_end': 230}, {'gene_symbol': 'ABO', 'cds_end': 689, 'biotype': 'protein_coding', 'gene_id': 'ENSG00000281879', 'variant_allele': 'T', 'cdna_end': 689, 'protein_end': 230, 'impact': 'MODERATE', 'cdna_start': 689, 'hgnc_id': 'HGNC:79', 'amino_acids': 'G/D', 'sift_score': 0, 'codons': 'gGc/gAc', 'transcript_id': 'ENST00000644755', 'consequence_terms': ['missense_variant'], 'strand': -1, 'sift_prediction': 'deleterious', 'gene_symbol_source': 'HGNC', 'polyphen_prediction': 'probably_damaging', 'polyphen_score': 0.996, 'cds_start': 689, 'protein_start': 230}, {'gene_symbol_source': 'HGNC', 'sift_prediction': 'deleterious', 'protein_start': 230, 'polyphen_score': 0.996, 'cds_start': 689, 'polyphen_prediction': 'probably_damaging', 'codons': 'gGc/gAc', 'amino_acids': 'G/D', 'sift_score': 0, 'consequence_terms': ['missense_variant'], 'strand': -1, 'transcript_id': 'ENST00000645810', 'impact': 'MODERATE', 'cdna_start': 701, 'hgnc_id': 'HGNC:79', 'gene_id': 'ENSG00000281879', 'gene_symbol': 'ABO', 'biotype': 'protein_coding', 'cds_end': 689, 'cdna_end': 701, 'protein_end': 230, 'variant_allele': 'T'}], 'input': 'rs56116432', 'assembly_name': 'GRCh38', 'allele_string': 'C/T'}, {'colocated_variants': [{'allele_string': 'HGMD_MUTATION', 'end': 140753336, 'start': 140753336, 'phenotype_or_disease': 1, 'id': 'CM112509', 'seq_region_name': '7', 'strand': 1}, {'allele_string': 'COSMIC_MUTATION', 'somatic': 1, 'id': 'COSV56056643', 'strand': 1, 'seq_region_name': '7', 'var_synonyms': {'COSMIC': ['COSM476']}, 'end': 140753336, 'phenotype_or_disease': 1, 'start': 140753336}, {'id': 'COSV56065204', 'somatic': 1, 'strand': 1, 'seq_region_name': '7', 'var_synonyms': {'COSMIC': ['COSM18443']}, 'end': 140753336, 'start': 140753336, 'phenotype_or_disease': 1, 'allele_string': 'COSMIC_MUTATION'}, {'allele_string': 'COSMIC_MUTATION', 'seq_region_name': '7', 'strand': 1, 'id': 'COSV56080151', 'somatic': 1, 'start': 140753336, 'phenotype_or_disease': 1, 'end': 140753336, 'var_synonyms': {'COSMIC': ['COSM6137']}}], 'end': 140753336, 'id': 'COSM476', 'strand': 1, 'start': 140753336, 'most_severe_consequence': '?', 'seq_region_name': '7', 'input': 'COSM476', 'assembly_name': 'GRCh38', 'allele_string': 'COSMIC_MUTATION'}]\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!curl 'https://rest.ensembl.org/vep/human/id' -H 'Content-type:application/json' \\\n",
        "-H 'Accept:application/json' -X POST -d '{ \"ids\" : [\"rs56116432\", \"COSM476\" ] }'"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "lIqubSc9gYMH",
        "outputId": "db6c3b16-973b-4b3f-c3fc-4d1eda3e01eb"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "[{\"most_severe_consequence\":\"missense_variant\",\"assembly_name\":\"GRCh38\",\"end\":133256042,\"allele_string\":\"C/T\",\"strand\":1,\"input\":\"rs56116432\",\"transcript_consequences\":[{\"gene_symbol\":\"ABO\",\"variant_allele\":\"T\",\"gene_symbol_source\":\"HGNC\",\"cdna_start\":718,\"hgnc_id\":\"HGNC:79\",\"consequence_terms\":[\"non_coding_transcript_exon_variant\"],\"gene_id\":\"ENSG00000175164\",\"impact\":\"MODIFIER\",\"cdna_end\":718,\"strand\":-1,\"biotype\":\"processed_transcript\",\"transcript_id\":\"ENST00000453660\"},{\"cds_end\":686,\"cdna_end\":711,\"cds_start\":686,\"biotype\":\"protein_coding\",\"variant_allele\":\"T\",\"amino_acids\":\"G/D\",\"gene_symbol_source\":\"HGNC\",\"polyphen_score\":0.995,\"gene_id\":\"ENSG00000175164\",\"sift_score\":0,\"impact\":\"MODERATE\",\"transcript_id\":\"ENST00000538324\",\"protein_start\":229,\"sift_prediction\":\"deleterious\",\"codons\":\"gGc/gAc\",\"strand\":-1,\"gene_symbol\":\"ABO\",\"polyphen_prediction\":\"probably_damaging\",\"cdna_start\":711,\"protein_end\":229,\"consequence_terms\":[\"missense_variant\"],\"hgnc_id\":\"HGNC:79\"},{\"gene_id\":\"ENSG00000175164\",\"sift_score\":0,\"impact\":\"MODERATE\",\"protein_start\":229,\"transcript_id\":\"ENST00000611156\",\"sift_prediction\":\"deleterious\",\"codons\":\"gGc/gAc\",\"strand\":-1,\"gene_symbol\":\"ABO\",\"polyphen_prediction\":\"probably_damaging\",\"cdna_start\":711,\"protein_end\":229,\"consequence_terms\":[\"missense_variant\"],\"hgnc_id\":\"HGNC:79\",\"cds_end\":686,\"cdna_end\":711,\"cds_start\":686,\"biotype\":\"protein_coding\",\"variant_allele\":\"T\",\"amino_acids\":\"G/D\",\"gene_symbol_source\":\"HGNC\",\"polyphen_score\":0.998},{\"transcript_id\":\"ENST00000647353\",\"biotype\":\"processed_transcript\",\"strand\":-1,\"impact\":\"MODIFIER\",\"gene_id\":\"ENSG00000175164\",\"consequence_terms\":[\"intron_variant\",\"non_coding_transcript_variant\"],\"hgnc_id\":\"HGNC:79\",\"gene_symbol_source\":\"HGNC\",\"variant_allele\":\"T\",\"gene_symbol\":\"ABO\"},{\"transcript_id\":\"ENST00000651471\",\"biotype\":\"processed_transcript\",\"strand\":-1,\"impact\":\"MODIFIER\",\"gene_id\":\"ENSG00000175164\",\"hgnc_id\":\"HGNC:79\",\"consequence_terms\":[\"downstream_gene_variant\"],\"gene_symbol_source\":\"HGNC\",\"variant_allele\":\"T\",\"distance\":64,\"gene_symbol\":\"ABO\"},{\"gene_symbol\":\"ABO\",\"variant_allele\":\"T\",\"hgnc_id\":\"HGNC:79\",\"consequence_terms\":[\"intron_variant\"],\"gene_symbol_source\":\"HGNC\",\"gene_id\":\"ENSG00000175164\",\"strand\":-1,\"biotype\":\"protein_coding\",\"transcript_id\":\"ENST00000679909\",\"impact\":\"MODIFIER\"},{\"distance\":3445,\"gene_symbol\":\"ABO\",\"variant_allele\":\"T\",\"consequence_terms\":[\"upstream_gene_variant\"],\"hgnc_id\":\"HGNC:79\",\"gene_symbol_source\":\"HGNC\",\"gene_id\":\"ENSG00000175164\",\"strand\":-1,\"biotype\":\"processed_transcript\",\"transcript_id\":\"ENST00000680600\",\"impact\":\"MODIFIER\"}],\"colocated_variants\":[{\"seq_region_name\":\"9\",\"end\":133256042,\"frequencies\":{\"T\":{\"gnomade_amr\":0.002312,\"eur\":0.0109,\"eas\":0,\"gnomadg_eas\":0,\"gnomade_asj\":0.002299,\"gnomadg_fin\":0.01281,\"gnomadg_ami\":0,\"amr\":0.0014,\"afr\":0,\"gnomade_afr\":0.0007352,\"gnomadg\":0.003052,\"gnomade_fin\":0.01381,\"gnomadg_oth\":0.00239,\"gnomadg_nfe\":0.003927,\"gnomade_nfe\":0.003597,\"gnomade_sas\":0.001379,\"gnomade_oth\":0.005249,\"gnomadg_sas\":0.0006206,\"gnomadg_afr\":0.0005082,\"gnomade\":0.003617,\"gnomadg_asj\":0.001441,\"gnomadg_mid\":0,\"gnomade_eas\":0,\"gnomadg_amr\":0.001767,\"sas\":0.001}},\"allele_string\":\"C/T\",\"id\":\"rs56116432\",\"start\":133256042,\"strand\":1}],\"seq_region_name\":\"9\",\"start\":133256042,\"id\":\"rs56116432\"},{\"allele_string\":\"C/T\",\"strand\":1,\"input\":\"rs56116432\",\"end\":133256189,\"assembly_name\":\"GRCh38\",\"most_severe_consequence\":\"missense_variant\",\"id\":\"rs56116432\",\"start\":133256189,\"seq_region_name\":\"CHR_HG2030_PATCH\",\"colocated_variants\":[{\"end\":133256189,\"seq_region_name\":\"CHR_HG2030_PATCH\",\"strand\":1,\"id\":\"rs56116432\",\"start\":133256189,\"allele_string\":\"C/T\"}],\"transcript_consequences\":[{\"consequence_terms\":[\"missense_variant\"],\"hgnc_id\":\"HGNC:79\",\"protein_end\":230,\"cdna_start\":689,\"polyphen_prediction\":\"probably_damaging\",\"gene_symbol\":\"ABO\",\"codons\":\"gGc/gAc\",\"sift_prediction\":\"deleterious\",\"strand\":-1,\"protein_start\":230,\"transcript_id\":\"ENST00000626615\",\"impact\":\"MODERATE\",\"sift_score\":0,\"gene_id\":\"ENSG00000281879\",\"polyphen_score\":0.996,\"gene_symbol_source\":\"HGNC\",\"amino_acids\":\"G/D\",\"variant_allele\":\"T\",\"biotype\":\"protein_coding\",\"cds_start\":689,\"cdna_end\":689,\"cds_end\":689},{\"gene_id\":\"ENSG00000281879\",\"sift_score\":0,\"impact\":\"MODERATE\",\"protein_start\":230,\"transcript_id\":\"ENST00000644422\",\"strand\":-1,\"sift_prediction\":\"deleterious\",\"codons\":\"gGc/gAc\",\"gene_symbol\":\"ABO\",\"polyphen_prediction\":\"probably_damaging\",\"cdna_start\":701,\"protein_end\":230,\"consequence_terms\":[\"missense_variant\"],\"hgnc_id\":\"HGNC:79\",\"cds_end\":689,\"cdna_end\":701,\"cds_start\":689,\"biotype\":\"protein_coding\",\"amino_acids\":\"G/D\",\"variant_allele\":\"T\",\"gene_symbol_source\":\"HGNC\",\"polyphen_score\":0.996},{\"gene_symbol\":\"ABO\",\"polyphen_prediction\":\"probably_damaging\",\"cdna_start\":689,\"protein_end\":230,\"consequence_terms\":[\"missense_variant\"],\"hgnc_id\":\"HGNC:79\",\"gene_id\":\"ENSG00000281879\",\"sift_score\":0,\"impact\":\"MODERATE\",\"protein_start\":230,\"transcript_id\":\"ENST00000644755\",\"strand\":-1,\"sift_prediction\":\"deleterious\",\"codons\":\"gGc/gAc\",\"variant_allele\":\"T\",\"amino_acids\":\"G/D\",\"gene_symbol_source\":\"HGNC\",\"polyphen_score\":0.996,\"cds_end\":689,\"cdna_end\":689,\"cds_start\":689,\"biotype\":\"protein_coding\"},{\"hgnc_id\":\"HGNC:79\",\"consequence_terms\":[\"missense_variant\"],\"protein_end\":230,\"cdna_start\":701,\"polyphen_prediction\":\"probably_damaging\",\"gene_symbol\":\"ABO\",\"sift_prediction\":\"deleterious\",\"codons\":\"gGc/gAc\",\"strand\":-1,\"transcript_id\":\"ENST00000645810\",\"protein_start\":230,\"impact\":\"MODERATE\",\"sift_score\":0,\"gene_id\":\"ENSG00000281879\",\"polyphen_score\":0.996,\"gene_symbol_source\":\"HGNC\",\"variant_allele\":\"T\",\"amino_acids\":\"G/D\",\"biotype\":\"protein_coding\",\"cds_start\":689,\"cdna_end\":701,\"cds_end\":689}]},{\"assembly_name\":\"GRCh38\",\"most_severe_consequence\":\"?\",\"strand\":1,\"allele_string\":\"COSMIC_MUTATION\",\"input\":\"COSM476\",\"end\":140753336,\"seq_region_name\":\"7\",\"colocated_variants\":[{\"allele_string\":\"HGMD_MUTATION\",\"strand\":1,\"start\":140753336,\"id\":\"CM112509\",\"seq_region_name\":\"7\",\"end\":140753336,\"phenotype_or_disease\":1},{\"strand\":1,\"id\":\"COSV56056643\",\"start\":140753336,\"allele_string\":\"COSMIC_MUTATION\",\"var_synonyms\":{\"COSMIC\":[\"COSM476\"]},\"phenotype_or_disease\":1,\"somatic\":1,\"end\":140753336,\"seq_region_name\":\"7\"},{\"start\":140753336,\"id\":\"COSV56065204\",\"strand\":1,\"allele_string\":\"COSMIC_MUTATION\",\"var_synonyms\":{\"COSMIC\":[\"COSM18443\"]},\"phenotype_or_disease\":1,\"end\":140753336,\"somatic\":1,\"seq_region_name\":\"7\"},{\"strand\":1,\"id\":\"COSV56080151\",\"start\":140753336,\"var_synonyms\":{\"COSMIC\":[\"COSM6137\"]},\"allele_string\":\"COSMIC_MUTATION\",\"phenotype_or_disease\":1,\"somatic\":1,\"end\":140753336,\"seq_region_name\":\"7\"}],\"id\":\"COSM476\",\"start\":140753336}]"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!wget -q --header='Content-type:application/json' --header='Accept:application/json' \\\n",
        "--post-data='{ \"ids\" : [\"rs56116432\", \"COSM476\" ] }' \\\n",
        "'https://rest.ensembl.org/vep/human/id'  -O -"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "_tHgaylXg2xx",
        "outputId": "53ce7c95-c5be-47b4-e600-acdfd64a21e2"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "[{\"transcript_consequences\":[{\"consequence_terms\":[\"non_coding_transcript_exon_variant\"],\"variant_allele\":\"T\",\"gene_symbol_source\":\"HGNC\",\"hgnc_id\":\"HGNC:79\",\"cdna_end\":718,\"gene_symbol\":\"ABO\",\"gene_id\":\"ENSG00000175164\",\"strand\":-1,\"cdna_start\":718,\"impact\":\"MODIFIER\",\"biotype\":\"processed_transcript\",\"transcript_id\":\"ENST00000453660\"},{\"codons\":\"gGc/gAc\",\"cdna_end\":711,\"amino_acids\":\"G/D\",\"variant_allele\":\"T\",\"consequence_terms\":[\"missense_variant\"],\"gene_symbol_source\":\"HGNC\",\"transcript_id\":\"ENST00000538324\",\"gene_id\":\"ENSG00000175164\",\"impact\":\"MODERATE\",\"hgnc_id\":\"HGNC:79\",\"sift_prediction\":\"deleterious\",\"cds_start\":686,\"polyphen_score\":0.995,\"cds_end\":686,\"sift_score\":0,\"protein_end\":229,\"protein_start\":229,\"biotype\":\"protein_coding\",\"gene_symbol\":\"ABO\",\"polyphen_prediction\":\"probably_damaging\",\"strand\":-1,\"cdna_start\":711},{\"amino_acids\":\"G/D\",\"cdna_end\":711,\"codons\":\"gGc/gAc\",\"consequence_terms\":[\"missense_variant\"],\"variant_allele\":\"T\",\"gene_symbol_source\":\"HGNC\",\"transcript_id\":\"ENST00000611156\",\"gene_id\":\"ENSG00000175164\",\"impact\":\"MODERATE\",\"sift_prediction\":\"deleterious\",\"hgnc_id\":\"HGNC:79\",\"cds_start\":686,\"polyphen_score\":0.998,\"sift_score\":0,\"cds_end\":686,\"protein_end\":229,\"protein_start\":229,\"biotype\":\"protein_coding\",\"gene_symbol\":\"ABO\",\"polyphen_prediction\":\"probably_damaging\",\"strand\":-1,\"cdna_start\":711},{\"hgnc_id\":\"HGNC:79\",\"gene_symbol_source\":\"HGNC\",\"consequence_terms\":[\"intron_variant\",\"non_coding_transcript_variant\"],\"variant_allele\":\"T\",\"transcript_id\":\"ENST00000647353\",\"biotype\":\"processed_transcript\",\"impact\":\"MODIFIER\",\"gene_id\":\"ENSG00000175164\",\"strand\":-1,\"gene_symbol\":\"ABO\"},{\"gene_symbol_source\":\"HGNC\",\"consequence_terms\":[\"downstream_gene_variant\"],\"variant_allele\":\"T\",\"distance\":64,\"hgnc_id\":\"HGNC:79\",\"impact\":\"MODIFIER\",\"gene_symbol\":\"ABO\",\"strand\":-1,\"gene_id\":\"ENSG00000175164\",\"transcript_id\":\"ENST00000651471\",\"biotype\":\"processed_transcript\"},{\"impact\":\"MODIFIER\",\"strand\":-1,\"gene_id\":\"ENSG00000175164\",\"gene_symbol\":\"ABO\",\"transcript_id\":\"ENST00000679909\",\"biotype\":\"protein_coding\",\"gene_symbol_source\":\"HGNC\",\"consequence_terms\":[\"intron_variant\"],\"variant_allele\":\"T\",\"hgnc_id\":\"HGNC:79\"},{\"strand\":-1,\"gene_id\":\"ENSG00000175164\",\"gene_symbol\":\"ABO\",\"impact\":\"MODIFIER\",\"biotype\":\"processed_transcript\",\"transcript_id\":\"ENST00000680600\",\"consequence_terms\":[\"upstream_gene_variant\"],\"variant_allele\":\"T\",\"gene_symbol_source\":\"HGNC\",\"hgnc_id\":\"HGNC:79\",\"distance\":3445}],\"strand\":1,\"seq_region_name\":\"9\",\"most_severe_consequence\":\"missense_variant\",\"assembly_name\":\"GRCh38\",\"input\":\"rs56116432\",\"start\":133256042,\"colocated_variants\":[{\"strand\":1,\"end\":133256042,\"seq_region_name\":\"9\",\"frequencies\":{\"T\":{\"eas\":0,\"gnomadg_oth\":0.00239,\"amr\":0.0014,\"gnomadg_ami\":0,\"sas\":0.001,\"gnomadg_eas\":0,\"gnomade_sas\":0.001379,\"gnomade_afr\":0.0007352,\"eur\":0.0109,\"gnomade_oth\":0.005249,\"gnomade_fin\":0.01381,\"gnomadg_mid\":0,\"gnomadg\":0.003052,\"gnomade_nfe\":0.003597,\"gnomadg_amr\":0.001767,\"gnomade_amr\":0.002312,\"gnomadg_fin\":0.01281,\"gnomadg_sas\":0.0006206,\"gnomadg_nfe\":0.003927,\"gnomadg_afr\":0.0005082,\"gnomadg_asj\":0.001441,\"afr\":0,\"gnomade_eas\":0,\"gnomade_asj\":0.002299,\"gnomade\":0.003617}},\"start\":133256042,\"allele_string\":\"C/T\",\"id\":\"rs56116432\"}],\"allele_string\":\"C/T\",\"end\":133256042,\"id\":\"rs56116432\"},{\"most_severe_consequence\":\"missense_variant\",\"seq_region_name\":\"CHR_HG2030_PATCH\",\"assembly_name\":\"GRCh38\",\"strand\":1,\"transcript_consequences\":[{\"transcript_id\":\"ENST00000626615\",\"impact\":\"MODERATE\",\"gene_id\":\"ENSG00000281879\",\"cdna_end\":689,\"amino_acids\":\"G/D\",\"codons\":\"gGc/gAc\",\"gene_symbol_source\":\"HGNC\",\"consequence_terms\":[\"missense_variant\"],\"variant_allele\":\"T\",\"biotype\":\"protein_coding\",\"protein_start\":230,\"cdna_start\":689,\"strand\":-1,\"gene_symbol\":\"ABO\",\"polyphen_prediction\":\"probably_damaging\",\"polyphen_score\":0.996,\"cds_start\":689,\"hgnc_id\":\"HGNC:79\",\"sift_prediction\":\"deleterious\",\"cds_end\":689,\"sift_score\":0,\"protein_end\":230},{\"amino_acids\":\"G/D\",\"cdna_end\":701,\"codons\":\"gGc/gAc\",\"gene_symbol_source\":\"HGNC\",\"consequence_terms\":[\"missense_variant\"],\"variant_allele\":\"T\",\"transcript_id\":\"ENST00000644422\",\"impact\":\"MODERATE\",\"gene_id\":\"ENSG00000281879\",\"cds_start\":689,\"polyphen_score\":0.996,\"hgnc_id\":\"HGNC:79\",\"sift_prediction\":\"deleterious\",\"sift_score\":0,\"protein_end\":230,\"cds_end\":689,\"protein_start\":230,\"biotype\":\"protein_coding\",\"cdna_start\":701,\"gene_symbol\":\"ABO\",\"polyphen_prediction\":\"probably_damaging\",\"strand\":-1},{\"strand\":-1,\"gene_symbol\":\"ABO\",\"polyphen_prediction\":\"probably_damaging\",\"cdna_start\":689,\"biotype\":\"protein_coding\",\"protein_start\":230,\"protein_end\":230,\"sift_score\":0,\"cds_end\":689,\"hgnc_id\":\"HGNC:79\",\"sift_prediction\":\"deleterious\",\"polyphen_score\":0.996,\"cds_start\":689,\"gene_id\":\"ENSG00000281879\",\"impact\":\"MODERATE\",\"transcript_id\":\"ENST00000644755\",\"variant_allele\":\"T\",\"consequence_terms\":[\"missense_variant\"],\"gene_symbol_source\":\"HGNC\",\"codons\":\"gGc/gAc\",\"cdna_end\":689,\"amino_acids\":\"G/D\"},{\"sift_score\":0,\"protein_end\":230,\"cds_end\":689,\"hgnc_id\":\"HGNC:79\",\"sift_prediction\":\"deleterious\",\"cds_start\":689,\"polyphen_score\":0.996,\"gene_symbol\":\"ABO\",\"polyphen_prediction\":\"probably_damaging\",\"strand\":-1,\"cdna_start\":701,\"protein_start\":230,\"biotype\":\"protein_coding\",\"consequence_terms\":[\"missense_variant\"],\"variant_allele\":\"T\",\"gene_symbol_source\":\"HGNC\",\"cdna_end\":701,\"amino_acids\":\"G/D\",\"codons\":\"gGc/gAc\",\"gene_id\":\"ENSG00000281879\",\"impact\":\"MODERATE\",\"transcript_id\":\"ENST00000645810\"}],\"input\":\"rs56116432\",\"start\":133256189,\"allele_string\":\"C/T\",\"colocated_variants\":[{\"seq_region_name\":\"CHR_HG2030_PATCH\",\"allele_string\":\"C/T\",\"id\":\"rs56116432\",\"start\":133256189,\"strand\":1,\"end\":133256189}],\"end\":133256189,\"id\":\"rs56116432\"},{\"strand\":1,\"assembly_name\":\"GRCh38\",\"seq_region_name\":\"7\",\"most_severe_consequence\":\"?\",\"start\":140753336,\"input\":\"COSM476\",\"colocated_variants\":[{\"allele_string\":\"HGMD_MUTATION\",\"id\":\"CM112509\",\"start\":140753336,\"seq_region_name\":\"7\",\"end\":140753336,\"phenotype_or_disease\":1,\"strand\":1},{\"end\":140753336,\"phenotype_or_disease\":1,\"strand\":1,\"allele_string\":\"COSMIC_MUTATION\",\"id\":\"COSV56056643\",\"var_synonyms\":{\"COSMIC\":[\"COSM476\"]},\"somatic\":1,\"start\":140753336,\"seq_region_name\":\"7\"},{\"id\":\"COSV56065204\",\"allele_string\":\"COSMIC_MUTATION\",\"somatic\":1,\"start\":140753336,\"var_synonyms\":{\"COSMIC\":[\"COSM18443\"]},\"seq_region_name\":\"7\",\"end\":140753336,\"phenotype_or_disease\":1,\"strand\":1},{\"seq_region_name\":\"7\",\"var_synonyms\":{\"COSMIC\":[\"COSM6137\"]},\"start\":140753336,\"somatic\":1,\"allele_string\":\"COSMIC_MUTATION\",\"id\":\"COSV56080151\",\"strand\":1,\"phenotype_or_disease\":1,\"end\":140753336}],\"allele_string\":\"COSMIC_MUTATION\",\"end\":140753336,\"id\":\"COSM476\"}]"
          ]
        }
      ]
    }
  ]
}